## Making Plots and Tables for the Final Manuscript

The following code details the creation of the final tables and plots that were included in the manuscript (including supplementary material). 

These include the following figures:

- Figure 1 = Forest plot(s) of genetic effect on age at onset, A = full PRS model, B = PRS less ABCG2.
    
- Figure 2 = Effect of individual ABCG2 variants, A = rs2231142, B = rs10011796.

- Figure 3 = Forest plot of PRS effect on presence of tophaceous gout (two panels, before and after adjustment for disease duration).

- Supp Figure 1 = Manhattan plot for gout GWAS.

- Supp Figure 2 = Tophaceous disease ABCG2 models and PRS without ABCG2.

- Supp Figure 3 = Relationship between flares and PRS (plot as faceted scatter plot, and/or as grouped category plots).

And the following tables:

- Table 1 = Gout cohort statistics/demographics (only key variables).

- Supp Table 1 = Table describing GWAS results and which SNPs are part of the PRS.

- Supp Table 2 = Missing data.

- Supp Table 3 = Full cohort stats.

- Supp Table 4 = All model results in table format.

There are also several other analyses included at the end of the document.

```{r dataprep4}
# Datasets
load(path(scratch_path, "Output/all_pheno_prs.RData"))

# Making a categorical flare frequency variable (FLARE_CAT) and setting all control gout severity traits to NA, then preparing the data further for models.
all_pheno_prs <- all_pheno_prs %>%
  mutate(FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"),
                                       "One every one to two months",
                                       "One or more per month"),
                            labels = c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52"),
                            ordered = TRUE),
         AGE1ATK = case_when(GOUT ~ AGE1ATK),
         DURATION = case_when(GOUT ~ DURATION),
         NUMATK = case_when(GOUT ~ NUMATK),
         TOPHIGOUT = case_when(GOUT ~ TOPHIGOUT),
         ULT = case_when(GOUT ~ ULT),
         SEX = factor(SEX, levels = c("Male", "Female")),
         GOUT2 = factor(GOUT, levels = c(TRUE, FALSE), labels = c("Gout", "Control")),
         ANCESTRY_GOUT = factor(case_when(GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "European Gout",
                                          !GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "European Control",
                                          GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian Gout",
                                          !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian Control",
                                          GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian Gout - NP",
                                          !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian Control - NP",
                                          GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian Gout",
                                          !GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian Control"),
                                levels = c("European Gout", 
                                           "European Control", 
                                           "East Polynesian Gout", 
                                           "East Polynesian Control", 
                                           "East Polynesian Gout - NP", 
                                           "East Polynesian Control - NP", 
                                           "West Polynesian Gout", 
                                           "West Polynesian Control")),
         ANCESTRY_GOUT_SEX = factor(case_when(ANCESTRY_GOUT == "European Gout" & SEX == "Male" ~ "European Gout - male",
                                              ANCESTRY_GOUT == "European Gout" & SEX == "Female" ~ "European Gout - female",
                                              ANCESTRY_GOUT == "European Control" & SEX == "Male" ~ "European Control - male",
                                              ANCESTRY_GOUT == "European Control" & SEX == "Female" ~ "European Control - female",
                                              ANCESTRY_GOUT == "East Polynesian Gout" & SEX == "Male" ~ "East Polynesian Gout - male",
                                              ANCESTRY_GOUT == "East Polynesian Gout" & SEX == "Female" ~ "East Polynesian Gout - female",
                                              ANCESTRY_GOUT == "East Polynesian Control" & SEX == "Male" ~ "East Polynesian Control - male",
                                              ANCESTRY_GOUT == "East Polynesian Control" & SEX == "Female" ~ "East Polynesian Control - female",
                                              ANCESTRY_GOUT == "East Polynesian Gout - NP" & SEX == "Male" ~ "East Polynesian Gout - NP - male",
                                              ANCESTRY_GOUT == "East Polynesian Gout - NP" & SEX == "Female" ~ "East Polynesian Gout - NP - female",
                                              ANCESTRY_GOUT == "East Polynesian Control - NP" & SEX == "Male" ~ "East Polynesian Control - NP - male",
                                              ANCESTRY_GOUT == "East Polynesian Control - NP" & SEX == "Female" ~ "East Polynesian Control - NP - female",
                                              ANCESTRY_GOUT == "West Polynesian Gout" & SEX == "Male" ~ "West Polynesian Gout - male",
                                              ANCESTRY_GOUT == "West Polynesian Gout" & SEX == "Female" ~ "West Polynesian Gout - female",
                                              ANCESTRY_GOUT == "West Polynesian Control" & SEX == "Male" ~ "West Polynesian Control - male",
                                              ANCESTRY_GOUT == "West Polynesian Control" & SEX == "Female" ~ "West Polynesian Control - female"), 
                                    levels = c("European Gout - male", 
                                               "European Gout - female", 
                                               "European Control - male", 
                                               "European Control - female", 
                                               "East Polynesian Gout - male", 
                                               "East Polynesian Gout - female", 
                                               "East Polynesian Control - male", 
                                               "East Polynesian Control - female", 
                                               "East Polynesian Gout - NP - male", 
                                               "East Polynesian Gout - NP - female", 
                                               "East Polynesian Control - NP - male", 
                                               "East Polynesian Control - NP - female", 
                                               "West Polynesian Gout - male", 
                                               "West Polynesian Gout - female", 
                                               "West Polynesian Control - male", 
                                               "West Polynesian Control - female")),
         SEX_GOUT = factor(case_when(GOUT & SEX == "Male" ~ "Male Gout",
                                     GOUT & SEX == "Female" ~ "Female Gout",
                                     !GOUT & SEX == "Male" ~ "Male Control",
                                     !GOUT & SEX == "Female" ~ "Female Control"), 
                           levels = c("Male Gout", 
                                      "Female Gout",
                                      "Male Control",
                                      "Female Control")),
         COHORT_GOUT = factor(case_when(GOUT & Pheno.Study == "UK Biobank" ~ "UK Biobank - Gout",
                                        !GOUT & Pheno.Study == "UK Biobank" ~ "UK Biobank - Control",
                                        GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") & Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease") ~ "Aus/NZ - Gout",
                                        !GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") & Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease") ~ "Aus/NZ - Control",
                                        GOUT & Pheno.Study == "EuroGout" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "GlobalGout - Gout",
                                        !GOUT & Pheno.Study == "EuroGout" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "GlobalGout - Control",
                                        Pheno.Study == "Ardea: 401" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - LASSO",
                                        Pheno.Study == "Ardea: CLEAR1" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CLEAR1",
                                        Pheno.Study == "Ardea: CLEAR2" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CLEAR2",
                                        Pheno.Study == "Ardea: CRYSTAL" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CRYSTAL",
                                        Pheno.Study == "Ardea: LIGHT" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - LIGHT",
                                        GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian - Gout",
                                        !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian - Control",
                                        GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian - Gout - NP",
                                        !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian - Control - NP",
                                        GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian - Gout",
                                        !GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian - Control"), 
                              levels = c("UK Biobank - Gout", 
                                         "UK Biobank - Control", 
                                         "Aus/NZ - Gout", 
                                         "Aus/NZ - Control", 
                                         "GlobalGout - Gout", 
                                         "GlobalGout - Control", 
                                         "Ardea - LASSO", 
                                         "Ardea - CLEAR1", 
                                         "Ardea - CLEAR2", 
                                         "Ardea - CRYSTAL", 
                                         "Ardea - LIGHT", 
                                         "East Polynesian - Gout", 
                                         "East Polynesian - Control", 
                                         "East Polynesian - Gout - NP", 
                                         "East Polynesian - Control - NP", 
                                         "West Polynesian - Gout", 
                                         "West Polynesian - Control"))) %>% 
  filter(!is.na(COHORT_GOUT),
         !is.na(AGECOL),
         !is.na(PRS),
         (GOUT & !(is.na(AGE1ATK) & is.na(NUMATK) & is.na(TOPHIGOUT)) | Pheno.Study == "UK Biobank" | !GOUT))

# Making list with each of the cohort subsets. This will make it easier to manipulate the data for models.
data_list <- list("UK Biobank - Gout - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           GOUT,
                                           Pheno.Study == "UK Biobank"),
                  "UK Biobank - Gout - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           GOUT,
                                           Pheno.Study == "UK Biobank"),
                  "UK Biobank - Control - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           !GOUT,
                                           Pheno.Study == "UK Biobank"),
                  "UK Biobank - Control - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           !GOUT,
                                           Pheno.Study == "UK Biobank"),
                  "Aus/NZ European - Gout - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                           Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "Aus/NZ European - Gout - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                           Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "Aus/NZ European - Control - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                           Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "Aus/NZ European - Control - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                           Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "GlobalGout - Gout - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           GOUT,
                                           Pheno.Study == "EuroGout",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "GlobalGout - Gout - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           GOUT,
                                           Pheno.Study == "EuroGout",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "GlobalGout - Control - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           !GOUT,
                                           Pheno.Study == "EuroGout",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "GlobalGout - Control - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           !GOUT,
                                           Pheno.Study == "EuroGout",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LASSO - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           Pheno.Study == "Ardea: 401",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LASSO - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           Pheno.Study == "Ardea: 401",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR1 - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           Pheno.Study == "Ardea: CLEAR1",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR1 - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           Pheno.Study == "Ardea: CLEAR1",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR2 - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           Pheno.Study == "Ardea: CLEAR2",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR2 - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           Pheno.Study == "Ardea: CLEAR2",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CRYSTAL - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           Pheno.Study == "Ardea: CRYSTAL",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CRYSTAL - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           Pheno.Study == "Ardea: CRYSTAL",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LIGHT - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           Pheno.Study == "Ardea: LIGHT",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LIGHT - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           Pheno.Study == "Ardea: LIGHT",
                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "East Polynesian - Gout - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study != "Ngati Porou"),
                  "East Polynesian - Gout - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study != "Ngati Porou"),
                  "East Polynesian - Control - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study != "Ngati Porou"),
                  "East Polynesian - Control - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study != "Ngati Porou"),
                  "East Polynesian - Gout - NP - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study == "Ngati Porou"),
                  "East Polynesian - Gout - NP - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study == "Ngati Porou"),
                  "East Polynesian - Control - NP - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study == "Ngati Porou"),
                  "East Polynesian - Control - NP - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("East Polynesian"),
                                           Pheno.Study == "Ngati Porou"),
                  "West Polynesian - Gout - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  "West Polynesian - Gout - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           GOUT,
                                           Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  "West Polynesian - Control - Male" = all_pheno_prs %>% filter(SEX == "Male",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  "West Polynesian - Control - Female" = all_pheno_prs %>% filter(SEX == "Female",
                                           !GOUT,
                                           Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")))

cohortstring <- c("UK Biobank - Gout - Male",
                  "UK Biobank - Gout - Female",
                  "UK Biobank - Control - Male",
                  "UK Biobank - Control - Female",
                  "Aus/NZ European - Gout - Male",
                  "Aus/NZ European - Gout - Female",
                  "Aus/NZ European - Control - Male",
                  "Aus/NZ European - Control - Female",
                  "GlobalGout - Gout - Male",
                  "GlobalGout - Gout - Female",
                  "GlobalGout - Control - Male",
                  "GlobalGout - Control - Female",
                  "Ardea - LASSO - Male",
                  "Ardea - LASSO - Female",
                  "Ardea - CLEAR1 - Male",
                  "Ardea - CLEAR1 - Female",
                  "Ardea - CLEAR2 - Male",
                  "Ardea - CLEAR2 - Female",
                  "Ardea - CRYSTAL - Male",
                  "Ardea - CRYSTAL - Female",
                  "Ardea - LIGHT - Male",
                  "Ardea - LIGHT - Female",
                  "East Polynesian - Gout - Male",
                  "East Polynesian - Gout - Female",
                  "East Polynesian - Control - Male",
                  "East Polynesian - Control - Female",
                  "East Polynesian - Gout - NP - Male",
                  "East Polynesian - Gout - NP - Female",
                  "East Polynesian - Control - NP - Male",
                  "East Polynesian - Control - NP - Female",
                  "West Polynesian - Gout - Male",
                  "West Polynesian - Gout - Female",
                  "West Polynesian - Control - Male",
                  "West Polynesian - Control - Female")

clean_names <- c("UK&nbsp;Biobank<br>Gout<br>Male",
                 "UK&nbsp;Biobank<br>Gout<br>Female",
                 "UK&nbsp;Biobank<br>Control<br>Male",
                 "UK&nbsp;Biobank<br>Control<br>Female",
                 "Aus/NZ&nbsp;European<br>Gout<br>Male",
                 "Aus/NZ&nbsp;European<br>Gout<br>Female",
                 "Aus/NZ&nbsp;European<br>Control<br>Male",
                 "Aus/NZ&nbsp;European<br>Control<br>Female",
                 "GlobalGout<br>Gout<br>Male",
                 "GlobalGout<br>Gout<br>Female",
                 "GlobalGout<br>Control<br>Male",
                 "GlobalGout<br>Control<br>Female",
                 "Ardea<br>LASSO<br>Gout<br>Male",
                 "Ardea<br>LASSO<br>Gout<br>Female",
                 "Ardea<br>CLEAR1<br>Gout<br>Male",
                 "Ardea<br>CLEAR1<br>Gout<br>Female",
                 "Ardea<br>CLEAR2<br>Gout<br>Male",
                 "Ardea<br>CLEAR2<br>Gout<br>Female",
                 "Ardea<br>CRYSTAL<br>Gout<br>Male",
                 "Ardea<br>CRYSTAL<br>Gout<br>Female",
                 "Ardea<br>LIGHT<br>Gout<br>Male",
                 "Ardea<br>LIGHT<br>Gout<br>Female",
                 "East&nbsp;Polynesian<br>Gout<br>Male",
                 "East&nbsp;Polynesian<br>Gout<br>Female",
                 "East&nbsp;Polynesian<br>Control<br>Male",
                 "East&nbsp;Polynesian<br>Control<br>Female",
                 "East&nbsp;Polynesian<br>Gout<br>Male<br>NP",
                 "East&nbsp;Polynesian<br>Gout<br>Female<br>NP",
                 "East&nbsp;Polynesian<br>Control<br>Male<br>NP",
                 "East&nbsp;Polynesian<br>Control<br>Female<br>NP",
                 "West&nbsp;Polynesian<br>Gout<br>Male",
                 "West&nbsp;Polynesian<br>Gout<br>Female",
                 "West&nbsp;Polynesian<br>Control<br>Male",
                 "West&nbsp;Polynesian<br>Control<br>Female")

# Loading GWAS summary tables.
load(here("Output/UKBB_Gene_OR.RData"))
load(here("Output/Tin_Gene_OR.RData"))

# Loading model results.
load(here("Output/GoutModels.RData"))
load(here("Output/OnsetModels.RData"))
load(here("Output/TophiModels.RData"))
```

```{r functions4}
# Defining functions for use throughout this section.
# This function will take any numeric variable and produce a report of the form "<mean> ± <sd>".
report <- function(x) {
    if(sum(is.na(x)) != length(x)) {
      paste0(sprintf(mean(x, na.rm = TRUE), fmt = "%#.1f"), " ± ", sprintf(sd(x, na.rm = TRUE), fmt = "%#.1f"))
    } else {
      paste0("NA")
    }
}

# This function will take any numeric variable and produce a report of the form "<median> (<lower quartile> - <upper quartile>)".
report_median <- function(x) {
    if(sum(is.na(x)) != length(x)) {
      paste0(median(x, na.rm =T), " (", summary(x)[[2]], " - ", summary(x)[[5]], ")")
    } else {
      paste0("NA")
    }
}

# This function will take any TRUE/FALSE variable and produce a report of the form "<N TRUE> (<% TRUE>)".
sumreport <- function(x) {
  if(sum(is.na(x)) != length(x)){
    paste0(sum(x, na.rm = TRUE), " (", sprintf((mean(x, na.rm = TRUE) * 100), fmt = "%#.1f"), ")")
  } else {
      paste0("NA")
  }
}

# This function transposes a dataframe.
transpose_df <- function(df) {
  t_df <- data.table::transpose(df)
  colnames(t_df) <- rownames(df)
  rownames(t_df) <- colnames(df)
  t_df <- t_df %>%
    rownames_to_column() %>%
    as_tibble() %>% 
    row_to_names(row_number = 1)
  return(t_df)
}

# This function will take any variable and report the missingness as either "All", "None", or the form "<N missing> (<% missing>)".
missing <- function(x){
  if(sum(is.na(x)) == length(x)) {
    return("All")
    } else if(sum(!is.na(x)) == length(x)){
      return("None")
      } else {
      paste0(format(sum(is.na(x)), big.mark = ","), " (", format(round((sum(is.na(x)) / length(x) * 100), digits = 1), nsmall = 1), ")")
  }
}
```

### Main figures and tables

The following forest plots are for the effect of the various PRS or individual variants on either age at onset or tophaceous disease, with tophaceous disease models additionally adjusted for duration. The individual variant models require P < 0.05/2 for significance. Where appropriate, these were tested via meta-analysis to reflect the results from the full PRS models. 

Initially, I plotted the effect of the PRS, PRS less ABCG2 and the urate PRS on age at onset.

```{r OnsetvsPRSfigs, eval = F}
# Preparing age at onset vs PRS model data for meta-analysis.
tmp <- OnsetModels %>% 
  filter(Predictor == "PRS") %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         Beta = case_when(is.na(LCL) ~ NA_real_,
                          TRUE ~ Beta),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male                  N",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"),
         N = format(N, big.mark = ","))

# Running meta-analysis.
onset <- metagen(TE = Beta, 
                 seTE = SE, 
                 studlab = Cohort,
                 subgroup = Label,
                 data = tmp)

# Preparing to produce a tiff file of the following plot.
tiff(file = here("Output/Plots/OnsetvsPRS.tiff"), units = "in", width = 7.0, height = 8.5, res = 300)

# Producing the forest plot for output in the tiff.
forest(onset, 
       xlim = c(-15, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T,
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of Gout PRS on Age at Gout Onset\nIncluding ABCG2\n", ""),
       rightlabs = c("Beta (years)", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F)

# Saving the tiff file.
dev.off()

# Preparing age at onset vs PRS less ABCG2 model data for meta-analysis.
tmp <- OnsetModels %>% 
  filter(Predictor == "PRS_noABCG2") %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         Beta = case_when(is.na(LCL) ~ NA_real_,
                          TRUE ~ Beta),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male                  N",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"),
         N = format(N, big.mark = ","))

# Running meta-analysis.
onset <- metagen(TE = Beta, 
                 seTE = SE, 
                 studlab = Cohort,
                 subgroup = Label,
                 data = tmp)

# Preparing to produce a tiff file of the following plot.
tiff(file = here("Output/Plots/OnsetvsPRS2.tiff"), units = "in", width = 7.0, height = 8.5, res = 300)

# Producing the forest plot for output in the tiff.
forest(onset, 
       xlim = c(-15, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T,
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of Gout PRS on Age at Gout Onset\nExcluding ABCG2\n", ""),
       rightlabs = c("Beta (years)", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F)

# Saving the tiff file.
dev.off()
```

Plotting the effect of the individual ABCG2 variants on age at onset. 

```{r OnsetvsABCG2figs, eval = F}
# Preparing age at onset vs ABCG2 variants model data for meta-analysis.
tmp <- OnsetModels %>% 
  filter(str_detect(Predictor, "rs")) %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         Beta = case_when(is.na(LCL) ~ NA_real_,
                          TRUE ~ Beta),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male                  N",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"),
         N = format(N, big.mark = ","))

# Making list of unique predictors (i.e. both SNPs)
snps <- unique(as.character(tmp$Predictor))

# For each SNP.
for(i in snps){
  
  # Filtering the tmp file to only include that SNP.
  tmp2 <- tmp %>% 
           filter(Predictor == i)
  
  # Running the meta-analysis for that SNP.
  assign(paste0(i), metagen(TE = Beta, 
                            seTE = SE, 
                            studlab = Cohort,
                            subgroup = Label,
                            data = tmp2))
}

# Preparing to produce tiff file for the first SNP.
tiff(file = here(paste0("Output/Plots/Onsetvs", snps[1], ".tiff")), units = "in", width = 7.0, height = 8.5, res = 300)

# Producing the forest plot for output in the tiff.
forest(get(snps[1]), 
       xlim = c(-15, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N"),
       rightcols = c("effect", "ci"),
       leftlabs = c(paste0("Change in Age at Gout Onset\nper ", snps[1], " allele"), ""),
       rightlabs = c("Beta (years)", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F)

# Saving the tiff file.
dev.off()

# Preparing to produce tiff file for the second SNP.
tiff(file = here(paste0("Output/Plots/Onsetvs", snps[2], ".tiff")), units = "in", width = 7.0, height = 8.5, res = 300)

# Producing the forest plot for output in the tiff.
forest(get(paste0(snps[2])), 
       xlim = c(-15, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N"),
       rightcols = c("effect", "ci"),
       leftlabs = c(paste0("Change in Age at Gout Onset\nper ", snps[2], " allele"), ""),
       rightlabs = c("Beta (years)", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F)

# Saving the tiff file.
dev.off()
```

Plotting the effect of the PRS, PRS less ABCG2 and the urate PRS on tophaceous gout.

```{r TophivsPRSfigs, eval = F}
# Preparing tophaceous gout vs gout PRS (unadjusted) model data for meta-analysis.
tmp <- TophiModels %>% 
  filter(Predictor == "PRS",
         !str_detect(Covariates, "DURATION"),
         `N case` > 20,
         `N control` > 20) %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         `log-odds` = case_when(LCL == 0 ~ NA_real_,
                        TRUE ~ `log-odds`),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"))

# Running meta-analysis.
tophi <- metagen(TE = `log-odds`, 
                 seTE = SE, 
                 studlab = Cohort,
                 subgroup = Label,
                 data = tmp, 
                 sm = "OR")

# Preparing to produce tiff file.
tiff(file = here("Output/Plots/TophivsPRS.tiff"), units = "in", width = 6.75, height = 6.25, res = 300)

# Producing the forest plot for output in the tiff.
forest(tophi, 
       xlim = c(0.2, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N case", "N control"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of Gout PRS on\nTophaceous Gout", "N\nwith", "N\nwithout"),
       rightlabs = c("OR", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F,
       smlab = "Unadjusted")

# Saving the tiff file.
dev.off()

# Preparing tophaceous gout vs gout PRS (adjusted) model data for meta-analysis.
tmp <- TophiModels %>% 
  filter(Predictor == "PRS",
         str_detect(Covariates, "DURATION"),
         `N case` > 20,
         `N control` > 20) %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         `log-odds` = case_when(LCL == 0 ~ NA_real_,
                        TRUE ~ `log-odds`),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"))

# Running meta-analysis.
tophi_adj <- metagen(TE = `log-odds`, 
                     seTE = SE, 
                     studlab = Cohort,
                     subgroup = Label,
                     data = tmp, 
                     sm = "OR")

# Preparing to produce tiff file.
tiff(file = here("Output/Plots/TophivsPRS_adj.tiff"), units = "in", width = 6.75, height = 6.25, res = 300)

# Producing the forest plot for output in the tiff.
forest(tophi_adj, 
       xlim = c(0.2, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N case", "N control"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of Gout PRS on\nTophaceous Gout", "N\nwith", "N\nwithout"),
       rightlabs = c("OR", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F,
       smlab = "Duration Adjusted")

# Saving the tiff file.
dev.off()

# Preparing tophaceous gout vs gout PRS less ABCG2 (unadjusted) model data for meta-analysis.
tmp <- TophiModels %>% 
  filter(Predictor == "PRS_noABCG2",
         !str_detect(Covariates, "DURATION"),
         `N case` > 20,
         `N control` > 20) %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         `log-odds` = case_when(LCL == 0 ~ NA_real_,
                        TRUE ~ `log-odds`),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"))

# Running meta-analysis.
tophi <- metagen(TE = `log-odds`, 
                 seTE = SE, 
                 studlab = Cohort,
                 subgroup = Label,
                 data = tmp, 
                 sm = "OR")

# Preparing to produce tiff file.
tiff(file = here("Output/Plots/TophivsPRS2.tiff"), units = "in", width = 6.75, height = 6.25, res = 300)

# Producing the forest plot for output in the tiff.
forest(tophi, 
       xlim = c(0.2, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N case", "N control"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of Gout PRS on\nTophaceous Gout", "N\nwith", "N\nwithout"),
       rightlabs = c("OR", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F,
       smlab = "Unadjusted\nExcluding ABCG2")

# Saving the tiff file.
dev.off()
```

Plotting the effect of the ABCG2 variants on tophaceous gout.

```{r TophivsABCG2figs, eval = F}
# Preparing tophaceous gout vs rs2231142 (unadjusted) model data for meta-analysis.
tmp <- TophiModels %>% 
  filter(Predictor == "rs2231142",
         !str_detect(Covariates, "DURATION"),
         `N case` > 20,
         `N control` > 20) %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         `log-odds` = case_when(LCL == 0 ~ NA_real_,
                        TRUE ~ `log-odds`),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"))

# Running meta-analysis.
tophi <- metagen(TE = `log-odds`, 
                 seTE = SE, 
                 studlab = Cohort,
                 subgroup = Label,
                 data = tmp, 
                 sm = "OR")

# Preparing to produce tiff file.
tiff(file = here("Output/Plots/Tophivsrs2231142.tiff"), units = "in", width = 6.75, height = 6.25, res = 300)

# Producing the forest plot for output in the tiff.
forest(tophi, 
       xlim = c(0.2, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N case", "N control"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of rs2231142 on\nTophaceous Gout", "N\nwith", "N\nwithout"),
       rightlabs = c("OR", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F,
       smlab = "")

# Saving the tiff file.
dev.off()

# Preparing tophaceous gout vs rs10011796 (unadjusted) model data for meta-analysis.
tmp <- TophiModels %>% 
  filter(Predictor == "rs10011796",
         !str_detect(Covariates, "DURATION"),
         `N case` > 20,
         `N control` > 20) %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         `log-odds` = case_when(LCL == 0 ~ NA_real_,
                        TRUE ~ `log-odds`),
         Label = case_when(!str_detect(Cohort, "Polynesian") & Sex == "Male" ~ "European Male",
                           !str_detect(Cohort, "Polynesian") & Sex == "Female" ~ "European Female",
                           str_detect(Cohort, "East Polynesian") & Sex == "Male" ~ "East Polynesian Male",
                           str_detect(Cohort, "East Polynesian") & Sex == "Female" ~ "East Polynesian Female",
                           str_detect(Cohort, "West Polynesian") & Sex == "Male" ~ "West Polynesian Male",
                           str_detect(Cohort, "West Polynesian") & Sex == "Female" ~ "West Polynesian Female"))

# Running meta-analysis.
tophi <- metagen(TE = `log-odds`, 
                 seTE = SE, 
                 studlab = Cohort,
                 subgroup = Label,
                 data = tmp, 
                 sm = "OR")

# Preparing to produce tiff file.
tiff(file = here("Output/Plots/Tophivsrs10011796.tiff"), units = "in", width = 6.75, height = 6.25, res = 300)

# Producing the forest plot for output in the tiff.
forest(tophi, 
       xlim = c(0.2, 5),
       random = F,
       print.tau2 = F,
       print.I2 = T, 
       col.square.lines = "darkgray", 
       col.diamond.lines = "darkgray",
       col.diamond = "#1e6b52",
       col.square = "#aa9767",
       col.study = "gray60",
       col.by = "black",
       leftcols = c("studlab", "N case", "N control"),
       rightcols = c("effect", "ci"),
       leftlabs = c("Effect of rs10011796 on\nTophaceous Gout", "N\nwith", "N\nwithout"),
       rightlabs = c("OR", "[95% CI]"),
       addrows.below.overall = 2,
       overall = F,
       print.subgroup.name = F,
       smlab = "")

# Saving the tiff file.
dev.off()
```

<br>

The following table describes the cohort statistics for every variable that I have deemed to have sufficient non-missing data. If it is missing at more than 50% in a single cohort then that cohort will be set to "too much missing". If there are no cohorts with fewer than 30% missing (excluding UKBB) then the variable is removed from the plot. 

```{r Table1a}
# Preparing data_list for table of demographics.
data_list2 <- data_list

# Removing UK Biobank and control cohorts.
for(i in length(data_list2):1){
  if(str_detect(names(data_list2)[[i]], "UK Biobank|Control")){
    data_list2[[i]] <- NULL
  }
}

# Removing cohorts with fewer than 20 individuals.
for(i in length(data_list2):1){
  if(nrow(data_list2[[i]]) < 20){
    data_list2[[i]] <- NULL
  }
}

# Producing table summarizing the cohorts of interest for the variables of interest.
table1 <- tibble("Cohort" = cohortstring[which(cohortstring %in% names(data_list2))], 
                 "N" = unlist(lapply(data_list2, function(x) format(nrow(x), big.mark = ","))),
                 "Age at\nrecruitment,\nyears" = unlist(lapply(data_list2, function(x) report(x$AGECOL))),
                 "Age at onset,\nyears" = unlist(lapply(data_list2, function(x) report(x$AGE1ATK))),
                 "Disease\nduration,\nyears" = unlist(lapply(data_list2, function(x) report(x$DURATION))),
                 "Flares in\nprevious\nyear" = unlist(lapply(data_list2, function(x) report_median(x$NUMATK))),
                 "Tophaceous\ndisease" = unlist(lapply(data_list2, function(x) sumreport(x$TOPHIGOUT))),
                 "Gout PRS" = unlist(lapply(data_list2, function(x) report(x$PRS))),
                 "Urate PRS" = unlist(lapply(data_list2, function(x) report(x$Urate_PRS))))

# Transposing the table and preparing it for printing.
table1 <- transpose_df(table1) %>% 
  column_to_rownames(var = "Cohort") %>% 
  mutate(across(.cols = 1:ncol(table1), ~ str_replace_all(string = .x, pattern = " ", replacement = "&nbsp;")))

# Further preparing the table for printing.
row.names(table1) <- str_replace_all(row.names(table1), " ", "&nbsp;")

# Printing the table.
table1 %>% 
  kable(col.names = clean_names[which(cohortstring %in% names(data_list2))],
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px") %>% 
  footnote("Flares reported as median (inter-quartile range). All other numeric variables reported as mean ± sd. All binary variables reported as N (%).")
```

<br>

### Supplementary figures and tables

Manhattan plot for the gout GWAS showing the locations of the SNPs that were used in the PRS.

```{r Manhattan, eval = F}
# Loading in summary statistics file.
load(path(scratch_path, "Output/sumstat_final.RData"))

# Filtering summary stats to keep only SNPs with P < 0.01.
GOUT_pValues <- sumstat_final %>% 
  filter(P < 0.01) %>% 
  arrange(CHR, BP)

# Extracting lead SNPs for labeling.
tmp <- GOUT_pValues %>% 
  filter(RSID %in% UKBB_Gene_OR$RSID) %>% 
  mutate("Gene" = UKBB_Gene_OR$Locus_Name)

# Extracting all non-lead SNPs for labeling.
tmp2 <- GOUT_pValues %>% 
  filter(!(SNP %in% tmp$SNP)) %>% 
  mutate("Gene" = NA)

# Joining back together.
GOUT_pValues <- full_join(tmp, tmp2) %>% 
  arrange(CHR, BP)

# Preparing data for manhattan plot.
GOUT_pValues2 <- GOUT_pValues %>% 
  
  # Grouping by chromosome.
  group_by(CHR) %>% 
  
  # Computing chromosome size as new column.
  summarise(chr_len = max(BP)) %>% 
  
  # Calculating cumulative position of each chromosome.
  mutate(tot = cumsum(chr_len) - chr_len) %>%
  
  # Removing chromosome length column.
  select(-chr_len) %>%
  
  # Adding this info to the initial dataset.
  left_join(GOUT_pValues, ., by = "CHR") %>%
  
  # Arranging by chr/bp.
  arrange(CHR, BP) %>%
  
  # Adding a cumulative position of each SNP.
  mutate(BPcum = BP + tot) %>%
  
  # Adding highlight and annotation information.
  mutate(is_highlight = ifelse(RSID %in% UKBB_Gene_OR$RSID, "yes", "no")) %>%
  mutate(is_annotate = ifelse(RSID %in% UKBB_Gene_OR$RSID, "yes", "no")) 

# Marking center of each chromosome in axisdf object.
axisdf <- GOUT_pValues2 %>% 
  group_by(CHR) %>% 
  summarize(center = (max(BPcum) + min(BPcum)) / 2)

# Plotting the manhattan plot, with the cumulative SNP location on the x-axis and -log10(P) on the y-axis.
ggplot(GOUT_pValues2, aes(x = BPcum, y = -log10(P))) +
  
  # Plotting each SNP, colored based on the CHR.
  geom_point(aes(color = as.factor(CHR)), alpha = 0.8, size = 1.3) +
  
  # Setting colors of chromosomes.
  scale_color_manual(values = rep(c("#1e6b52", "#aa9767"), 22)) +
  
  # Adding significance threshold.
  geom_hline(yintercept = -log10(5e-8), colour = "red") +
  
  # Customizing x-axis based on CHR center points.
  scale_x_continuous(label = axisdf$CHR, breaks = axisdf$center) +
  
  # Removing space between plot area and x-axis.
  scale_y_continuous(expand = c(0, 0), limits = c(0, 250)) +
  
  # Adding highlighted data points.
  geom_point(data = subset(GOUT_pValues2, is_highlight == "yes"), color = "orange", size = 2) +
  
  # Adding label using ggrepel to avoid overlapping.
  geom_label_repel(aes(label = Gene), size = 3, box.padding = 0.5, force_pull = 0.5, nudge_y = 3, max.overlaps = Inf) +
  
  # Adding x-axis label.
  xlab("Chromosome") +
  
  # Adding title to plot.
  ggtitle("UK Biobank Gout GWAS Results") +
  
  # Setting theme to bw setting.
  theme_bw() +
  
  # Further customizing the theme.
  theme( 
    legend.position = "none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )

# Saving the image (must be done on biocmerriserver2).
ggsave(filename = here("Output/Plots/Manhattan.tiff"), width = 10, height = 5, dpi = 600)
```

<br>

Preparing table of GWAS results for PRS.

```{r GWASTable, eval = F}
# Making pooled West Polynesian table.
WestPoly <- rbind(data_list[[31]], data_list[[32]], data_list[[33]], data_list[[34]])

# Making pooled East Polynesian table.
EastPoly <- rbind(data_list[[23]], data_list[[24]], data_list[[25]], data_list[[26]], data_list[[27]], data_list[[28]], data_list[[29]], data_list[[30]])

# Pulling out list of SNPs.
SNPlist <- UKBB_Gene_OR$RSID

# Making function for extracting the minor allele frequency for a set of SNPs.
get_maf <- function(cohort, snps){
  
  # Making test dataframe which is each SNP as a factorized column.
  test <- cohort %>% 
    select(all_of(snps)) %>% 
    mutate_all(factor, levels = 0:2) %>% 
    na.omit()
  
  # Making empty vector.
  tmp <- c()
  
  # For each column in test.
  for(i in 1:ncol(test)){
    
    # Calculating the allele frequency for the effect allele (1 encoded).
    tmp[i] <- ((sum(test[[i]] == 1) + (sum(test[[i]] == 2) * 2)) / (2 * nrow(test))) %>% sprintf(fmt = "%#.3f")
  }
  
  # Output the tmp vector.
  return(tmp)
}

# Calculating EAF for West Polynesian cohort.
tmp1 <- get_maf(cohort = WestPoly, snps = SNPlist)

# Calculating EAF for East Polynesian cohort.
tmp2 <- get_maf(cohort = EastPoly, snps = SNPlist)

# Preparing lead SNP table for supplementary table format.
tmp <- UKBB_Gene_OR %>% 
  mutate(`Locus (chr:lower-upper)` = paste0(CHR, ":", BP1, "-", BP2),
         `Base Pair Position` = str_remove_all(format(BP, big.mark = ","), " "),
         `Odds ratio` = sprintf(OR, fmt = "%#.2f"),
         `95%-CI` = paste0("[", sprintf(L95, fmt = "%#.2f"), ", ", sprintf(U95, fmt = "%#.2f"), "]"),
         `Odds ratio (Marginal)` = str_replace(sprintf(OR_old, fmt = "%#.2f"), "NA", NA_character_),
         `95%-CI (Marginal)` = case_when(!is.na(L95_old) ~ paste0("[", sprintf(L95_old, fmt = "%#.2f"), ", ", sprintf(U95_old, fmt = "%#.2f"), "]")),
         `EAF (UK Biobank)` = sprintf(EAF, fmt = "%#.3f"),
         `EAF (East Polynesian)` = tmp2,
         `EAF (West Polynesian)` = tmp1,
         `p-value` = formatC(P, format = "e", digits = 2),
         `p-value (Marginal)` = formatC(P_old, format = "e", digits = 2)) %>% 
  rename(`Lead SNP rsID` = RSID,
         `Effect allele` = Effect_Allele,
         `Alternate allele` = Alternate_Allele,
         `Locus name` = Locus_Name) %>% 
  select(`Locus (chr:lower-upper)`, `Lead SNP rsID`, `Base Pair Position`, `Effect allele`, `Alternate allele`, `Odds ratio`, `95%-CI`, `p-value`, `Locus name`, `EAF (UK Biobank)`, `EAF (East Polynesian)`, `EAF (West Polynesian)`, `Odds ratio (Marginal)`, `95%-CI (Marginal)`, `p-value (Marginal)`)

# Writing out supplementary table 1.
write_delim(tmp, file = here("Output/Tables/UKBB_Gene_OR.txt"), delim = "\t")

# Preparing lead SNP of Tin Urate PRS table for supplementary table format.
tmp <- Tin_Gene_OR %>% 
  mutate(`Locus (chr:lower-upper)` = paste0(CHR, ":", BP1, "-", BP2),
         `Base Pair Position` = str_remove_all(format(BP, big.mark = ","), " "),
         Beta = sprintf(Beta, fmt = "%#.3f"),
         `95%-CI` = paste0("[", sprintf(L95, fmt = "%#.3f"), ", ", sprintf(U95, fmt = "%#.3f"), "]"),
         `p-value` = formatC(P, format = "e", digits = 2)) %>% 
  rename(`Lead SNP rsID` = RSID,
         `Effect allele` = Effect_Allele,
         `Alternate allele` = Alternate_Allele,
         `Locus name` = Locus_Name) %>% 
  select(`Locus (chr:lower-upper)`, `Lead SNP rsID`, `Base Pair Position`, `Effect allele`, `Alternate allele`, Beta, `95%-CI`, `p-value`, `Locus name`)

# Writing out supplementary table 1.
write_delim(tmp, file = here("Output/Tables/Tin_Gene_OR.txt"), delim = "\t")
```

<br>

Producing missing data supplementary table.

```{r MissingData3}
# Producing table of missing data statistics.
table1 <- tibble("Cohort" = cohortstring,
                 "N" = unlist(lapply(data_list, function(x) format(nrow(x), big.mark = ","))),
                 "Age at Collection" = unlist(lapply(data_list, function(x) missing(x$AGECOL))),
                 "Serum Urate" = unlist(lapply(data_list, function(x) missing(x$URATE))),
                 "ULT" = unlist(lapply(data_list, function(x) missing(x$ULT))),
                 "Age at Onset" = unlist(lapply(data_list, function(x) missing(x$AGE1ATK))),
                 "Disease Duration" = unlist(lapply(data_list, function(x) missing(x$DURATION))),
                 "Flares" = unlist(lapply(data_list, function(x) missing(x$NUMATK))),
                 "Tophi" = unlist(lapply(data_list, function(x) missing(x$TOPHIGOUT))),
                 "PRS" = unlist(lapply(data_list, function(x) missing(x$PRS))),
                 "Prophylaxis" = unlist(lapply(data_list, function(x) missing(x$PROPHY))),
                 "BMI" = unlist(lapply(data_list, function(x) missing(x$BMI))),
                 "Hypertension" = unlist(lapply(data_list, function(x) missing(x$HYPERTENSION))),
                 "Type 2 Diabetes" = unlist(lapply(data_list, function(x) missing(x$DIABETES))),
                 "Heart Disease" = unlist(lapply(data_list, function(x) missing(x$HEART))),
                 "Kidney Disease" = unlist(lapply(data_list, function(x) missing(x$KIDNEY))),
                 "Dyslipidemia" = unlist(lapply(data_list, function(x) missing(x$LIPIDS))),
                 "Stroke" = unlist(lapply(data_list, function(x) missing(x$STROKE))),
                 "Alcoholic Drinks / Week" = unlist(lapply(data_list, function(x) missing(x$TOTALALC))),
                 "Sugar-Sweetened Drinks / Week" = unlist(lapply(data_list, function(x) missing(x$SUGDRINK))),
                 "Current Smoker" = unlist(lapply(data_list, function(x) missing(x$CURSMOKE))),
                 "Family History of Gout" = unlist(lapply(data_list, function(x) missing(x$FAMGOUT))),
                 "No. Relatives w/ Gout" = unlist(lapply(data_list, function(x) missing(x$FAMGOUTNUM))))

# Transposing dataframe.
table1 <- transpose_df(table1) %>% 
  column_to_rownames(var = "Cohort")

# Writing out supplementary table.
write_delim(table1, file = here("Output/Tables/missing.txt"), delim = "\t")

# Preparing for printing in Rmd.
table1 <- table1 %>% 
  mutate(across(.cols = 1:ncol(table1), ~ str_replace(string = .x, pattern = " ", replacement = "&nbsp;")))

# Preparing for printing in Rmd.
row.names(table1) <- str_replace(row.names(table1), " ", "&nbsp;")

# Printing table.
table1 %>% 
  kable(col.names = clean_names,
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px") %>% 
  footnote("'All' = all missing, 'None' = none missing")
```

<br>

Producing full cohort statistics supplementary table.

```{r CohortStats}
# Producing table of cohort statistics.
table1 <- tibble("Cohort" = cohortstring, 
                 "N" = unlist(lapply(data_list, function(x) format(nrow(x), big.mark = ","))),
                 "Age at Collection (years)" = unlist(lapply(data_list, function(x) report(x$AGECOL))),
                 "Serum Urate (mg/dL)" = unlist(lapply(data_list, function(x) report(x$URATE))),
                 "ULT" = unlist(lapply(data_list, function(x) sumreport(x$ULT))),
                 "Age at Onset (years)" = unlist(lapply(data_list, function(x) report(x$AGE1ATK))),
                 "Disease Duration (years)" = unlist(lapply(data_list, function(x) report(x$DURATION))),
                 "Number of Flares in Last Year" = unlist(lapply(data_list, function(x) report_median(x$NUMATK))),
                 "Presence of Tophi" = unlist(lapply(data_list, function(x) sumreport(x$TOPHIGOUT))),
                 "PRS" = unlist(lapply(data_list, function(x) report(x$PRS))),
                 "Prophylaxis" = unlist(lapply(data_list, function(x) sumreport(x$PROPHY))),
                 "BMI" = unlist(lapply(data_list, function(x) report(x$BMI))),
                 "Hypertension" = unlist(lapply(data_list, function(x) sumreport(x$HYPERTENSION))),
                 "Type 2 Diabetes" = unlist(lapply(data_list, function(x) sumreport(x$DIABETES))),
                 "Heart Disease" = unlist(lapply(data_list, function(x) sumreport(x$HEART))),
                 "Kidney Disease" = unlist(lapply(data_list, function(x) sumreport(x$KIDNEY))),
                 "Dyslipidemia" = unlist(lapply(data_list, function(x) sumreport(x$LIPIDS))),
                 "Stroke" = unlist(lapply(data_list, function(x) sumreport(x$STROKE))),
                 "Alcoholic Drinks / Week" = unlist(lapply(data_list, function(x) report(x$TOTALALC))),
                 "Sugar-Sweetened Drinks / Week" = unlist(lapply(data_list, function(x) report(x$SUGDRINK))),
                 "Current Smoker" = unlist(lapply(data_list, function(x) sumreport(x$CURSMOKE))),
                 "Family History of Gout" = unlist(lapply(data_list, function(x) sumreport(x$FAMGOUT))),
                 "No. Relatives w/ Gout" = unlist(lapply(data_list, function(x) report(x$FAMGOUTNUM))))

# Transposing dataframe.
table1 <- transpose_df(table1) %>% 
  column_to_rownames(var = "Cohort")

# Writing out table.
write_delim(table1, file = here("Output/Tables/demographics.txt"), delim = "\t")

# Preparing for printing.
table1 <- table1 %>% 
  mutate(across(.cols = 1:ncol(table1), ~ str_replace(string = .x, pattern = " ", replacement = "&nbsp;")))

# Preparing for printing.
row.names(table1) <- str_replace(row.names(table1), " ", "&nbsp;")

# Printing table.
table1 %>% 
  kable(col.names = clean_names,
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px") %>% 
  footnote("Flares reported as median (inter-quartile range). All other numeric variables reported as mean ± sd. All binary variables reported as N (%).")
```

<br>

Producing supplementary table of model results.

```{r goutmodeltable, eval = F}
# Preparing gout model results table (excluding East Polynesians for now).
NotEast <- GoutModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         LCL = sprintf(LCL, fmt = "%#.2f"),
         UCL = sprintf(UCL, fmt = "%#.2f"),
         OR = sprintf(OR, fmt = "%#.2f"),
         CI = paste0("[", LCL, ", ", UCL, "]"),
         Adjusted = case_when(str_detect(Covariates, "AGECOL") ~ TRUE,
                              TRUE ~ FALSE)) %>% 
  filter(!str_detect(Cohort, "East Polynesian")) %>% 
  select(Cohort, `N case`, `N control`, Sex, Adjusted, Predictor, OR, CI, Pval)

# Producing East Polynesian gout results table.
tmp <- GoutModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         Adjusted = case_when(str_detect(Covariates, "AGECOL") ~ TRUE,
                              TRUE ~ FALSE),
         `log-odds` = case_when(`N case` < 20 | `N control` < 20 ~ NA_real_,
                        TRUE ~ `log-odds`)) %>% 
  filter(str_detect(Cohort, "East Polynesian"),
         !is.na(`log-odds`)) %>% 
  select(Cohort, `N case`, `N control`, Sex, Predictor, Adjusted, `log-odds`, SE)

# Keeping unadjusted models.
tmp2 <- tmp %>% 
  filter(!Adjusted)

# Making list of SNPs.
snplist <- as.character(unique(GoutModels$Predictor))

# Preparing tibble with named columns of specified types (using placeholder row).
out <- tibble("Predictor" = "test",
              "Male OR" = 1,
              "Male LCL" = 1,
              "Male UCL" = 1,
              "Male Pval" = 1,
              "Female OR" = 1,
              "Female LCL" = 1,
              "Female UCL" = 1,
              "Female Pval" = 1)

# For each predictor in snplist.
for(i in seq_along(snplist)){
  
  # Keeping only models with that predictor.
  tmp3 <- tmp2 %>% 
    filter(Predictor == snplist[i])
  
  # Running meta-analysis.
  gout <- metagen(TE = `log-odds`, 
                  seTE = SE, 
                  studlab = Cohort,
                  subgroup = Sex,
                  data = tmp3,
                  sm = "OR")
  
  # Adding meta-analysis results to out table.
  out <- rbind(out,
               list(snplist[i],
                    exp(gout$TE.fixed.w[[1]]),
                    exp(gout$lower.fixed.w[[1]]),
                    exp(gout$upper.fixed.w[[1]]),
                    gout$pval.fixed.w[[1]],
                    exp(gout$TE.fixed.w[[2]]),
                    exp(gout$lower.fixed.w[[2]]),
                    exp(gout$upper.fixed.w[[2]]),
                    gout$pval.fixed.w[[2]]))
}

# Cleaning up results table.
out1 <- out %>% 
  slice(-1) %>% 
  mutate(Cohort = "East Polynesian",
         Adjusted = FALSE,
         `Male LCL` = sprintf(`Male LCL`, fmt = "%#.2f"),
         `Male UCL` = sprintf(`Male UCL`, fmt = "%#.2f"),
         `Male OR` = sprintf(`Male OR`, fmt = "%#.2f"),
         `Male CI` = paste0("[", `Male LCL`, ", ", `Male UCL`, "]"),
         `Female LCL` = sprintf(`Female LCL`, fmt = "%#.2f"),
         `Female UCL` = sprintf(`Female UCL`, fmt = "%#.2f"),
         `Female OR` = sprintf(`Female OR`, fmt = "%#.2f"),
         `Female CI` = paste0("[", `Female LCL`, ", ", `Female UCL`, "]")) %>% 
  select(Cohort, Adjusted, Predictor, `Male OR`, `Male CI`, `Male Pval`, `Female OR`, `Female CI`, `Female Pval`)

# Extracting adjusted models.
tmp2 <- tmp %>% 
  filter(Adjusted)

# Making table for outputting meta-analysis results.
out <- tibble("Predictor" = "test",
              "Male OR" = 1,
              "Male LCL" = 1,
              "Male UCL" = 1,
              "Male Pval" = 1,
              "Female OR" = 1,
              "Female LCL" = 1,
              "Female UCL" = 1,
              "Female Pval" = 1)

# For each predictor.
for(i in seq_along(snplist)){
  
  # Extracting models with that predictor.
  tmp3 <- tmp2 %>% 
    filter(Predictor == snplist[i])
  
  # Running meta-analysis.
  gout <- metagen(TE = `log-odds`, 
                   seTE = SE, 
                   studlab = Cohort,
                   subgroup = Sex,
                   data = tmp3,
                   sm = "OR")
  
  # Summarizing meta-analysis results.
  out <- rbind(out,
               list(snplist[i],
                    exp(gout$TE.fixed.w[[1]]),
                    exp(gout$lower.fixed.w[[1]]),
                    exp(gout$upper.fixed.w[[1]]),
                    gout$pval.fixed.w[[1]],
                    exp(gout$TE.fixed.w[[2]]),
                    exp(gout$lower.fixed.w[[2]]),
                    exp(gout$upper.fixed.w[[2]]),
                    gout$pval.fixed.w[[2]]))
}

# Cleaning up results table.
out2 <- out %>% 
  slice(-1) %>% 
  mutate(Cohort = "East Polynesian",
         Adjusted = TRUE,
         `Male LCL` = sprintf(`Male LCL`, fmt = "%#.2f"),
         `Male UCL` = sprintf(`Male UCL`, fmt = "%#.2f"),
         `Male OR` = sprintf(`Male OR`, fmt = "%#.2f"),
         `Male CI` = paste0("[", `Male LCL`, ", ", `Male UCL`, "]"),
         `Female LCL` = sprintf(`Female LCL`, fmt = "%#.2f"),
         `Female UCL` = sprintf(`Female UCL`, fmt = "%#.2f"),
         `Female OR` = sprintf(`Female OR`, fmt = "%#.2f"),
         `Female CI` = paste0("[", `Female LCL`, ", ", `Female UCL`, "]")) %>% 
  select(Cohort, Adjusted, Predictor, `Male OR`, `Male CI`, `Male Pval`, `Female OR`, `Female CI`, `Female Pval`)

# Recombining meta-analysis results.
EastPoly <- rbind(out1, out2)

# Combining all gout model results.
final <- rbind(NotEast, EastPoly)

# Writing out gout model results supplementary table.
write_delim(final, file = here("Output/Tables/GoutMod.txt"), delim = "\t")
```

<br>

Doing the same for age at onset models.

```{r onsetmodeltable, eval = F}
# Preparing West Polynesian onset models table.
tmp <- OnsetModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         LCL = sprintf(LCL, fmt = "%#.2f"),
         UCL = sprintf(UCL, fmt = "%#.2f"),
         Beta = sprintf(Beta, fmt = "%#.2f"),
         CI = paste0("[", LCL, ", ", UCL, "]")) %>% 
  filter(str_detect(Cohort, "West Polynesian")) %>% 
  select(Cohort, Sex, Predictor, Beta, CI, Pval)

# Extracting male model results.
tmp1 <- tmp %>% 
  filter(Sex == "Male") %>% 
  rename("Male Beta" = Beta,
         "Male CI" = CI,
         "Male Pval" = Pval) %>% 
  select(-Sex)

# Extracting female model results.
tmp2 <- tmp %>% 
  filter(Sex == "Female") %>% 
  rename("Female Beta" = Beta,
         "Female CI" = CI,
         "Female Pval" = Pval) %>% 
  select(-Sex)

# Joining results together.
WestPoly <- left_join(tmp1, tmp2)

# Preparing results table for meta-analyses.
tmp <- OnsetModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female")) %>% 
  select(Cohort, Sex, Predictor, Beta, SE)

# Extracting only European results.
tmp2 <- tmp %>% 
  filter(!str_detect(Cohort, "Polynesian"))

# Preparing output table.
out <- tibble("Predictor" = "test",
              "Male Beta" = 1,
              "Male LCL" = 1,
              "Male UCL" = 1,
              "Male Pval" = 1,
              "Female Beta" = 1,
              "Female LCL" = 1,
              "Female UCL" = 1,
              "Female Pval" = 1)

# For each predictor.
for(i in seq_along(snplist)){
  
  # Extracting models with that predictor.
  tmp3 <- tmp2 %>% 
    filter(Predictor == snplist[i])
  
  # Running meta-analysis.
  onset <- metagen(TE = Beta, 
                   seTE = SE, 
                   studlab = Cohort,
                   byvar = Sex,
                   data = tmp3)
  
  # Extracting summary of meta-analysis.
  out <- rbind(out,
               list(snplist[i],
                    onset$TE.fixed.w[[1]],
                    onset$lower.fixed.w[[1]],
                    onset$upper.fixed.w[[1]],
                    onset$pval.fixed.w[[1]],
                    onset$TE.fixed.w[[2]],
                    onset$lower.fixed.w[[2]],
                    onset$upper.fixed.w[[2]],
                    onset$pval.fixed.w[[2]]))
}

# Cleaning up European age at onset meta-analysis model results table.
Euro <- out %>% 
  slice(-1) %>% 
  mutate(Cohort = "European",
         `Male LCL` = sprintf(`Male LCL`, fmt = "%#.2f"),
         `Male UCL` = sprintf(`Male UCL`, fmt = "%#.2f"),
         `Male Beta` = sprintf(`Male Beta`, fmt = "%#.2f"),
         `Male CI` = paste0("[", `Male LCL`, ", ", `Male UCL`, "]"),
         `Female LCL` = sprintf(`Female LCL`, fmt = "%#.2f"),
         `Female UCL` = sprintf(`Female UCL`, fmt = "%#.2f"),
         `Female Beta` = sprintf(`Female Beta`, fmt = "%#.2f"),
         `Female CI` = paste0("[", `Female LCL`, ", ", `Female UCL`, "]")) %>% 
  select(Cohort, Predictor, `Male Beta`, `Male CI`, `Male Pval`, `Female Beta`, `Female CI`, `Female Pval`)

# Extracting East Polynesian model results.
tmp2 <- tmp %>% 
  filter(str_detect(Cohort, "East Polynesian"))

# Preparing output table.
out <- tibble("Predictor" = "test",
              "Male Beta" = 1,
              "Male LCL" = 1,
              "Male UCL" = 1,
              "Male Pval" = 1,
              "Female Beta" = 1,
              "Female LCL" = 1,
              "Female UCL" = 1,
              "Female Pval" = 1)

# For each predictor.
for(i in seq_along(snplist)){
  
  # Extracting models with that predictor.
  tmp3 <- tmp2 %>% 
    filter(Predictor == snplist[i])
  
  # Running meta-analysis.
  onset <- metagen(TE = Beta, 
                   seTE = SE, 
                   studlab = Cohort,
                   byvar = Sex,
                   data = tmp3)
  
  # Summarizing meta-analysis.
  out <- rbind(out,
               list(snplist[i],
                    onset$TE.fixed.w[[1]],
                    onset$lower.fixed.w[[1]],
                    onset$upper.fixed.w[[1]],
                    onset$pval.fixed.w[[1]],
                    onset$TE.fixed.w[[2]],
                    onset$lower.fixed.w[[2]],
                    onset$upper.fixed.w[[2]],
                    onset$pval.fixed.w[[2]]))
}

# Producing clean model output table for East Polynesian age at onset results.
EastPoly <- out %>% 
  slice(-1) %>% 
  mutate(Cohort = "East Polynesian",
         `Male LCL` = sprintf(`Male LCL`, fmt = "%#.2f"),
         `Male UCL` = sprintf(`Male UCL`, fmt = "%#.2f"),
         `Male Beta` = sprintf(`Male Beta`, fmt = "%#.2f"),
         `Male CI` = paste0("[", `Male LCL`, ", ", `Male UCL`, "]"),
         `Female LCL` = sprintf(`Female LCL`, fmt = "%#.2f"),
         `Female UCL` = sprintf(`Female UCL`, fmt = "%#.2f"),
         `Female Beta` = sprintf(`Female Beta`, fmt = "%#.2f"),
         `Female CI` = paste0("[", `Female LCL`, ", ", `Female UCL`, "]")) %>% 
  select(Cohort, Predictor, `Male Beta`, `Male CI`, `Male Pval`, `Female Beta`, `Female CI`, `Female Pval`)

# Combining results of all three ancestral groups.
final <- rbind(Euro, EastPoly, WestPoly)

# Writing out results.
write_delim(final, file = here("Output/Tables/OnsetMod.txt"), delim = "\t")
```

<br>

And finally for tophaceous gout models.

```{r tophimodeltable, eval = F}
# Preparing table of West Polynesian tophaceous disease models.
tmp <- TophiModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         LCL = sprintf(LCL, fmt = "%#.2f"),
         UCL = sprintf(UCL, fmt = "%#.2f"),
         OR = sprintf(OR, fmt = "%#.2f"),
         CI = paste0("[", LCL, ", ", UCL, "]"),
         Adjusted = case_when(str_detect(Covariates, "DURATION") ~ TRUE,
                              TRUE ~ FALSE),
         OR = case_when(`N case` < 20 | `N control` < 20 ~ NA_character_,
                        TRUE ~ OR),
         CI = case_when(`N case` < 20 | `N control` < 20 ~ NA_character_,
                        TRUE ~ CI),
         Pval = case_when(`N case` < 20 | `N control` < 20 ~ NA_real_,
                        TRUE ~ Pval)) %>% 
  select(Cohort, Sex, Adjusted, Predictor, OR, CI, Pval) %>% 
  filter(str_detect(Cohort, "West Polynesian"))

# Extracting male results.
tmp1 <- tmp %>% 
  filter(Sex == "Male") %>% 
  rename("Male OR" = OR,
         "Male CI" = CI,
         "Male Pval" = Pval) %>% 
  select(Cohort, Adjusted, Predictor, `Male OR`:`Male Pval`)

# Extracting female results.
tmp2 <- tmp %>% 
  filter(Sex == "Female") %>% 
  rename("Female OR" = OR,
         "Female CI" = CI,
         "Female Pval" = Pval) %>% 
  select(Cohort, Adjusted, Predictor, `Female OR`:`Female Pval`)

# Combining together.
WestPoly <- left_join(tmp1, tmp2)

# Preparing for European results summary.
tmp <- TophiModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         Adjusted = str_detect(Covariates, "DURATION"),
         `log-odds` = case_when(`N case` < 20 | `N control` < 20 ~ NA_real_,
                        TRUE ~ `log-odds`)) %>% 
  filter(!str_detect(Cohort, "Polynesian"),
         !is.na(`log-odds`)) %>% 
  select(Cohort, Sex, Predictor, Adjusted, `log-odds`, SE)

# Extracting unadjusted models only.
tmp2 <- tmp %>% 
  filter(!Adjusted)

# Preparing output table.
out <- tibble("Predictor" = "test",
              "Male OR" = 1,
              "Male LCL" = 1,
              "Male UCL" = 1,
              "Male Pval" = 1,
              "Female OR" = 1,
              "Female LCL" = 1,
              "Female UCL" = 1,
              "Female Pval" = 1)

# For each predictor.
for(i in seq_along(snplist)){
  
  # Extracting models with that predictor.
  tmp3 <- tmp2 %>% 
    filter(Predictor == snplist[i])
  
  # Running meta-analysis.
  tophi <- metagen(TE = `log-odds`, 
                   seTE = SE, 
                   studlab = Cohort,
                   byvar = Sex,
                   data = tmp3,
                   sm = "OR")
  
  # Summarizing meta-analysis.
  out <- rbind(out,
               list(snplist[i],
                    exp(tophi$TE.fixed.w[[1]]),
                    exp(tophi$lower.fixed.w[[1]]),
                    exp(tophi$upper.fixed.w[[1]]),
                    tophi$pval.fixed.w[[1]],
                    exp(tophi$TE.fixed.w[[2]]),
                    exp(tophi$lower.fixed.w[[2]]),
                    exp(tophi$upper.fixed.w[[2]]),
                    tophi$pval.fixed.w[[2]]))
}

# Cleaning up results.
out1 <- out %>% 
  slice(-1) %>% 
  mutate(Cohort = "European",
         Adjusted = FALSE,
         `Male LCL` = sprintf(`Male LCL`, fmt = "%#.2f"),
         `Male UCL` = sprintf(`Male UCL`, fmt = "%#.2f"),
         `Male OR` = sprintf(`Male OR`, fmt = "%#.2f"),
         `Male CI` = paste0("[", `Male LCL`, ", ", `Male UCL`, "]"),
         `Female LCL` = sprintf(`Female LCL`, fmt = "%#.2f"),
         `Female UCL` = sprintf(`Female UCL`, fmt = "%#.2f"),
         `Female OR` = sprintf(`Female OR`, fmt = "%#.2f"),
         `Female CI` = paste0("[", `Female LCL`, ", ", `Female UCL`, "]")) %>% 
  select(Cohort, Adjusted, Predictor, `Male OR`, `Male CI`, `Male Pval`, `Female OR`, `Female CI`, `Female Pval`)

# Extracting adjusted models.
tmp2 <- tmp %>% 
  filter(Adjusted)

# Preparing output table.
out <- tibble("Predictor" = "test",
              "Male OR" = 1,
              "Male LCL" = 1,
              "Male UCL" = 1,
              "Male Pval" = 1,
              "Female OR" = 1,
              "Female LCL" = 1,
              "Female UCL" = 1,
              "Female Pval" = 1)

# For each predictor.
for(i in seq_along(snplist)){
  
  # Extracting models with that predictor.
  tmp3 <- tmp2 %>% 
    filter(Predictor == snplist[i])
  
  # Running meta-analysis.
  tophi <- metagen(TE = `log-odds`, 
                   seTE = SE, 
                   studlab = Cohort,
                   byvar = Sex,
                   data = tmp3,
                   sm = "OR")
  
  # Summarizing results.
  out <- rbind(out,
               list(snplist[i],
                    exp(tophi$TE.fixed.w[[1]]),
                    exp(tophi$lower.fixed.w[[1]]),
                    exp(tophi$upper.fixed.w[[1]]),
                    tophi$pval.fixed.w[[1]],
                    exp(tophi$TE.fixed.w[[2]]),
                    exp(tophi$lower.fixed.w[[2]]),
                    exp(tophi$upper.fixed.w[[2]]),
                    tophi$pval.fixed.w[[2]]))
}

# Cleaning up results.
out2 <- out %>% 
  slice(-1) %>% 
  mutate(Cohort = "European",
         Adjusted = TRUE,
         `Male LCL` = sprintf(`Male LCL`, fmt = "%#.2f"),
         `Male UCL` = sprintf(`Male UCL`, fmt = "%#.2f"),
         `Male OR` = sprintf(`Male OR`, fmt = "%#.2f"),
         `Male CI` = paste0("[", `Male LCL`, ", ", `Male UCL`, "]"),
         `Female LCL` = sprintf(`Female LCL`, fmt = "%#.2f"),
         `Female UCL` = sprintf(`Female UCL`, fmt = "%#.2f"),
         `Female OR` = sprintf(`Female OR`, fmt = "%#.2f"),
         `Female CI` = paste0("[", `Female LCL`, ", ", `Female UCL`, "]")) %>% 
  select(Cohort, Adjusted, Predictor, `Male OR`, `Male CI`, `Male Pval`, `Female OR`, `Female CI`, `Female Pval`)

# Combining European results together.
Euro <- rbind(out1, out2)

# Preparing East Polynesian tophaceous disease model results.
tmp <- TophiModels %>% 
  mutate(Sex = case_when(str_detect(Cohort, "Male") ~ "Male",
                         str_detect(Cohort, "Female") ~ "Female"),
         Cohort = str_remove(Cohort, " - Male| - Female"),
         LCL = sprintf(LCL, fmt = "%#.2f"),
         UCL = sprintf(UCL, fmt = "%#.2f"),
         OR = sprintf(OR, fmt = "%#.2f"),
         CI = paste0("[", LCL, ", ", UCL, "]"),
         Adjusted = case_when(str_detect(Covariates, "DURATION") ~ TRUE,
                              TRUE ~ FALSE),
         OR = case_when(`N case` < 20 | `N control` < 20 ~ NA_character_,
                        TRUE ~ OR),
         CI = case_when(`N case` < 20 | `N control` < 20 ~ NA_character_,
                        TRUE ~ CI),
         Pval = case_when(`N case` < 20 | `N control` < 20 ~ NA_real_,
                        TRUE ~ Pval)) %>% 
  select(Cohort, Sex, Adjusted, Predictor, OR, CI, Pval) %>% 
  filter(str_detect(Cohort, "East Polynesian"))

# Extracting male results.
tmp1 <- tmp %>% 
  filter(Sex == "Male") %>% 
  rename("Male OR" = OR,
         "Male CI" = CI,
         "Male Pval" = Pval) %>% 
  select(Cohort, Adjusted, Predictor, `Male OR`:`Male Pval`)

# Extracting female results.
tmp2 <- tmp %>% 
  filter(Sex == "Female") %>% 
  rename("Female OR" = OR,
         "Female CI" = CI,
         "Female Pval" = Pval) %>% 
  select(Cohort, Adjusted, Predictor, `Female OR`:`Female Pval`)

# Combining East Polynesian model results.
EastPoly <- left_join(tmp1, tmp2)

# Combining all tophaceous gout model results.
final <- rbind(Euro, EastPoly, WestPoly)

# Writing out results.
write_delim(final, file = here("Output/Tables/TophiMod.txt"), delim = "\t")
```

<br>


### Urate vs Gout PRS

Comparing gout PRS to urate PRS for age at onset associations in men.

```{r UratevGoutPRS_Male, eval = F}
# Making list of PRS names.
prsnames <- c("PRS_std", "Urate_PRS_std")

# Preparing European dataset.
tmp <- all_pheno_prs %>%
  filter(Pheno.Study != "UK Biobank",
         !str_detect(COHORT_GOUT, "Polynesian"),
         GOUT,
         SEX == "Male",
         !is.na(Urate_PRS)) %>% 
  mutate(PRS_std = PRS / sd(PRS, na.rm = T),
         Urate_PRS_std = Urate_PRS / sd(Urate_PRS, na.rm = T))

# Making output vector.
out <- tibble("Outcome" = "test",
              "Predictor" = "test",
              "Beta_std" = 1,
              "LCL_std" = 1,
              "UCL_std" = 1,
              "P" = 1,
              "rsq" = 1,
              "Beta_std2" = 1,
              "Population" = "test",
              "N" = 1)

# Modeling each standardized PRS against age at onset.
for(i in seq_along(prsnames)){
  
  # Setting the full vector of variables to be the predictor and all covariates.
  variables <- c(prsnames[i], colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10)))
  variables2 <- c(str_remove(prsnames[i], "_std"), colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10)))
  
  # Making formula object.
  f <- formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
  f2 <- formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
  
  # Running model.
  assign(paste0("mod", i), lm(formula = f, data = tmp))
  assign(paste0("bmod", i), lm.beta(lm(formula = f2, data = tmp)))
  
  # Extracting model summaries.
  out1 <- list("Age at onset (years)",
               prsnames[i],
               round(summary(get(paste0("mod", i)))$coefficients[[2, 1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[2]], 3),
               formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2),
               round(rsq.partial(get(paste0("mod", i)), adj = T)$partial.rsq[1], 3), 
               round(get(paste0("bmod", i))$standardized.coefficients[[2]], 3),
               "European",
               nrow(tmp))
  
  # Adding results to output vector.
  out <- rbind(out, out1)
}
# formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2)
# Cleaning up output table.
out <- out %>% 
  slice(-1)

# Extracting East Polynesian individuals.
tmp <- all_pheno_prs %>%
  filter(str_detect(COHORT_GOUT, "East Polynesian"),
         GOUT,
         SEX == "Male",
         !is.na(Urate_PRS)) %>% 
  mutate(PRS_std = PRS / sd(PRS, na.rm = T),
         Urate_PRS_std = Urate_PRS / sd(Urate_PRS, na.rm = T))

# Modeling each standardized PRS against age at onset.
for(i in seq_along(prsnames)){
  
  # Setting the full vector of variables to be the predictor and all covariates.
  variables <- c(prsnames[i], colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  variables2 <- c(str_remove(prsnames[i], "_std"), colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  
  # Making formula object.
  f <- formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
  f2 <- formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
  
  # Running model.
  assign(paste0("mod", i), lm(f, data = tmp))
  assign(paste0("bmod", i), lm.beta(lm(f2, data = tmp)))
  
  # Extracting model summaries.
  out1 <- list("Age at onset (years)",
               prsnames[i],
               round(summary(get(paste0("mod", i)))$coefficients[[2, 1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[2]], 3),
               formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2),
               round(rsq.partial(get(paste0("mod", i)), adj = T)$partial.rsq[1], 3), 
               round(get(paste0("bmod", i))$standardized.coefficients[[2]], 3),
               "East Polynesian",
               nrow(tmp))
  
  # Adding results to output vector.
  out <- rbind(out, out1)
}

# Extracting West Polynesian individuals.
tmp <- all_pheno_prs %>%
  filter(str_detect(COHORT_GOUT, "West Polynesian"),
         GOUT,
         SEX == "Male",
         !is.na(Urate_PRS)) %>% 
  mutate(PRS_std = PRS / sd(PRS, na.rm = T),
         Urate_PRS_std = Urate_PRS / sd(Urate_PRS, na.rm = T))

# Modeling each standardized PRS against age at onset.
for(i in seq_along(prsnames)){
  
  # Setting the full vector of variables to be the predictor and all covariates.
  variables <- c(prsnames[i], colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  variables2 <- c(str_remove(prsnames[i], "_std"), colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  
  # Making formula object.
  f <- formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
  f2 <- formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
  
  # Running model.
  assign(paste0("mod", i), lm(f, data = tmp))
  assign(paste0("bmod", i), lm.beta(lm(f2, data = tmp)))
  
  # Extracting model summaries.
  out1 <- list("Age at onset (years)",
               prsnames[i],
               round(summary(get(paste0("mod", i)))$coefficients[[2, 1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[2]], 3),
               formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2),
               round(rsq.partial(get(paste0("mod", i)), adj = T)$partial.rsq[1], 3), 
               round(get(paste0("bmod", i))$standardized.coefficients[[2]], 3),
               "West Polynesian",
               nrow(tmp))
  
  # Adding results to output vector.
  out <- rbind(out, out1)
}

final <- out %>% 
  mutate(`95%-CI` = paste0("[", sprintf(LCL_std, fmt = "%#.2f"), ", ", sprintf(UCL_std, fmt = "%#.2f"), "]")) %>% 
  select(Outcome:Beta_std, `95%-CI`, P:N)

# Writing out results.
write_delim(final, file = here("Output/Tables/uratevsgoutPRS.txt"), delim = "\t")
```

<br>

### PRS Weighting Analysis

Comparing Gout PRS, Urate PRS, and unweighted PRS results for age at onset association in men only. Using each PRS, standardized in pooled cohorts of each ancestry, assessing their relative effects. Also producing standardized effect sizes using the lm.beta package and partial R-squared values.

```{r Weighting, eval = F}
# Making list of PRS names.
prsnames <- c("PRS2_std", "Urate_PRS2_std", "Unweighted_PRS_std")

# Preparing European dataset.
tmp <- all_pheno_prs %>%
  filter(Pheno.Study != "UK Biobank",
         !str_detect(COHORT_GOUT, "Polynesian"),
         GOUT,
         SEX == "Male") %>% 
  mutate(PRS2_std = PRS2 / sd(PRS2, na.rm = T),
         Urate_PRS2_std = Urate_PRS2 / sd(Urate_PRS2, na.rm = T),
         Unweighted_PRS_std = Unweighted_PRS / sd(Unweighted_PRS, na.rm = T))

# Making output vector.
out <- tibble("Outcome" = "test",
              "Predictor" = "test",
              "Beta_std" = 1,
              "LCL_std" = 1,
              "UCL_std" = 1,
              "P" = 1,
              "rsq" = 1,
              "Beta_std2" = 1,
              "Population" = "test",
              "N" = 1)

# Modeling each standardized PRS against age at onset.
for(i in seq_along(prsnames)){
  
  # Setting the full vector of variables to be the predictor and all covariates.
  variables <- c(prsnames[i], colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10)))
  variables2 <- c(str_remove(prsnames[i], "_std"), colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10)))
  
  # Making formula object.
  f <- formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
  f2 <- formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
  
  # Running model.
  assign(paste0("mod", i), lm(f, data = tmp))
  assign(paste0("bmod", i), lm.beta(lm(f2, data = tmp)))
  
  # Extracting model summaries.
  out1 <- list("Age at onset (years)",
               prsnames[i],
               round(summary(get(paste0("mod", i)))$coefficients[[2, 1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[2]], 3),
               formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2),
               round(rsq.partial(get(paste0("mod", i)), adj = T)$partial.rsq[1], 3), 
               round(get(paste0("bmod", i))$standardized.coefficients[[2]], 3),
               "European",
               nrow(tmp))
  
  # Adding results to output vector.
  out <- rbind(out, out1)
}
# Cleaning up output table.
out <- out %>% 
  slice(-1)

# Extracting East Polynesian individuals.
tmp <- all_pheno_prs %>%
  filter(str_detect(COHORT_GOUT, "East Polynesian"),
         GOUT,
         SEX == "Male") %>% 
  mutate(PRS2_std = PRS2 / sd(PRS2, na.rm = T),
         Urate_PRS2_std = Urate_PRS2 / sd(Urate_PRS2, na.rm = T),
         Unweighted_PRS_std = Unweighted_PRS / sd(Unweighted_PRS, na.rm = T))

# Modeling each standardized PRS against age at onset.
for(i in seq_along(prsnames)){
  
  # Setting the full vector of variables to be the predictor and all covariates.
  variables <- c(prsnames[i], colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  variables2 <- c(str_remove(prsnames[i], "_std"), colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  
  # Making formula object.
  f <- formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
  f2 <- formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
  
  # Running model.
  assign(paste0("mod", i), lm(f, data = tmp))
  assign(paste0("bmod", i), lm.beta(lm(f2, data = tmp)))
  
  # Extracting model summaries.
  out1 <- list("Age at onset (years)",
               prsnames[i],
               round(summary(get(paste0("mod", i)))$coefficients[[2, 1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[2]], 3),
               formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2),
               round(rsq.partial(get(paste0("mod", i)), adj = T)$partial.rsq[1], 3), 
               round(get(paste0("bmod", i))$standardized.coefficients[[2]], 3),
               "East Polynesian",
               nrow(tmp))
  
  # Adding results to output vector.
  out <- rbind(out, out1)
}

# Extracting West Polynesian individuals.
tmp <- all_pheno_prs %>%
  filter(str_detect(COHORT_GOUT, "West Polynesian"),
         GOUT,
         SEX == "Male") %>% 
  mutate(PRS2_std = PRS2 / sd(PRS2, na.rm = T),
         Urate_PRS2_std = Urate_PRS2 / sd(Urate_PRS2, na.rm = T),
         Unweighted_PRS_std = Unweighted_PRS / sd(Unweighted_PRS, na.rm = T))

# Modeling each standardized PRS against age at onset.
for(i in seq_along(prsnames)){
  
  # Setting the full vector of variables to be the predictor and all covariates.
  variables <- c(prsnames[i], colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  variables2 <- c(str_remove(prsnames[i], "_std"), colnames(tmp %>% select(Geno.PCVector1:Geno.PCVector10_Oc)))
  
  # Making formula object.
  f <- formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
  f2 <- formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
  
  # Running model.
  assign(paste0("mod", i), lm(f, data = tmp))
  assign(paste0("bmod", i), lm.beta(lm(f2, data = tmp)))
  
  # Extracting model summaries.
  out1 <- list("Age at onset (years)",
               prsnames[i],
               round(summary(get(paste0("mod", i)))$coefficients[[2, 1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[1]], 3), 
               round(confint.default(get(paste0("mod", i)))[2,][[2]], 3),
               formatC(summary(get(paste0("mod", i)))$coefficients[[2, 4]], format = "e", digits = 2),
               round(rsq.partial(get(paste0("mod", i)), adj = T)$partial.rsq[1], 3), 
               round(get(paste0("bmod", i))$standardized.coefficients[[2]], 3),
               "West Polynesian",
               nrow(tmp))
  
  # Adding results to output vector.
  out <- rbind(out, out1)
}

final <- out %>% 
  mutate(`95%-CI` = paste0("[", sprintf(LCL_std, fmt = "%#.2f"), ", ", sprintf(UCL_std, fmt = "%#.2f"), "]")) %>% 
  select(Outcome:Beta_std, `95%-CI`, P:N)

# Writing out results.
write_delim(final, file = here("Output/Tables/weightingscheme.txt"), delim = "\t")
```

For age at onset, this suggests that the gout PRS is consistently the best for associations, though it is only marginally better than the urate PRS. The unweighted PRS is consistently the worst.

<br>

### Standardized Sex Differences + Interaction Term

```{r menvwomen}
# Preparing pooled European cohort for analysis.
tmp <- all_pheno_prs %>%
  filter(Pheno.Study != "UK Biobank",
         !str_detect(COHORT_GOUT, "Polynesian"),
         GOUT) %>% 
  mutate(PRS_std = PRS / sd(PRS, na.rm = T))

# Running onset models in men and women separately.
mod1 <- lm(AGE1ATK ~ PRS_std, data = tmp %>% filter(SEX == "Male"))
mod2 <- lm(AGE1ATK ~ PRS_std, data = tmp %>% filter(SEX == "Female"))

# Analyzing results of models in men and women separately.
round(c(summary(mod1)$coefficients[[2, 1]], confint.default(mod1)[2,]), 3)
round(c(summary(mod2)$coefficients[[2, 1]], confint.default(mod2)[2,]), 3)

# Running interaction model.
mod3 <- lm(AGE1ATK ~ PRS_std*SEX, data = tmp)

# Analyzing interaction model shows significant interaction effect (p = 0.02).
summary(mod3)

# Plotting interaction.
ggplot(tmp, aes(x = PRS, y = AGE1ATK, color = SEX)) +
  geom_point(shape = 1, show.legend = FALSE) +
  geom_smooth(method = "lm", se = FALSE, color = "black") +
  facet_wrap(~ SEX, nrow = 2) +
  labs(x = "Gout Polygenic Risk Score",
       y = "Age at Onset (years)")

# Preparing pooled East Polynesian cohort for analysis.
tmp <- all_pheno_prs %>%
  filter(str_detect(COHORT_GOUT, "East Polynesian"),
         GOUT) %>% 
  mutate(PRS_std = PRS / sd(PRS, na.rm = T))

# Running onset models in men and women separately.
mod1 <- lm(AGE1ATK ~ PRS_std, data = tmp %>% filter(SEX == "Male"))
mod2 <- lm(AGE1ATK ~ PRS_std, data = tmp %>% filter(SEX == "Female"))

# Analyzing results of models in men and women separately.
round(c(summary(mod1)$coefficients[[2, 1]], confint.default(mod1)[2,]), 3)
round(c(summary(mod2)$coefficients[[2, 1]], confint.default(mod2)[2,]), 3)

# Running interaction model.
mod3 <- lm(AGE1ATK ~ PRS_std*SEX, data = tmp)

# Analyzing interaction model shows no significant interaction effect (p = 0.32).
summary(mod3)

# Preparing West Polynesian cohort for analysis.
tmp <- all_pheno_prs %>%
  filter(str_detect(COHORT_GOUT, "West Polynesian"),
         GOUT) %>% 
  mutate(PRS_std = PRS / sd(PRS, na.rm = T))

# Running onset models in men and women separately.
mod1 <- lm(AGE1ATK ~ PRS_std, data = tmp %>% filter(SEX == "Male"))
mod2 <- lm(AGE1ATK ~ PRS_std, data = tmp %>% filter(SEX == "Female"))

# Analyzing results of models in men and women separately.
round(c(summary(mod1)$coefficients[[2, 1]], confint.default(mod1)[2,]), 3)
round(c(summary(mod2)$coefficients[[2, 1]], confint.default(mod2)[2,]), 3)

# Running interaction model.
mod3 <- lm(AGE1ATK ~ PRS_std*SEX, data = tmp)

# Analyzing interaction model shows no significant interaction effect (p = 0.63).
summary(mod3)
```

<br>

### Relative Effect Sizes

Estimating the relative contribution of sex, age, gout genetic risk, BMI, and ancestry on age at onset. 

```{r PartialR2, eval = F}
# Preparing cohort.
tmp <- all_pheno_prs %>% 
  filter(!is.na(AGE1ATK),
         !is.na(PRS),
         GOUT) %>% 
  mutate(ANCESTRY = factor(case_when(str_detect(ANCESTRY_GOUT, "European") ~ "European",
                              str_detect(ANCESTRY_GOUT, "East Polynesian") ~ "East Polynesian",
                              str_detect(ANCESTRY_GOUT, "West Polynesian") ~ "West Polynesian"),
                           levels = c("European", "East Polynesian", "West Polynesian")))

# Running model.
mod <- lm(AGE1ATK ~ PRS + SEX + ANCESTRY + BMI, data = tmp)

# Extracting partial R-squared values.
partial_table <- tibble("Predictor" = rsq.partial(mod, adj = T)$variable,
                        "Partial R-squared" = rsq.partial(mod, adj = T)$partial.rsq,
                        "Percent variance explained" = round(rsq.partial(mod, adj = T)$partial.rsq * 100, 1))

# Printing table.
kable(partial_table)
```

<br>

### Flares plot.

Flares vs PRS

```{r FlarePlot, eval = F}
all_pheno_prs %>% 
  filter(NUMATK > 1,
         NUMATK < 53) %>% 
  mutate(Label = factor(case_when(COHORT_GOUT == "Aus/NZ - Gout" ~ "Aus/NZ",
                                  COHORT_GOUT == "GlobalGout - Gout" ~ "GlobalGout",
                                  str_detect(COHORT_GOUT, "Ardea") ~ as.character(COHORT_GOUT),
                                  COHORT_GOUT == "East Polynesian - Gout" ~ "East Polynesian",
                                  COHORT_GOUT == "East Polynesian - Gout - NP" ~ "East Polynesian - NP",
                                  COHORT_GOUT == "West Polynesian - Gout" ~ "West Polynesian"),
                        levels = c("Aus/NZ", "GlobalGout", "Ardea - LASSO", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - CRYSTAL", "Ardea - LIGHT", "East Polynesian", "East Polynesian - NP", "West Polynesian"))) %>% 
  ggplot(aes(x = PRS, y = NUMATK, color = SEX)) +
    geom_point(size = 0.5) +
    facet_wrap(~ Label, nrow = 2, ncol = 5) +
    labs(x = "Gout Polygenic Risk Score", y = "Number of Flares in Last Year (self-report)", color = "Sex")

ggsave(filename = here("Output/Plots/Flares.tiff"), width = 10, height = 5, dpi = 600)
```

