## GWAS Code

Murray first ran a genome-wide association study in the UK Biobank European cohort with gout as the outcome. This was done with a total of 27,287,012 variants (after imputation), and adjusted for age, sex, and the first 40 principal components. The code for this is seen below:

```{bash, engine.opts = '-l', eval = F}
# Save the following as do_gout_gwas.sh

file=$1
mem=36000

plink1.9b6.10 --vcf ${file}.vcf.gz --make-bed --out ${file} --memory $mem

plink1.9b6.10 --bfile ${file} --out ${file}_tmp --keep gout_gwas_keep_ids_w_sex.txt --pheno gout_gwas_covar.covar  --pheno-name plink_goutaff --update-sex gout_gwas_keep_ids_w_sex.txt --make-bed --memory $mem
# --geno 0.1 is default
plink1.9b6.10 --bfile ${file}_tmp --logistic sex --freq case-control --geno 0.1 --missing --ci 0.95 --maf 0.0001 --hwe 0.000001 --hardy --out gout_gwas/${file} --covar gout_gwas_covar.covar --covar-name Age,pc1-pc40 --memory $mem

rm ${file}.{bed,bim,fam} ${file}_tmp.{bed,bim,fam}

tr -s ' ' < gout_gwas/${file}.assoc.logistic | tr ' ' '\t' | sed 's/^\t//g' | sed 's/\t$//g' | gzip -c > gout_gwas/${file}.assoc.logistic.tsv.gz && rm gout_gwas/${file}.assoc.logistic
tr -s ' ' < gout_gwas/${file}.frq.cc  | tr ' ' '\t' | sed 's/^\t//g' | sed 's/\t$//g' | gzip -c > gout_gwas/${file}.frq.cc.tsv.gz && rm gout_gwas/${file}.frq.cc
tr -s ' ' < gout_gwas/${file}.hwe | tr ' ' '\t' | sed 's/^\t//g' |sed 's/\t$//g' | gzip -c > gout_gwas/${file}.hwe.tsv.gz && rm gout_gwas/${file}.hwe
tr -s ' ' < gout_gwas/${file}.imiss | tr ' ' '\t' | sed 's/^\t//g' | sed 's/\t$//g' | gzip -c > gout_gwas/${file}.imiss.tsv.gz && rm gout_gwas/${file}.imiss
tr -s ' ' < gout_gwas/${file}.lmiss | tr ' ' '\t' | sed 's/^\t//g' | sed 's/\t$//g' | gzip -c > gout_gwas/${file}.lmiss.tsv.gz && rm gout_gwas/${file}.lmiss
```

```{bash, engine.opts = '-l', eval = F}
# Then run this code

cd /Volumes/archive/merrimanlab_nobackup/ukbio/EGAD00010001474/splits

parallel --progress -P parallel_process.txt --tmp ./tmp/ 'bash do_gout_gwas.sh {}' ::: $(basename -s .vcf.gz -a $(ls bgenix_convert_chr*.gz ))

# These were then concatenated together into ukbb_gout-allcontrol_chr1-22.X.XY.add_unfiltered_p.tsv which was copied to the Data directory
```

Next, I filtered the summary stats to only include SNPs that were genotyped in the CoreExome chip. Additionally, I removed any indels, variants with MAF < 0.01, and X/Y chromosome SNPs, and ensured that all variants were biallelic.

```{r sumstat_filtering, eval = F}
CoreExPheno <- read_delim(here("Data/Phenotypes/CZ-MB1.2-QC1.10_MergedPhenotypes_20082020.txt"), delim = "\t")
  
All_CoreEx_ID <- read_delim("/Volumes/archive/merrimanlab/raid_backup/New_Zealand_Chip_data/CoreExome/QC_MergedBatches/Final_Data/CZ-MB1.2-QC1.10_CoreExome24-1.0-3_genotyped-QCd_rsIDconverted.fam", delim = " ", col_names = F)

CoreExPheno_Euro <- CoreExPheno %>% 
  filter(Geno.BroadAncestry == "European",
         Geno.SampleID %in% All_CoreEx_ID$X2,
         General.Use != "No",
         !(Pheno.Study %in% c("Auckland Controls", "Australian Controls", "ESR", "Rheumatoid Arthritis")))

CoreExPheno_Poly <- CoreExPheno %>% 
  filter(Geno.BroadAncestry == "Oceanian",
         Geno.SampleID %in% All_CoreEx_ID$X2,
         General.Use != "No",
         !(Pheno.Study %in% c("ESR", "Pacific Trust")),
         !is.na(Pheno.GoutSummary))

all_coreex_ids <- rbind(CoreExPheno_Euro, CoreExPheno_Poly) %>% 
  select(Geno.FamilyID, Geno.SampleID)

# write_delim(all_coreex_ids, delim = "\t", file = here("Output/Temp/all_coreex_ids.txt"), col_names = F)

rm(CoreExPheno, CoreExPheno_Euro, CoreExPheno_Poly, all_coreex_ids, All_CoreEx_ID)

# system(paste0("source ~/.bashrc; plink1.9b4.9 --bfile /Volumes/archive/merrimanlab/raid_backup/New_Zealand_Chip_data/CoreExome/QC_MergedBatches/Final_Data/CZ-MB1.2-QC1.10_CoreExome24-1.0-3_genotyped-QCd_rsIDconverted --keep ", here("Output/Temp/all_coreex_ids.txt"), " --geno 0.05 --make-bed --out ", here("Output/Temp/inCoreExGeno")))

geno <- read_delim(here("Output/Temp", "inCoreExGeno.bim"), delim = "\t", col_names = FALSE) %>% 
  mutate(CHR_BP = paste0(X1, "_", X4))

sumstat <- vroom(here("Data/GWAS/ukbb_gout-allcontrol_chr1-22.X.XY.add_unfiltered_p.tsv"), 
                 delim = "\t", 
                 col_names = TRUE)

sumstat2 <- sumstat %>% 
  mutate(CHR_BP = paste0(CHR, "_", BP)) %>% 
  filter(CHR %in% 1:22,
         CHR_BP %in% geno$CHR_BP,
         str_length(A1) == 1) %>% 
  separate(SNP, into = c("SNP1", "SNP2", "SNP3"), sep = ",", remove = FALSE) # total of 307,368 variants

# table(sumstat2$SNP3) # just chromosomes, can remove

sumstat2_2 <- sumstat2 %>% 
  select(-SNP3)

test1 <- sumstat2_2 %>% 
  filter(str_detect(SNP1, regex("^rs[0-9]+"))) # 307,066 variants have an rsID in SNP1 column
  
test1_1 <- test1 %>%
  filter(str_detect(SNP2, regex("^rs[0-9]+"))) # 74,955 of these have an rsID in SNP2 column

test1_1_1 <- test1_1 %>%
  filter(SNP1 != SNP2) # 63 variants have two rsIDs, for the most part SNP1 appears to be the newest rsID, but I will keep the extra rsID in a separate column

test1_2 <- test1 %>% 
  filter(str_detect(SNP2, regex("^[0-9]+:[0-9]+_[ACGT]+_[ACGT]+$"))) # 231,166 variants have rsID in SNP1 column and chr:bp_a1_a2 in SNP2 column

test1_2_1 <- test1_2 %>% 
  mutate(CHR2 = SNP2 %>% str_split(":", simplify = TRUE) %>% .[,1] %>% as.numeric(),
         BP2 = SNP2 %>% str_split(":", simplify = TRUE) %>% .[,2] %>% str_split("_", simplify = TRUE) %>% .[,1] %>% as.numeric(),
         Allele1 = SNP2 %>% str_split("_", simplify = TRUE) %>% .[,2],
         Allele2 = SNP2 %>% str_split("_", simplify = TRUE) %>% .[,3])

# sum(test1_2_1$BP != test1_2_1$BP2) # all BPs are equal

# sum(test1_2_1$CHR != test1_2_1$CHR2) # all CHRs are equal

# sum(test1_2_1$A1 != test1_2_1$Allele2) # Allele2 is not always A1

test1_2_2 <- test1_2_1 %>% 
  select(-CHR2, -BP2) %>% 
  filter(str_length(Allele1) == 1,
         str_length(Allele2) == 1) # removes a further 13 indels = 231,153 variants

test1_2_final <- test1_2_2 %>% 
  select(SNP, Allele1, Allele2)

test1_3 <- test1 %>%
  filter(!str_detect(SNP2, regex("^[0-9]+:[0-9]+_[ACGT]+_[ACGT]+$|^rs[0-9]+$"))) # 945 variants have neither rsID nor chr:bp_a1_a2 in SNP2 column

sum(!str_detect(test1_3$SNP2, regex("^Affx-[0-9]+$"))) # All of these are in the format Affx-<number>

test1_final <- test1 %>% 
  mutate(RSID = SNP1,
         ALT_RSID = case_when(str_detect(SNP2, regex("^rs[0-9]+$")) & SNP1 != SNP2 ~ SNP2, TRUE ~ NA_character_),
         AFFYID = case_when(str_detect(SNP2, regex("^Affx-[0-9]+$")) ~ SNP2, TRUE ~ NA_character_),
         SNP_ID = case_when(str_detect(SNP2, regex("^[0-9]+:[0-9]+_[ACGT]+_[ACGT]+$")) ~ SNP2, TRUE ~ NA_character_)) %>% 
  left_join(test1_2_final, by = "SNP")

test1_final2 <- test1_final %>% 
  filter(is.na(SNP_ID) | (!is.na(SNP_ID) & !is.na(Allele1))) # removing the indels

test2 <- sumstat2_2 %>% 
  filter(!str_detect(SNP1, regex("^rs[0-9]+"))) # 302 variants don't have an rsID in SNP1 column

test2_1 <- test2 %>% 
  filter(str_detect(SNP1, regex("^[0-9]+:[0-9]+_[ACGT]+_[ACGT]+$"))) # 300 have the SNP_ID format in SNP1

test2_1_1 <- test2_1 %>% 
  mutate(CHR2 = SNP1 %>% str_split(":", simplify = TRUE) %>% .[,1] %>% as.numeric(),
         BP2 = SNP1 %>% str_split(":", simplify = TRUE) %>% .[,2] %>% str_split("_", simplify = TRUE) %>% .[,1] %>% as.numeric(),
         Allele1 = SNP1 %>% str_split("_", simplify = TRUE) %>% .[,2],
         Allele2 = SNP1 %>% str_split("_", simplify = TRUE) %>% .[,3])

# sum(test2_1_1$BP != test2_1_1$BP2) # all BPs are equal

# sum(test2_1_1$CHR != test2_1_1$CHR2) # all CHRs are equal

# sum(test2_1_1$A1 != test2_1_1$Allele2) # Allele2 is not always A1

test2_1_2 <- test2_1_1 %>% 
  select(-CHR2, -BP2) %>% 
  filter(str_length(Allele1) == 1,
         str_length(Allele2) == 1)

# nrow(test2_1_2) - nrow(test2_1_1) # removed 12 indels

test2_1_final <- test2_1_2 %>% 
  select(SNP, Allele1, Allele2) # 288 total

test2_1_1 <- test2_1 %>%
  filter(str_detect(SNP2, regex("^rs[0-9]+"))) # 7 of the 300 have RSID in SNP2

test2_1_2 <- test2_1 %>%
  filter(str_detect(SNP2, regex("^[0-9]+:[0-9]+_[ACGT]+_[ACGT]+$"))) # 292 have SNP_ID in SNP2

# sum(test2_1_2$SNP1 != test2_1_2$SNP2) # All SNP_ID columns identical in these individuals

test2_1_3 <- test2_1 %>%
  filter(str_detect(SNP2, regex("^Affx-[0-9]+$"))) # 1 has AffyID in SNP2
  
test2_2 <- test2 %>%
  filter(str_detect(SNP1, regex("^Affx-[0-9]+$"))) # 2 have an AffyID in SNP1

test2_2_1 <- test2_2 %>%
  filter(str_detect(SNP2, regex("^Affx-[0-9]+$"))) # Both have AffyID in SNP2

# sum(test2_2_1$SNP1 != test2_2_1$SNP2) # All AffyID columns identical in these individuals

test2_final <- test2 %>% 
  mutate(RSID = case_when(str_detect(SNP2, regex("^rs[0-9]+$")) ~ SNP2, TRUE ~ NA_character_),
         ALT_RSID = NA_character_,
         AFFYID = case_when(str_detect(SNP2, regex("^Affx-[0-9]+$")) ~ SNP2, TRUE ~ NA_character_),
         SNP_ID = case_when(str_detect(SNP1, regex("^[0-9]+:[0-9]+_[ACGT]+_[ACGT]+$")) ~ SNP1, TRUE ~ NA_character_)) %>% 
  left_join(test2_1_final, by = "SNP")

test2_final2 <- test2_final %>% 
  filter(is.na(SNP_ID) | (!is.na(SNP_ID) & !is.na(Allele1))) # removing the indels

sumstat3 <- rbind(test1_final2, test2_final2) %>% arrange(CHR, BP) # 307,343 variants

remove <- ls()
remove <- as_tibble(remove) %>%
  filter(str_detect(value, "test"))
remove <- remove$value
rm(list = remove, remove)
rm(sumstat2_2, tmp)

tmp <- sumstat3 %>%
 select(CHR, BP) %>%
 unique()

for(i in 1:22) {
  write_delim(select(filter(tmp, CHR == i), BP), file = paste0(here("Output/Temp/"), "chr", i, "_snplist.txt"), delim = "\n")
}

# Now to pull out the MAF and alleles for all SNPs using the mfi files

#system(paste0('source ~/.bashrc; parallel "grep -Fwhf ', here("Output/Temp/"), 'chr{}_snplist.txt /Volumes/scratch/merrimanlab/ukbio/EGAD00010001474/splits/ukb_mfi_chr{}_v3.txt > ', here("Output/Temp/"), 'ukb_maf_info_chr{}.txt" ::: {1..22}'))

out <- tibble()
for(i in 1:22) {
  assign(paste0("chr", i, "_snps"), read_delim(here("Output/Temp", paste0("ukb_maf_info_chr", i, ".txt")), delim = "\t", col_names = FALSE) %>% mutate(CHR = i))
  out <- rbind(out, get(paste0("chr", i, "_snps")))
  rm(list = paste0("chr", i, "_snps"), i)
}

colnames(out) <- c("SNP1_mfi", "SNP2_mfi", "BP_mfi", "Allele1_mfi", "Allele2_mfi", "MAF_mfi", "Minor_Allele_mfi", "INFO_mfi", "CHR_mfi")

# sum(is.na(out$Allele1_mfi) | is.na(out$Allele2_mfi)) # all have allele1 and allele2

out <- out %>% 
  mutate(CHR_BP = paste0(CHR_mfi, "_", BP_mfi)) %>% 
  filter(Allele1_mfi %in% c("A", "C", "G", "T"),
         Allele2_mfi %in% c("A", "C", "G", "T"),
         MAF_mfi > 0.01,
         MAF_mfi < 0.99,
         INFO_mfi > 0.3) # 246,869 variants

sum(duplicated(out$CHR_BP)) # 224 multi-allelic sites

sumstat4 <- sumstat3 %>% 
  left_join(out, by = "CHR_BP") %>% 
  filter(!is.na(Allele1_mfi)) %>% 
  select(-CHR_mfi, -BP_mfi) # 246,017

geno2 <- geno %>% 
  filter(CHR_BP %in% sumstat4$CHR_BP,
         X5 %in% c("A", "C", "G", "T"),
         X6 %in% c("A", "C", "G", "T")) # only 244,655 variants

# length(unique(sumstat4$CHR_BP)) # same number as above, difference is multi-allelic sites

tmp <- sumstat4 %>% 
  filter(duplicated(CHR_BP)) %>% 
  pull(CHR_BP) %>% 
  unique() # 823 duplicated sites

tmp2 <- sumstat4 %>% 
  filter(!(CHR_BP %in% tmp)) # 244,015 biallelic sites

tmp3 <- sumstat4 %>% 
  filter(CHR_BP %in% tmp) # 2,002 total variants, need to go off the one that is in geno2

tmp4 <- tmp3 %>% 
  left_join(geno2, by = "CHR_BP") %>% 
  select(-X1, -X3, -X4) %>% 
  unique()

# test <- tmp4 %>% 
#   filter(Allele1 != Allele1_mfi | is.na(Allele1)) # Allele1_mfi can be changed to Allele1

tmp5 <- tmp4 %>% 
  filter(is.na(Allele2) | !is.na(Allele2) & Allele2 == Allele2_mfi)

tmp6 <- tmp5 %>% 
  filter((Allele1_mfi == X5 | Allele1_mfi == X6) & (Allele2_mfi == X5 | Allele2_mfi == X6))

tmp7a <- tmp6 %>% 
  filter(duplicated(CHR_BP))

tmp7b <- tmp6 %>% 
  filter(CHR_BP %in% tmp7a$CHR_BP)

tmp7 <- tmp6 %>% 
  filter(A1 == X5 | A1 == X6) # now all remaining variants are at unique locations

tmp2 <- sumstat4 %>% 
  filter(!(CHR_BP %in% tmp)) %>% 
  left_join(geno2, by = "CHR_BP") %>% 
  filter((Allele1_mfi == X5 | Allele1_mfi == X6) & (Allele2_mfi == X5 | Allele2_mfi == X6)) %>% 
  select(-X1, -X3, -X4)

sumstat4_1 <- rbind(tmp2, tmp7) %>% 
  arrange(CHR, BP) %>% 
  filter(Allele2 == Allele2_mfi | is.na(Allele2)) %>% 
  mutate(Allele1 = Allele1_mfi,
         Allele2 = Allele2_mfi) %>% 
  select(-Allele1_mfi, -Allele2_mfi)

# test <- sumstat4_1 %>% 
#   filter(!(A1 == X5 | A1 == X6))

sum(sumstat4_1$SNP1 != sumstat4_1$SNP2_mfi) # same

sum(sumstat4_1$SNP2 != sumstat4_1$SNP1_mfi) # 50 different

test <- sumstat4_1 %>% 
  filter(SNP2 != SNP1_mfi) # nothing to worry about, can remove mfi versions

sumstat5 <- sumstat4_1 %>% 
  rename(Effect_Allele = A1,
         INFO = INFO_mfi,
         MAF = MAF_mfi,
         Minor_Allele = Minor_Allele_mfi) %>% 
  select(-TEST, -NMISS, -CHR_BP, -X2, -X5, -X6, -SNP1_mfi, -SNP2_mfi)

test <- sumstat5 %>% 
  filter(Minor_Allele != Effect_Allele)

#summary(test$MAF) # all really close to 0.5 MAF, just need to flip OR, L95, and U95 then set Effect_Allele to Minor_Allele column

test <- test %>% 
  mutate(OR = 1/OR,
         tmp = 1/L95,
         tmp2 = 1/U95,
         L95 = tmp2,
         U95 = tmp,
         Effect_Allele = Minor_Allele) %>% 
  rename(EAF = MAF) %>% 
  select(-tmp, -tmp2, -Minor_Allele)

sumstat5_1 <- sumstat5 %>% 
  filter(Minor_Allele == Effect_Allele) %>% 
  select(-Minor_Allele) %>% 
  rename(EAF = MAF) %>% 
  rbind(test) %>% 
  arrange(CHR, BP)

test <- sumstat5_1 %>% 
  filter(Allele2 == Effect_Allele) %>% 
  rename(Alternate_Allele = Allele1) %>% 
  select(CHR, SNP, BP, Effect_Allele, Alternate_Allele, OR:SNP_ID, EAF:INFO)

test2 <- sumstat5_1 %>% 
  filter(Allele1 == Effect_Allele) %>% 
  rename(Alternate_Allele = Allele2) %>% 
  select(CHR, SNP, BP, Effect_Allele, Alternate_Allele, OR:SNP_ID, EAF:INFO)

sumstat_final <- rbind(test, test2) %>% 
  arrange(CHR, BP)

# save(sumstat_final, file = here("Output/sumstat_final.RData"))

rm(geno, geno2, out, sumstat, sumstat2, sumstat3, sumstat4, sumstat4_1, sumstat4_final, sumstat5, sumstat5_1, sumstat5_2, test, test2, tmp2, tmp3, tmp4, tmp5, tmp6, tmp7, tmp7a, tmp7b, tmp)
```

To get from these cleaned up summary statistics to the final list of SNPs for the PRS, I did the following:

1. I filtered out all SNPs with P-values greater than 5e-8.

2. I took each lead SNP within a 1 Mb window and used these to define 15 crude loci.

3. This list of lead SNPs was further filtered to only include one lead SNP per full locus.

    - The boundaries of these "full loci" were defined based on two consecutive genome-wide significant SNPs being more than 500 kb apart.
    
4. Next, SNPs in the UK Biobank BGEN files were extracted if they fit within the boundaries of these "full loci".

5. Conditional GWAS were run at each locus, conditioning on the lead SNP.

6. If there was a significant SNP (P < 5e-8) remaining after conditioning, the original lead SNP and the new lead SNP were used for a subsequent conditional GWAS at this locus.

7. This was repeated until no more significant SNPs (P < 5e-8) remained at each locus.

8. Locus zooms were plotted for each locus, using both the unconditioned and conditioned GWAS results.

9. Finally, the resulting list of 19 lead SNPs were saved in a single file ready for conversion to a PRS.

```{r new_loci_grouping, eval = F}
# Defining one SNP per locus ---------------------------------------------------------------------------
# First, filter out P < 5e-8 SNPs and arrange by P
load(here("Output/sumstat_final.RData"))
sumstat_signif <- sumstat_final %>% 
  filter(P <= 5e-8) %>% 
  arrange(P)

# Grouping into loci +- 500 kb of top SNPs
gout_top <- sumstat_signif %>% 
  slice(1)
gout2 <- sumstat_signif %>% 
  filter(!(CHR == gout_top$CHR[1] & BP %in% ((gout_top$BP[1] - 500000):(gout_top$BP[1] + 500000))))

while(nrow(gout2) > 0) {
  tmp <- gout2 %>% 
    slice(1)
  gout_top <- rbind(tmp, gout_top)
  gout2 <- gout2 %>% 
    filter(!(CHR == gout_top$CHR[1] & BP %in% ((gout_top$BP[1] - 500000):(gout_top$BP[1] + 500000))))
} 

gout_top <- gout_top %>% 
  arrange(CHR, BP)


# Finding regions of loci
sumstat_signif <- sumstat_signif %>% 
  arrange(CHR, BP)

out <- NA
for(i in 2:nrow(sumstat_signif)) {
  if(sumstat_signif$CHR[i] == sumstat_signif$CHR[i - 1]){
    out[i] <- sumstat_signif$BP[i] - sumstat_signif$BP[i - 1]
  } else {
    out[i] <- NA
  }
}

tmp <- sumstat_signif %>% 
  mutate(Diff = out,
         Diff2 = case_when(Diff < 500000 ~ Diff))

out <- sumstat_signif %>% slice(1)
for(i in 2:nrow(sumstat_signif)) {
  if(is.na(tmp$Diff2[i])){
    out <- rbind(out, sumstat_signif %>% slice(i - 1), sumstat_signif %>% slice(i))
  }
}
out <- rbind(out, sumstat_signif %>% slice(nrow(sumstat_signif)))

# Extracting regions
bgen_ranges <- out %>% select(CHR, BP)

tmp1 <- bgen_ranges %>% slice(seq(1, nrow(bgen_ranges), by = 2)) %>% rename(BP1 = BP)

tmp2 <- bgen_ranges %>% slice(seq(2, nrow(bgen_ranges), by = 2)) %>% rename(CHR.x = CHR, BP2 = BP)

bgen_ranges <- tmp1 %>% 
  cbind(tmp2) %>% 
  mutate(BP1 = BP1 - 50000,
         BP2 = BP2 + 50000) %>% 
  select(-CHR.x)

bgen_range1 <- bgen_ranges %>% 
  filter(CHR < 10) %>% 
  mutate(BGEN = paste0("0", CHR, ":", BP1, "-", BP2))

bgen_range2 <- bgen_ranges %>% 
  filter(CHR > 9) %>% 
  mutate(BGEN = paste0(CHR, ":", BP1, "-", BP2))

bgen_ranges <- rbind(bgen_range1, bgen_range2) %>% 
  arrange(CHR, BP1)

tmp <- bgen_ranges %>% 
  select(BGEN)

rm(bgen_range1, bgen_range2, tmp1, tmp2, out, gout2, i)
  
#write_delim(tmp, file = here("Output/Temp", "bgen_range.txt"), delim = "\n", col_names = F)

# Extracting all SNPs from biallelic sumstat that fit within boundaries of loci
loci <- tibble()
for(i in 1:nrow(bgen_ranges)){
  tmp <- sumstat_final %>% 
  filter(CHR == bgen_ranges$CHR[i] & between(BP, bgen_ranges$BP1[i], bgen_ranges$BP2[i]))
  loci <- rbind(loci, tmp)
}
loci <- loci %>% 
  mutate(SNP_ID2 = paste0(CHR, "_", BP, "_", Alternate_Allele, "_", Effect_Allele))

tmp <- loci %>% 
  mutate(BP2 = BP) %>% 
  select(CHR, BP, BP2, SNP)

#write_delim(tmp, file = here("Output/Temp", "loci_snps.txt"), delim = "\t", col_names = F)

out <- c()
for(i in 1:nrow(bgen_ranges)){
  tmp <- gout_top %>% 
    filter(CHR == bgen_ranges$CHR[i] & between(BP, bgen_ranges$BP1[i], bgen_ranges$BP2[i])) %>% 
    arrange(P) %>% 
    slice(1)
  out <- rbind(out, tmp)
}

gout_top <- out %>% 
  cbind(bgen_ranges %>% select(-CHR))

rm(tmp, bgen_ranges, out, i)

# Extracting all SNPs at loci from bgen files and converting to plink format -------------------------------
# system(paste0('source ~/.bashrc; parallel "bgenix -g /Volumes/archive/merrimanlab_nobackup/ukbio/EGAD00010001474/ukb_imp_chr{}_v3.bgen -vcf -incl-range ', here("Output/Temp", "bgen_range.txt"), ' | bcftools reheader -h /Volumes/archive/merrimanlab_nobackup/ukbio/EGAD00010001474/bgen_to_vcf/new_header.txt | bcftools annotate --rename-chrs /Volumes/archive/merrimanlab_nobackup/ukbio/EGAD00010001474/bgen_to_vcf/rename_contigs.txt | bgzip -c > ', here("Output/Temp", "chr"), '{}_forclumping.vcf.gz" ::: ', paste(unique(gout_top$CHR), collapse = " ")))

# system(paste0('source ~/.bashrc; parallel "plink1.9b4.9 --vcf ', here("Output/Temp/"), 'chr{}_forclumping.vcf.gz --extract range ', here("Output/Temp/loci_snps.txt"), ' --pheno ', here("Data/GWAS", "gout_gwas_covar.covar"), ' --pheno-name plink_goutaff --update-sex ', here("Data/GWAS", "gout_gwas_keep_ids_w_sex.txt"), ' --geno 0.1 --maf 0.01 --hwe 0.000001 --make-bed --out ', here("Output/Temp/"), 'chr{}_tmp" ::: ', paste(unique(gout_top$CHR), collapse = " ")))


# Reading the bim files into R and converting their identifiers to just the rsid
file_names <- list.files(here("Output/Temp/"))[str_detect(list.files(here("Output/Temp/")), "_tmp.bim")]

for(i in file_names){
  assign(i, read_delim(paste0(here("Output/Temp/"), i), delim = "\t", col_names = F))
  assign(i, get(i) %>% left_join(loci, (by = c("X1" = "CHR", "X4" = "BP"))) %>% mutate(SNP_clean = case_when(is.na(RSID) ~ SNP_ID, TRUE ~ RSID)))
  # assign(paste0(i, "_notequal"), get(i) %>% filter(X2 != SNP)) # all identical
  assign(paste0("new_", i), get(i) %>% select(X1, SNP_clean, X3:X6))
  write_delim(get(paste0("new_", i)), file = paste0(here("Output/Temp/"), i), delim = "\t", col_names = F)
}

rm(list = ls()[str_detect(ls(), ".bim")], i, file_names)



# Running the conditional GWAS ----------------------------------------
# Split up the plink files to have one locus per file (saves on computational time)
gout_top2 <- gout_top %>% 
  select(CHR, BP1, BP2, RSID)

for(i in 1:nrow(gout_top2)){
  tmp <- gout_top2 %>% slice(i)
  write_delim(tmp, file = paste0(here("Output/Temp/"), "extractrange_", tmp$RSID, ".txt"), delim = "\t", col_names = F)
}

#system(paste0('source ~/.bashrc; parallel --xapply "plink1.9b6.10 --bfile {1}/Output/Temp/chr{2}_tmp --extract range {1}/Output/Temp/extractrange_{3}.txt --make-bed --out {1}/Output/Temp/{3}" ::: ', paste(rep(here(), nrow(gout_top)), collapse = " "), ' ::: ', paste(gout_top$CHR, collapse = " "), ' ::: ', paste(gout_top$RSID, collapse = " ")))

#system(paste0('source ~/.bashrc; parallel "cat {1}/Output/Temp/{2}.bim | cut -f 2 > {1}/Output/Temp/{2}_snps; split -d -n l/10 {1}/Output/Temp/{2}_snps {1}/Output/Temp/{2}_snps_split" ::: ', here(), ' ::: ', paste(gout_top$RSID, collapse = " ")))

# First round of conditioning

#system(paste0('source ~/.bashrc; parallel "echo {2} >> {1}/Output/Temp/{2}_snps_split{3}" ::: ', here(), ' ::: ', paste(gout_top$RSID, collapse = " "), ' ::: ', paste(paste0(0, 0:9), collapse = " ")))

system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --bfile {1}/Output/Temp/{2} --extract {1}/Output/Temp/{2}_snps_split{3} --logistic sex --ci 0.95 --covar {1}/Data/GWAS/gout_gwas_covar.covar --covar-name Age,pc1-pc40 --condition {2} --out {1}/Output/Temp/{2}_split{3}" ::: ', here(), ' ::: ', paste(gout_top$RSID, collapse = " "), ' ::: ', paste(paste0(0, 0:9), collapse = " ")))

# Reading all the outputs into R
file_names <- list.files(here("Output/Temp/"))[str_detect(list.files(here("Output/Temp/")), regex("rs[0-9]+_split[0-9]+.assoc.logistic"))]

for(i in file_names){
  assign(i, read.table(paste0(here("Output/Temp/"), i), header = T) %>% filter(TEST == "ADD"))
}

tmp <- c()
for(i in 1:nrow(gout_top)){
  tmp3 <- c()
  for(j in 0:9){
    tmp2 <- get(paste0(gout_top$RSID[i], "_split0", j, ".assoc.logistic"))
    tmp3 <- rbind(tmp3, tmp2)
  }
  assign(paste0(gout_top$RSID[i], "_gwas"), tmp3 %>% na.omit())
  tmp3 <- tmp3 %>% select(SNP, P) %>% arrange(P) %>% slice(1)
  tmp <- rbind(tmp, tmp3)
}
tmp <- tmp %>% 
  rename(new_lead = SNP, new_p = P)
gout_top2 <- gout_top %>% 
  cbind(tmp)

gout_top_resid <- gout_top2 %>%
  filter(new_p < 5e-8)

rm(list = ls()[str_detect(ls(), ".assoc")], i, tmp, tmp2, file_names, tmp3, j)


# Second round of conditioning
system(paste0('source ~/.bashrc; parallel "echo {4} >> {1}/Output/Temp/{2}_snps_split{3}" ::: ', here(), ' ::: ', paste(gout_top_resid$RSID, collapse = " "), ' ::: ', paste(paste0(0, 0:9), collapse = " "), ' ::: ', paste(gout_top_resid$new_lead, collapse = " ")))

system(paste0('source ~/.bashrc; parallel --xapply "echo $', paste0("'{2}\n{3}'"), ' > {1}/Output/Temp/{2}_2" ::: ', here(), ' ::: ', paste(gout_top_resid$RSID, collapse = " "), ' ::: ', paste(gout_top_resid$new_lead, collapse = " ")))

system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --bfile {1}/Output/Temp/{2} --extract {1}/Output/Temp/{2}_snps_split{3} --logistic sex --ci 0.95 --covar {1}/Data/GWAS/gout_gwas_covar.covar --covar-name Age,pc1-pc40 --condition-list {1}/Output/Temp/{2}_2 --out {1}/Output/Temp/{2}_split{3}_2" ::: ', here(), ' ::: ', paste(gout_top_resid$RSID, collapse = " "), ' ::: ', paste(paste0(0, 0:9), collapse = " ")))

file_names <- list.files(here("Output/Temp/"))[str_detect(list.files(here("Output/Temp/")), regex("rs[0-9]+_split[0-9]+_2.assoc.logistic"))]

for(i in file_names){
  assign(i, read.table(paste0(here("Output/Temp/"), i), header = T) %>% filter(TEST == "ADD"))
}

tmp <- c()
for(i in 1:nrow(gout_top_resid)){
  tmp3 <- c()
  for(j in 0:9){
    tmp2 <- get(paste0(gout_top_resid$RSID[i], "_split0", j, "_2.assoc.logistic"))
    tmp3 <- rbind(tmp3, tmp2)
  }
  assign(paste0(gout_top_resid$RSID[i], "_gwas2"), tmp3 %>% na.omit())
  tmp3 <- tmp3 %>% select(SNP, P) %>% arrange(P) %>% slice(1)
  tmp <- rbind(tmp, tmp3)
}

tmp <- tmp %>% 
  rename(new_lead2 = SNP, new_p2 = P)
gout_top3 <- gout_top_resid %>% 
  cbind(tmp)

gout_top_resid2 <- gout_top3 %>%
  filter(new_p2 < 5e-8)

rm(list = ls()[str_detect(ls(), ".assoc")], i, tmp, tmp2, file_names, tmp3, j)

# Third round of conditioning
system(paste0('source ~/.bashrc; parallel "echo {4} >> {1}/Output/Temp/{2}_snps_split{3}" ::: ', here(), ' ::: ', paste(gout_top_resid2$RSID, collapse = " "), ' ::: ', paste(paste0(0, 0:9), collapse = " "), ' ::: ', paste(gout_top_resid2$new_lead2, collapse = " ")))

system(paste0('source ~/.bashrc; parallel --xapply "echo $', paste0("'{2}\n{3}\n{4}'"), ' > {1}/Output/Temp/{2}_3" ::: ', here(), ' ::: ', paste(gout_top_resid2$RSID, collapse = " "), ' ::: ', paste(gout_top_resid2$new_lead, collapse = " "), ' ::: ', paste(gout_top_resid2$new_lead2, collapse = " ")))

system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --bfile {1}/Output/Temp/{2} --extract {1}/Output/Temp/{2}_snps_split{3} --logistic sex --ci 0.95 --covar {1}/Data/GWAS/gout_gwas_covar.covar --covar-name Age,pc1-pc40 --condition-list {1}/Output/Temp/{2}_3 --out {1}/Output/Temp/{2}_split{3}_3" ::: ', here(), ' ::: ', paste(gout_top_resid2$RSID, collapse = " "), ' ::: ', paste(paste0(0, 0:9), collapse = " ")))

file_names <- list.files(here("Output/Temp/"))[str_detect(list.files(here("Output/Temp/")), regex("rs[0-9]+_split[0-9]+_3.assoc.logistic"))]

for(i in file_names){
  assign(i, read.table(paste0(here("Output/Temp/"), i), header = T) %>% filter(TEST == "ADD"))
}

tmp <- c()
for(i in 1:nrow(gout_top_resid2)){
  tmp3 <- c()
  for(j in 0:9){
    tmp2 <- get(paste0(gout_top_resid2$RSID[i], "_split0", j, "_3.assoc.logistic"))
    tmp3 <- rbind(tmp3, tmp2)
  }
  assign(paste0(gout_top_resid2$RSID[i], "_gwas3"), tmp3 %>% na.omit())
  tmp3 <- tmp3 %>% select(SNP, P) %>% arrange(P) %>% slice(1)
  tmp <- rbind(tmp, tmp3)
}

tmp <- tmp %>% 
  rename(new_lead3 = SNP, new_p3 = P)
gout_top4 <- gout_top_resid2 %>% 
  cbind(tmp)

gout_top_resid3 <- gout_top4 %>%
  filter(new_p3 < 5e-8)

rm(list = ls()[str_detect(ls(), ".assoc")], i, tmp, tmp2, file_names, tmp3, j)



# Locus zooms ---------------------------------------
# Loading in code and gene list
source(here("Script/Functions/locus_zoom.R"))

UCSC_GRCh37_Genes_UniqueList.txt <- as.data.frame(read_delim(here("Data/GWAS/UCSC_GRCh37_Genes_UniqueList.txt"), delim = "\t"))


# Plotting locus zooms of original GWAS

# Calculating LD
system(paste0('source ~/.bashrc; parallel --xapply "plink1.9b6.10 --bfile ', here("Output/Temp/"), 'chr{1}_tmp --r2 inter-chr --ld-snp {2} --ld-window-r2 0 --out ', here("Output/Temp/"), 'chr{1}_{2}_ld" ::: ', paste(gout_top$CHR, collapse = " "), ' ::: ', paste(gout_top$RSID, collapse = " ")))

# Reading the LD reports back into R
for(i in 1:nrow(gout_top)){
  assign(paste0("chr", gout_top$CHR[i], "_", gout_top$RSID[i], "_ld"), read_table(paste0(here("Output/Temp/"), "chr", gout_top$CHR[i], "_", gout_top$RSID[i], "_ld.ld")))
}

# Making full list of SNPs for labelling
first_round <- gout_top_resid %>% 
  select(new_lead) %>% 
  rename(RSID = new_lead)

second_round <- gout_top_resid2 %>% 
  select(new_lead2) %>% 
  rename(RSID = new_lead2)

gout_top_full <- gout_top %>% 
  select(RSID) %>% 
  rbind(first_round, second_round) %>% 
  left_join(loci, by = "RSID") %>% 
  arrange(CHR, BP)

# Plotting the locus zooms
for(i in 1:nrow(gout_top)){
  locus.zoom(data = loci %>% mutate(SNP = RSID) %>% filter(!is.na(SNP), CHR == gout_top$CHR[i] & between(BP, gout_top$BP1[i], gout_top$BP2[i])),
             region = c(gout_top$CHR[i], gout_top$BP1[i], gout_top$BP2[i]),
             offset_bp = 0,
             ld.file = get(paste0("chr", gout_top$CHR[i], "_", gout_top$RSID[i], "_ld")),
             genes.data = UCSC_GRCh37_Genes_UniqueList.txt,
             plot.title = paste0("Unconditioned ", gout_top$RSID[i], " Locus Zoom"),
             file.name = paste0(here("Output/Plots/"), "Chr", gout_top$CHR[i], "_", gout_top$BP1[i], "_", gout_top$BP2[i], "_", gout_top$RSID[i], "_unconditioned", ".jpg"),
             secondary.snp = gout_top_full$RSID,
             secondary.label = TRUE)
}


# Plotting locus zooms of first round of conditioning

# Remaking LD based on ALL new lead SNPs
system(paste0('source ~/.bashrc; parallel --xapply "plink1.9b6.10 --bfile ', here("Output/Temp/"), 'chr{1}_tmp --r2 inter-chr --ld-snp {2} --ld-window-r2 0 --out ', here("Output/Temp/"), 'chr{1}_{2}_ld" ::: ', paste(gout_top2$CHR, collapse = " "), ' ::: ', paste(gout_top2$new_lead, collapse = " ")))

# Reading the LD reports back into R
for(i in 1:nrow(gout_top2)){
  assign(paste0("chr", gout_top2$CHR[i], "_", gout_top2$new_lead[i], "_ld"), read_table(paste0(here("Output/Temp/"), "chr", gout_top2$CHR[i], "_", gout_top2$new_lead[i], "_ld.ld")))
}

# Plotting locus zooms
for(i in 1:nrow(gout_top2)){
  locus.zoom(data = get(paste0(gout_top2$RSID[i], "_gwas")),
             region = c(gout_top$CHR[i], gout_top2$BP1[i], gout_top2$BP2[i]),
             offset_bp = 0,
             ld.file = get(paste0("chr", gout_top2$CHR[i], "_", gout_top2$new_lead[i], "_ld")),
             genes.data = UCSC_GRCh37_Genes_UniqueList.txt,
             plot.title = paste0("Conditioned on ", gout_top2$RSID[i]),
             file.name = paste0(here("Output/Plots/"), "Chr", gout_top2$CHR[i], "_", gout_top2$BP1[i], "_", gout_top2$BP2[i], "_", gout_top2$RSID[i], "_condition_", gout_top2$RSID[i], ".jpg"),
             secondary.snp = gout_top_full$RSID,
             secondary.label = TRUE)
}


# Plotting locus zooms of second round of conditioning

# Remaking LD based on new lead SNPs
system(paste0('source ~/.bashrc; parallel --xapply "plink1.9b6.10 --bfile ', here("Output/Temp/"), 'chr{1}_tmp --r2 inter-chr --ld-snp {2} --ld-window-r2 0 --out ', here("Output/Temp/"), 'chr{1}_{2}_ld" ::: ', paste(gout_top3$CHR, collapse = " "), ' ::: ', paste(gout_top3$new_lead2, collapse = " ")))

# Reading the LD reports back into R
for(i in 1:nrow(gout_top3)){
  assign(paste0("chr", gout_top3$CHR[i], "_", gout_top3$new_lead2[i], "_ld"), read_table(paste0(here("Output/Temp/"), "chr", gout_top3$CHR[i], "_", gout_top3$new_lead2[i], "_ld.ld")))
}

# Plotting locus zooms
for(i in 1:nrow(gout_top3)){
  locus.zoom(data = get(paste0(gout_top3$RSID[i], "_gwas2")),
             region = c(gout_top3$CHR[i], gout_top3$BP1[i], gout_top3$BP2[i]),
             offset_bp = 0,
             ld.file = get(paste0("chr", gout_top3$CHR[i], "_", gout_top3$new_lead2[i], "_ld")),
             genes.data = UCSC_GRCh37_Genes_UniqueList.txt,
             plot.title = paste0("Conditioned on ", gout_top3$RSID[i], " and ", gout_top3$new_lead[i]),
             file.name = paste0(here("Output/Plots/"), "Chr", gout_top3$CHR[i], "_", gout_top3$BP1[i], "_", gout_top3$BP2[i], "_", gout_top3$RSID[i], "_condition_", gout_top3$RSID[i], "and", gout_top3$new_lead[i], ".jpg"),
             secondary.snp = gout_top_full$RSID,
             secondary.label = TRUE)
}


# Plotting locus zooms of third round of conditioning

# Remaking LD based on new lead SNP
system(paste0('source ~/.bashrc; parallel --xapply "plink1.9b6.10 --bfile ', here("Output/Temp/"), 'chr{1}_tmp --r2 inter-chr --ld-snp {2} --ld-window-r2 0 --out ', here("Output/Temp/"), 'chr{1}_{2}_ld" ::: ', paste(gout_top4$CHR, collapse = " "), ' ::: ', paste(gout_top4$new_lead3, collapse = " ")))

# Reading the LD reports back into R
for(i in 1:nrow(gout_top4)){
  assign(paste0("chr", gout_top4$CHR[i], "_", gout_top4$new_lead3[i], "_ld"), read_table(paste0(here("Output/Temp/"), "chr", gout_top4$CHR[i], "_", gout_top4$new_lead3[i], "_ld.ld")))
}

# Plotting locus zooms
for(i in 1:nrow(gout_top4)){
  locus.zoom(data = get(paste0(gout_top4$RSID[i], "_gwas3")),
             region = c(gout_top4$CHR[i], gout_top4$BP1[i], gout_top4$BP2[i]),
             offset_bp = 0,
             ld.file = get(paste0("chr", gout_top4$CHR[i], "_", gout_top4$new_lead3[i], "_ld")),
             genes.data = UCSC_GRCh37_Genes_UniqueList.txt,
             plot.title = paste0("Conditioned on ", gout_top4$RSID[i], " and ", gout_top4$new_lead[i], " and ", gout_top4$new_lead2[i]),
             file.name = paste0(here("Output/Plots/"), "Chr", gout_top4$CHR[i], "_", gout_top4$BP1[i], "_", gout_top4$BP2[i], "_", gout_top4$RSID[i], "_condition_", gout_top4$RSID[i], "and", gout_top4$new_lead[i], "and", gout_top4$new_lead2[i], ".jpg"),
             secondary.snp = gout_top_full$RSID,
             secondary.label = TRUE)
}



# Combining all GWAS results together into final list ----------------------------------------------
regions <- gout_top %>%
  select(CHR, BP1, BP2)

out <- c()
for(i in 1:nrow(regions)){
  tmp <- gout_top_full %>% 
    filter(CHR == regions$CHR[i] & between(BP, regions$BP1[i], regions$BP2[i])) %>% 
    mutate(BP1 = regions$BP1[i], 
           BP2 = regions$BP2[i])
  out <- rbind(out, tmp)
}

gout_top_full <- out

# For each of the loci with multiple SNPs, I need to test the association of each SNP after adjusting for all others at the locus
# So first lets pull out all SNPs from those loci
multi_snps <- gout_top_full %>% 
  filter(BP1 %in% names(table(gout_top_full$BP1)[table(gout_top_full$BP1) > 1]))

tmp <- multi_snps %>% select(RSID)

write_delim(tmp, file = here("Output/Temp/snps_to_extract.txt"), delim = "\n", col_names = F)

system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --bfile ', here("Output/Temp/"), 'chr{1}_tmp --extract ', here("Output/Temp/snps_to_extract.txt"), ' --make-bed --out ', here("Output/Temp/"), 'chr{1}_test" ::: ', paste(unique(multi_snps$CHR), collapse = " ")))

system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'chr4_test --bmerge ', here("Output/Temp/"), 'chr11_test --make-bed --out ', here("Output/Temp/"), 'merged_test'))

for(i in 4:7){
  system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'merged_test --logistic sex --ci 0.95 --covar ', here("Data/GWAS", "gout_gwas_covar.covar"), ' --covar-name Age,pc1-pc40 --condition ', multi_snps$RSID[i], ' --out ', here("Output/Temp/"), 'final_gwas_', multi_snps$RSID[i]))
}

for(i in 1:3){
  write_delim(multi_snps %>% slice(1:3) %>% select(RSID) %>% filter(RSID != multi_snps$RSID[i]), file = paste0(here("Output/Temp/"), "conditionlist_", multi_snps$RSID[i], ".txt"), delim = "\n", col_names = F)
  
  system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'merged_test --logistic sex --ci 0.95 --covar ', here("Data/GWAS", "gout_gwas_covar.covar"), ' --covar-name Age,pc1-pc40 --condition-list ', here("Output/Temp/"), 'conditionlist_', multi_snps$RSID[i], '.txt --out ', here("Output/Temp/"), 'final_gwas_', multi_snps$RSID[i]))
}

file_names <- list.files(here("Output/Temp/"))[str_detect(list.files(here("Output/Temp/")), "final_gwas_.+logistic")]

for(i in file_names){
  assign(i, read.table(paste0(here("Output/Temp/"), i), header = T) %>% filter(TEST == "ADD"))
}

tmp <- c()
for(i in 1:3){
  tmp2 <- get(paste0("final_gwas_", multi_snps$RSID[i], ".assoc.logistic")) %>% slice(i)
  tmp <- rbind(tmp, tmp2)
}

tmp2 <- c()
for(i in 4:7){
  tmp3 <- get(paste0("final_gwas_", multi_snps$RSID[i], ".assoc.logistic")) %>% slice(-(1:3))
  tmp2 <- rbind(tmp2, tmp3)
}

tmp2 <- tmp2 %>% slice(5, 2, 15, 12)

multi_snps2 <- rbind(tmp, tmp2)

tmp2 <- multi_snps2 %>% 
  select(CHR:BP, OR:U95, P)

multi_snps3 <- left_join(multi_snps, tmp2, by = c("CHR", "BP"))

system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'merged_test --r2 inter-chr --ld-window-r2 0 --out ', here("Output/Temp/"), 'merged_ld'))

merged_ld <- read_table(paste0(here("Output/Temp/"), "merged_ld.ld"))

# test multicollinearity in SLC2A9 model in next Rmd

single_snps <- gout_top_full %>% 
  filter(!(BP1 %in% names(table(gout_top_full$BP1)[table(gout_top_full$BP1) > 1])))

single_snps2 <- single_snps %>% 
  select(CHR, RSID, BP:Alternate_Allele, OR:U95, P, EAF, INFO, BP1, BP2)

multi_snps4 <- multi_snps3 %>% 
  select(CHR, RSID, BP:Alternate_Allele, OR.y:U95.y, P.y, OR.x:U95.x, P.x, EAF, INFO, BP1, BP2) %>% 
  rename(OR = OR.y,
         SE = SE.y, 
         L95 = L95.y,
         U95 = U95.y,
         P = P.y,
         OR_old = OR.x,
         SE_old = SE.x, 
         L95_old = L95.x,
         U95_old = U95.x,
         P_old = P.x)

gout_top_final <- full_join(multi_snps4, single_snps2) %>% 
  arrange(CHR, BP)

# Flipping allele order + OR etc so effect allele is always the gout risk allele + labelling based on locus zooms
smallOR <- gout_top_final %>% 
  filter(OR < 1) %>% 
  mutate(OR = as.numeric(signif(1/OR, digits = 4)),
         tmp_L = as.numeric(signif(1/L95, digits = 4)),
         tmp_U = as.numeric(signif(1/U95, digits = 4)),
         U95 = tmp_L,
         L95 = tmp_U,
         OR_old = as.numeric(round(1/OR_old, digits = 3)),
         tmp_L_old = as.numeric(round(1/L95_old, digits = 3)),
         tmp_U_old = as.numeric(round(1/U95_old, digits = 3)),
         U95_old = tmp_L_old,
         L95_old = tmp_U_old,
         EAF = 1 - EAF) %>% 
  rename(allele2 = Effect_Allele,
         allele1 = Alternate_Allele) %>% 
  rename(Alternate_Allele = allele2,
         Effect_Allele = allele1) %>% 
  select(CHR:BP, Effect_Allele, Alternate_Allele, OR:BP2)
bigOR <- gout_top_final %>% 
  filter(OR > 1)
gout_top_final <- full_join(smallOR, bigOR) %>% 
  arrange(CHR, BP) %>% 
  mutate(Locus_Name = c("PDZK1", "TRIM46", "GCKR", "SFMBT1", "SLC2A9", "SLC2A9", "SLC2A9", "ABCG2", "ABCG2", "SLC17A1", "ZSCAN31", "MLXIPL", "SLC16A9", "SLC22A11", "SLC22A11", "OVOL1", "R3HDM2", "MLXIP", "PNPLA3"))

UKBB_Gene_OR <- gout_top_final
  
save(UKBB_Gene_OR, file = here("Output/UKBB_Gene_OR.RData"))

# Cleaning up
rm(list = ls()[str_detect(ls(), "^chr|^gout_top|^final_|gwas$")], bigOR, first_round, multi_snps, multi_snps2, multi_snps3, multi_snps4, merged_ld, out, regions, second_round, single_snps, smallOR, third_round, tmp, tmp2, tmp3, file_names, i, sumstat_signif, single_snps2, loci, sumstat_final, UCSC_GRCh37_Genes_UniqueList.txt, check.rsid, elog10, gene.position, get.ld, get.region, locus.zoom, merge.gene.colour, merge.plot.dat, plot.locus, plot.secondary.point, read.plink.loci, round.up, subset.data)
```

### Locus-Zooms {.tabset}

All of the locus zooms are plotted below in separate tabs:

```{r Plots_tables, warning = F, message = F}
file_names <- list.files(here("Output/Plots"), full.names = T)[str_detect(list.files(here("Output/Plots"), full.names = T), "Chr")]

tmp <- file_names %>% 
  as_tibble() %>% 
  separate(value, sep = "_", into = c(NA, "X2", "BP1", NA, NA, "Cond", "CondSNPs")) %>% 
  rownames_to_column() %>% 
  mutate(Cond1 = case_when(Cond == "unconditioned.jpg" ~ FALSE, TRUE ~ TRUE), 
         BP1 = as.numeric(BP1), 
         rowname = as.numeric(rowname)) %>% 
  separate(X2, sep = "/", into = c(NA, NA, NA, NA, NA, NA, NA, "CHR")) %>%
  mutate(CHR = as.numeric(str_replace(CHR, "Chr", ""))) %>% 
  arrange(CHR, BP1, Cond1, CondSNPs)

tmp1 <- tmp %>% 
  pull(BP1) %>% 
  unique() %>% 
  as_tibble() %>% 
  rename(BP1 = value)

load(here("Output/UKBB_Gene_OR.RData"))

tmp2 <- UKBB_Gene_OR %>% 
  pull(Locus_Name) %>% 
  unique() %>% 
  as_tibble() %>% 
  rename(Locus_Name = value) %>% 
  cbind(tmp1)

tmp3 <- tmp %>% 
  left_join(tmp2, by = "BP1")

tmp4 <- tmp3 %>% 
  mutate(CondSNPs2 = str_remove(CondSNPs, ".jpg")) %>% 
  separate(CondSNPs2, sep = "and", into = c("SNP1", "SNP2", "SNP3", "SNP4")) %>% 
  mutate(SNPs = case_when(is.na(SNP2) ~ SNP1,
                          !is.na(SNP2) & is.na(SNP3) ~ str_c(SNP1, SNP2, sep = " and "),
                          !is.na(SNP3) & is.na(SNP4) ~ str_c(SNP1, SNP2, SNP3, sep = " and "),
                          !is.na(SNP4) ~ str_c(SNP1, SNP2, SNP3, SNP4, sep = " and ")),
         Plot_Name = case_when(!Cond1 ~ paste0(Locus_Name, " (Uncond.)"),
                               Cond1 ~ paste0(Locus_Name, " (Cond. on ", SNPs, ")")))

file_names2 <- file_names[tmp$rowname]

names(file_names2) <- tmp4$Plot_Name

template <- c(
    "#### {{nm}}\n",
    "```{r, echo = FALSE}\n",
    "include_graphics(file_names2['{{nm}}'])\n",
    "```\n",
    "\n"
  )

plots <- lapply(
  tmp4$Plot_Name, 
  function(nm) knit_expand(text = template)
)
```

`r knitr::knit(text = unlist(plots))`

### Summary

In summary, I produced a list of SNPs that will be used to create a PRS. This uses SNPs genotyped on the Human CoreExome v1.0 chip. This will ensure that no imputation was done on Polynesian individuals, which should make the genotypes more reliable. Three of the 15 total loci had more than one partially independent genome-wide significant signal. These were at SLC2A9 (3 hits), ABCG2 (2 hits), and SLC22A11 (2 hits). 
