# Preparing Phenotype files and making PRS

The purpose of this document is to generate cleaned up phenotype files for each cohort, with the European and Polynesian polygenic risk scores (PRSs) included for each. It contains the code for going from the raw phenotype and genotype data (in combination with the SNP lists generated before) to the finalized data frames for analysis.

```{r dataprep, message = FALSE, warning = FALSE, cache = TRUE}
# Loading PRS SNPs for European and Polynesian analyses
load(here("Output/UKBB_Gene_OR.RData"))
load(here("Output/Poly_Gene_OR.RData"))


# Making files for extracting SNPs from the CoreExome VCFs using plink
if(!file.exists(here("Output/Temp/UKBB_SNPs_Plink.txt"))){
  for_plink <- UKBB_Gene_OR %>% 
    select(CHR, BP, RSID) %>% 
    mutate(BP2 = BP) %>% 
    select(CHR, BP, BP2, RSID)
  write_delim(for_plink, file = here("Output/Temp/UKBB_SNPs_Plink.txt"), col_names = F)
  rm(for_plink)
}

if(!file.exists(here("Output/Temp/Poly_SNPs_Plink.txt"))){
  for_plink <- Poly_Gene_OR %>% 
    select(CHR, BP, RSID) %>% 
    mutate(BP2 = BP) %>% 
    select(CHR, BP, BP2, RSID)
  write_delim(for_plink, file = here("Output/Temp/Poly_SNPs_Plink.txt"), col_names = F)
  rm(for_plink)
}

# Extracting SNPs from the CoreExome VCFs using plink

#system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --vcf /Volumes/archive/merrimanlab/raid_backup/New_Zealand_Chip_data/CoreExome/QC_MergedBatches/Imputed_Genotypes/QC1-10_Impute_EUR_only/CZ-MB1.2-QC1.10_EUR_imputed_chr{}.vcf.gz --extract range ', here("Output/Temp/UKBB_SNPs_Plink.txt"), ' --make-bed --out ', here("Output/Temp/UKBB_SNPs_"), 'chr{}" ::: ', paste(unique(UKBB_Gene_OR$CHR), collapse = " ")))

#system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --bfile ', here("Output/Temp/UKBB_SNPs_"), 'chr{} --set-missing-var-ids @:# --make-bed --out ', here("Output/Temp/UKBB_SNPs_"), 'chr{}_2" ::: ', paste(unique(UKBB_Gene_OR$CHR), collapse = " ")))

#write_delim(as_tibble(paste0(here("Output/Temp/"), "UKBB_SNPs_chr", unique(UKBB_Gene_OR$CHR), "_2")), file = here("Output/Temp/mergefile_prs1.txt"), delim = "\n", col_names = F)

#system(paste0('source ~/.bashrc; plink1.9b6.10 --merge-list ', here("Output/Temp/"), 'mergefile_prs1.txt --make-bed --out ', here("Output/Temp/"), 'merged_PRS_UKBB'))

#system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'merged_PRS_UKBB --recode --out ', here("Output/Temp/"), 'merged_PRS_UKBB'))

#system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile /Volumes/archive/merrimanlab/raid_backup/New_Zealand_Chip_data/CoreExome/QC_MergedBatches/Final_Data/CZ-MB1.2-QC1.10_CoreExome24-1.0-3_genotyped-QCd_rsIDconverted --extract range ', here("Output/Temp/Poly_SNPs_Plink.txt"), ' --make-bed --out ', here("Output/Temp/Poly_SNPs")))

#system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'Poly_SNPs --recode --out ', here("Output/Temp/"), 'Poly_SNPs'))



# Making bgen file for extracting SNPs from the UK Biobank
if(!file.exists(here("Output/Temp/PRS_SNPs_BGEN.txt"))){
  tmp <- UKBB_Gene_OR %>%
    select(CHR, BP)

  tmp2 <- Poly_Gene_OR %>%
    select(CHR, BP)

  tmp3 <- rbind(tmp, tmp2) %>%
    arrange(CHR, BP) %>%
    unique()

  bgen_range1 <- tmp3 %>%
    filter(CHR < 10) %>%
    mutate(BGEN = paste0("0", CHR, ":", BP, "-", BP))

  bgen_range2 <- tmp3 %>%
    filter(CHR > 9) %>%
    mutate(BGEN = paste0(CHR, ":", BP, "-", BP))

  bgen_range <- rbind(bgen_range1, bgen_range2) %>%
    arrange(CHR, BP) %>%
    select(BGEN)

  write_delim(bgen_range, file = here("Output/Temp/PRS_SNPs_BGEN.txt"), delim = "\n", col_names = F)

  rm(tmp, tmp2, tmp3, bgen_range1, bgen_range2, bgen_range)
}

# Extracting SNPs from UK Biobank and converting to merged plink file

#system(paste0('source ~/.bashrc; parallel "bgenix -g /Volumes/scratch/merrimanlab/ukbio/EGAD00010001474/ukb_imp_chr{}_v3.bgen -vcf -incl-range ', here("Output/Temp", "PRS_SNPs_BGEN.txt"), ' | bcftools reheader -h /Volumes/scratch/merrimanlab/ukbio/EGAD00010001474/bgen_to_vcf/new_header.txt | bcftools annotate --rename-chrs /Volumes/scratch/merrimanlab/ukbio/EGAD00010001474/bgen_to_vcf/rename_contigs.txt | bgzip -c > ', here("Output/Temp", "chr"), '{}_forPRS.vcf.gz" ::: ', paste(unique(UKBB_Gene_OR$CHR), collapse = " ")))

#system(paste0('source ~/.bashrc; parallel "plink1.9b4.9 --vcf ', here("Output/Temp/"), 'chr{}_forPRS.vcf.gz --make-bed --out ', here("Output/Temp/"), 'chr{}_PRS" ::: ', paste(unique(UKBB_Gene_OR$CHR), collapse = " ")))

#write_delim(as_tibble(paste0(here("Output/Temp/"), "chr", unique(UKBB_Gene_OR$CHR), "_PRS")), file = here("Output/Temp/mergefile_prs.txt"), delim = "\n", col_names = F)

#system(paste0('source ~/.bashrc; plink1.9b6.10 --merge-list ', here("Output/Temp/"), 'mergefile_prs.txt --make-bed --out ', here("Output/Temp/"), 'merged_PRS'))

#system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/Temp/"), 'merged_PRS --recode --out ', here("Output/Temp/"), 'merged_PRS'))



# Making phenotype files ----------------------------------------------------------------------------------------
if(file.exists(here("Output/Phenotypes.RData"))){
  load(here("Output/Phenotypes.RData"))
  } else {
  # CoreExome QC 1-10 Phenotype file
  CoreExPheno <- read_delim(here("Data/Phenotypes/CZ-MB1.2-QC1.10_MergedPhenotypes_20082020.txt"), delim = "\t")
  
  # European cohorts = All Ardea (split into LASSO and other), EuroGout (includes EireGout), Gout in Aotearoa (combine with AGRIA + DM + RD + NP + LPA => ANZ cohort)
  All_Euro_ID <- read_delim(here("Output/Temp/merged_PRS_UKBB.fam"), delim = " ", col_names = F)
  
  CoreExPheno_Euro <- CoreExPheno %>% 
    filter(Geno.BroadAncestry == "European",
           Geno.SampleID %in% All_Euro_ID$X2,
           General.Use != "No",
           !(Pheno.Study %in% c("Auckland Controls", "Australian Controls", "ESR", "Rheumatoid Arthritis")))
  # 1,146 NU, 455 HU controls (1,601 total) + 4,694 gout = either GP or ACR or self-report
  
  # Polynesian cohorts = AGRIA, All Ardea, DM, EuroGout, Aotearoa (both new + old), LPA, Ngati Porou, RD => all split into East and West
  All_CoreEx_ID <- read_delim("/Volumes/archive/merrimanlab/raid_backup/New_Zealand_Chip_data/CoreExome/QC_MergedBatches/Final_Data/CZ-MB1.2-QC1.10_CoreExome24-1.0-3_genotyped-QCd_rsIDconverted.fam", delim = " ", col_names = F)
  
  CoreExPheno_Poly <- CoreExPheno %>% 
    filter(Geno.BroadAncestry == "Oceanian",
           Geno.SampleID %in% All_CoreEx_ID$X2,
           General.Use != "No",
           !(Pheno.Study %in% c("ESR", "Pacific Trust")),
           !is.na(Pheno.GoutSummary))
  # 1,021 NU, 248 HU controls (1,269 total) + 1,380 gout
  
  rm(CoreExPheno, All_CoreEx_ID, All_Euro_ID)
  
  
  # Extracting relevant phenotype information from each cohort
  # Phenotypes of interest are: gout, age at recruitment, sex, age at gout onset, flares in last year, presence of tophi, current allopurinol usage, disease duration, serum urate
  # Can get gout, age, sex, ULT usage, serum urate from CoreExPheno file (reliable as done by Tanya)
  # Also need PCs for all and East/West ancestry (Geno.PolynesianGroup) for Polynesians
  # Also need to determine which (if any) are duplicates based on geneticduplicate column
  # Also need to filter out sex mismatches and ethnicity mismatches (none in Euro)
  # First extracting all Aus/NZ European and Polynesian controls (from Gout in Aotearoa, NPH, RD, DM, AGRIA)
  CoreExPheno_Euro_Control <- CoreExPheno_Euro %>% 
    filter(Pheno.GoutSummary %in% c("Control", "HyperU"),
           Pheno.Study %in% c("Gout in Aotearoa", "Ngati Porou", "Renal Disease", "Diabetes Mellitus", "AGRIA")) %>% 
    select(Pheno.SampleID, Geno.SampleID, Pheno.Study, Pheno.GoutSummary, Pheno.CollectionUrate, Pheno.Gender, Pheno.Age, Geno.PCVector1:Geno.PCVector10)
  
  CoreExPheno_Poly_Control <- CoreExPheno_Poly %>% 
    filter(Pheno.GoutSummary %in% c("Control", "HyperU"),
           Pheno.Study %in% c("Gout in Aotearoa", "Ngati Porou", "Renal Disease", "Diabetes Mellitus", "AGRIA")) %>% 
    select(Pheno.SampleID, Geno.SampleID, Pheno.Study, Pheno.GoutSummary, Pheno.CollectionUrate, Geno.PolynesianGroup, Pheno.Gender, Pheno.Age, Geno.PCVector1:Geno.PCVector10, Geno.PCVector1_Oc:Geno.PCVector10_Oc)
  
  CoreExPheno_EastPoly_Control <- CoreExPheno_Poly_Control %>% 
    filter(Geno.PolynesianGroup == "East Polynesian")
  
  CoreExPheno_WestPoly_Control <- CoreExPheno_Poly_Control %>% 
    filter(Geno.PolynesianGroup == "West Polynesian")
  
  rm(CoreExPheno_Poly_Control)
  
  
  # Now extracting important phenos from CoreExPheno for Euro and Poly gout patients
  CoreExPheno_Euro_Gout <- CoreExPheno_Euro %>% 
    filter(Pheno.GoutSummary == "Gout") %>% 
    select(Pheno.SampleID, Pheno.Study, Pheno.Gender, Pheno.Age, Pheno.GoutSummary:Pheno.WithdrawReason, Geno.SampleID, Geno.GeneticDuplicate:Geno.PCVector10, Geno.BroadAncestry, Notes)
  
  CoreExPheno_Poly_Gout <- CoreExPheno_Poly %>% 
    filter(Pheno.GoutSummary == "Gout") %>% 
    select(Pheno.SampleID, Pheno.Study, Pheno.Gender, Pheno.Age, Pheno.GoutSummary:Pheno.WithdrawReason, Geno.SampleID, Geno.GeneticDuplicate:Geno.PCVector10, Geno.BroadAncestry:Geno.PCVector10_Oc, Geno.PolynesianGroup:Notes)
  
  rm(CoreExPheno_Euro, CoreExPheno_Poly)
  
  # # Now the second CoreEx phenotype file (nothing of use)
  # CoreExPheno2 <- read_delim("Phenotypes/CoreExPheno150618.txt", delim = "\t") %>% 
  #   filter(!(STUDYCOHORT %in% c(5, 8, 9, 11, 13, 14, 15)),
  #          !is.na(STUDYCOHORT))
  # 
  # rm(CoreExPheno2)
  
  
  # So I just need to extract/derive flares, tophi, onset and duration from original SNPmax files, thus only need Gout files
  # Throughout, duration will be derived from onset - age (from CoreExPheno) + 1 (because I am deriving it from ages in years, there is up to 1 additional year duration that needs to be accounted for)
  # Filter cohorts to only include those in the CoreExPheno_Euro or CoreExPheno_Poly files
  # Going through each cohort alphabetically: AGRIA, Ardea - Ironwood (Euro + Poly), Ardea - LASSO (Euro + Poly), DM (Euro + Poly), EuroGout (Euro + Poly), Gout in Aotearoa (Euro + Poly), LPA (Euro + Poly), Ngati Porou (Euro + Poly), RD (Euro + Poly)
  # AGRIA
  ages <- full_join(CoreExPheno_Euro_Gout, CoreExPheno_Poly_Gout) %>% 
    select(Pheno.SampleID, Pheno.Age, Pheno.Study) %>% 
    rename(IID = Pheno.SampleID,
           AGECOL = Pheno.Age)
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "AGRIA")
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/AGRIA.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(1, 2), labels = c("FALSE", "TRUE")))
  }
  
  agriapheno <- read_delim(here("Data/Phenotypes/AGRIAPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT,
           TOPHIGOUT2 = TOPHIGOUT,
           AGE1ATK = AGEGOUTDOX) %>% 
    left_join(ages) %>% 
    mutate(NUMATK = rep(NA, nrow(.)),
           TOPHUS = as.logical(factor(TOPHUS, levels = c(1, 2, 3), labels = c("FALSE", "TRUE", "NA"))),
           GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHIGOUT2 = logicfactor(TOPHIGOUT2),
           TOPHIGOUT = case_when(TOPHUS == TRUE | GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  rm(tmp, tmp2, logicfactor)
  
  
  # Ardea - Ironwood (CLEAR1, CLEAR2, CRYSTAL, LIGHT)
  # European in Ironwood
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: CLEAR1", "Ardea: CLEAR2", "Ardea: CRYSTAL", "Ardea: LIGHT"))
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/Ardea_Ironwood.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(0, 1), labels = c("FALSE", "TRUE")))
  }
  
  ironwood_euro_pheno <- read_delim(here("Data/Phenotypes/ArdeaPheno.txt"), delim = "\t") %>% 
    filter(SUBJID %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGE) %>% 
    rename(IID = SUBJID,
           NUMATK = GFNUM,
           TOPHIGOUT2 = TOPHIGOUT,
           DURATION2 = GFDUR) %>% 
    left_join(ages, by = "IID") %>%
    mutate(CRITBFL = logicfactor(CRITBFL),
           PHNM8FL = logicfactor(PHNM8FL),
           TOPHIFN = logicfactor(TOPHIFN),
           TOPHIGOUT2 = as.logical(factor(TOPHIGOUT2, levels = c("Non-tophaceous gout", "Tophaceous gout"), labels = c("FALSE", "TRUE"))),
           TOPHIGOUT = case_when(CRITBFL == TRUE | PHNM8FL == TRUE | TOPHIFN == TRUE | TOPHIGOUT2 == TRUE | BLTOPHN %in% 1:5 ~ TRUE, is.na(CRITBFL) & is.na(PHNM8FL) & is.na(TOPHIFN) & is.na(BLTOPHN) & is.na(TOPHIGOUT2) ~ NA, TRUE ~ FALSE),
           AGFIDDT = year(as_date(AGFIDDT)),
           AGE1ATK = AGFIDDT - BRTHDTC,
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, STUDYACRONYM)
  
  # Polynesians in Ironwood
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: CLEAR1", "Ardea: CLEAR2", "Ardea: CRYSTAL", "Ardea: LIGHT"))
  
  ironwood_poly_pheno <- read_delim(here("Data/Phenotypes/ArdeaPheno.txt"), delim = "\t") %>% 
    filter(SUBJID %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGE) %>% 
    rename(IID = SUBJID,
           NUMATK = GFNUM,
           TOPHIGOUT2 = TOPHIGOUT,
           DURATION2 = GFDUR) %>% 
    left_join(ages, by = "IID") %>%
    mutate(CRITBFL = logicfactor(CRITBFL),
           PHNM8FL = logicfactor(PHNM8FL),
           TOPHIFN = logicfactor(TOPHIFN),
           TOPHIGOUT2 = as.logical(factor(TOPHIGOUT2, levels = c("Non-tophaceous gout", "Tophaceous gout"), labels = c("FALSE", "TRUE"))),
           TOPHIGOUT = case_when(CRITBFL == TRUE | PHNM8FL == TRUE | TOPHIFN == TRUE | TOPHIGOUT2 == TRUE | BLTOPHN %in% 1:5 ~ TRUE, is.na(CRITBFL) & is.na(PHNM8FL) & is.na(TOPHIFN) & is.na(BLTOPHN) & is.na(TOPHIGOUT2) ~ NA, TRUE ~ FALSE),
           AGFIDDT = year(as_date(AGFIDDT)),
           AGE1ATK = AGFIDDT - BRTHDTC,
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, STUDYACRONYM)
  
  rm(tmp, tmp2, logicfactor)
  
  
  # Ardea - LASSO
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/Ardea_LASSO_Flare.txt"), delim = "\t", col_names = F)
  
  lassopheno1 <- read_delim(here("Data/Phenotypes/ArdeaLassoPhenoFlare.txt"), delim = "\t") %>% 
    select(tmp2$X1)
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/Ardea_LASSO_LabChem.txt"), delim = "\t", col_names = F)
  
  lassopheno2 <- read_delim(here("Data/Phenotypes/ArdeaLassoPhenoLabChem.txt"), delim = "\t") %>% 
    select(tmp2$X1)
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/Ardea_LASSO_Main.txt"), delim = "\t", col_names = F)
  
  lassopheno3 <- read_delim(here("Data/Phenotypes/ArdeaLassoPhenoMain.txt"), delim = "\t") %>% 
    select(tmp2$X1) %>% 
    filter(SUBJID %in% lassopheno1$SUBJID)
  
  tmp <- full_join(lassopheno3, lassopheno2)
  
  lassopheno <- full_join(tmp, lassopheno1)
  
  rm(lassopheno1, lassopheno2, lassopheno3, tmp, tmp2)
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "Ardea: 401")
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(0, 1), labels = c("FALSE", "TRUE")))
  }
  
  lasso_euro_pheno <- lassopheno %>% 
    filter(DNAID %in% tmp$Pheno.SampleID) %>% 
    mutate(IID = as.character(DNAID)) %>% 
    rename(NUMATK = GFNUM,
           DURATION2 = GFDUR) %>% 
    left_join(filter(ages, Pheno.Study == "Ardea: 401"), by = "IID") %>% 
    mutate(TOPHIGOUT = logicfactor(BLTPHFN),
           AGFIDDT = year(AGFIDDT),
           AGE1ATK = AGFIDDT - year(BRTHDTC),
           DURATION = AGECOL - AGE1ATK + 1,
           AGE1ATK = case_when(!is.na(AGE1ATK) ~ AGE1ATK, is.na(AGE1ATK) ~ AGECOL - DURATION2),
           DURATION = case_when(!is.na(DURATION) ~ DURATION, is.na(DURATION) ~ AGECOL - AGE1ATK + 1)) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
    
  
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study == "Ardea: 401")
  
  lasso_poly_pheno <- lassopheno %>% 
    filter(DNAID %in% tmp$Pheno.SampleID) %>% 
    mutate(IID = as.character(DNAID)) %>% 
    rename(NUMATK = GFNUM,
           DURATION2 = GFDUR) %>% 
    left_join(filter(ages, Pheno.Study == "Ardea: 401"), by = "IID") %>% 
    mutate(TOPHIGOUT = logicfactor(BLTPHFN),
           AGFIDDT = year(AGFIDDT),
           AGE1ATK = AGFIDDT - year(BRTHDTC),
           DURATION = AGECOL - AGE1ATK + 1,
           AGE1ATK = case_when(!is.na(AGE1ATK) ~ AGE1ATK, is.na(AGE1ATK) ~ AGECOL - DURATION2),
           DURATION = case_when(!is.na(DURATION) ~ DURATION, is.na(DURATION) ~ AGECOL - AGE1ATK + 1)) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  rm(lassopheno, tmp, logicfactor)
    
  
  # Ardea - Other (232 other Ardea study participants (594 and 3170)) - asked ruth if they have separate pheno files - she said we aren't allowed to use them - Tony suggested I just say Tony said why not use them
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(str_detect(Pheno.Study, "Ardea: 3170") | str_detect(Pheno.Study, "Ardea: 594"))
  
  other_ardea_euro_pheno <- read_delim(here("Data/Phenotypes/ArdeaPheno.txt"), delim = "\t") %>% 
    filter(SUBJID %in% tmp$Pheno.SampleID)
  
  other_ardea_euro_pheno2 <- read_delim(here("Data/Phenotypes/ArdeaLassoPhenoMain.txt"), delim = "\t") %>% 
    filter(DNAID %in% tmp$Pheno.SampleID | SUBJID %in% tmp$Pheno.SampleID)
  
  rm(tmp, other_ardea_euro_pheno, other_ardea_euro_pheno2)
  
  
  # DM
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "Diabetes Mellitus")
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/DM.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(1, 2), labels = c("FALSE", "TRUE")))
  }
  
  dm_euro_pheno <- read_delim(here("Data/Phenotypes/DMPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT = case_when(TOPHUS == TRUE | GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study == "Diabetes Mellitus")
  
  dm_poly_pheno <- read_delim(here("Data/Phenotypes/DMPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT = case_when(TOPHUS == TRUE | GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  rm(tmp, tmp2, logicfactor)
  
  
  # EuroGout
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "EuroGout")
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/EuroGout.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(0, 1), labels = c("FALSE", "TRUE")))
  }
  
  eurogout_euro_pheno <- read_delim(here("Data/Phenotypes/EuroGoutPheno.txt"), delim = "\t") %>% 
    filter(SUBJECT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1) %>% 
    rename(IID = SUBJECT,
           AGE1ATK = AGEFIRSTATTK,
           NUMATK = NUMATTACKS,
           DURATION2 = DURATIONGOUT) %>% 
    left_join(filter(ages, Pheno.Study == "EuroGout"), by = "IID") %>% 
    mutate(TOPHUS = logicfactor(TOPHUS),
           NUMTOPHI = as.factor(NUMTOPHI),
           ACRB = logicfactor(ACRB),
           ACRC8 = logicfactor(ACRC8),
           TOPHIGOUT = case_when(TOPHUS == TRUE | NUMTOPHI %in% c(1, 2, 3) | ACRB == TRUE | ACRC8 == TRUE ~ TRUE, is.na(TOPHUS) & is.na(NUMTOPHI) & is.na(ACRB) & is.na(ACRC8) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1,
           DURATION = case_when(!is.na(DURATION) ~ DURATION, is.na(DURATION) ~ DURATION2),
           AGE1ATK = case_when(!is.na(AGE1ATK) ~ AGE1ATK, is.na(AGE1ATK) ~ AGECOL - DURATION - 1)) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study == "EuroGout")
  
  eurogout_poly_pheno <- read_delim(here("Data/Phenotypes/EuroGoutPheno.txt"), delim = "\t") %>% 
    filter(SUBJECT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1) %>% 
    rename(IID = SUBJECT,
           AGE1ATK = AGEFIRSTATTK,
           NUMATK = NUMATTACKS,
           DURATION2 = DURATIONGOUT) %>% 
    left_join(filter(ages, Pheno.Study == "EuroGout"), by = "IID") %>% 
    mutate(TOPHUS = logicfactor(TOPHUS),
           NUMTOPHI = as.factor(NUMTOPHI),
           ACRB = logicfactor(ACRB),
           ACRC8 = logicfactor(ACRC8),
           TOPHIGOUT = case_when(TOPHUS == TRUE | NUMTOPHI %in% c(1, 2, 3) | ACRB == TRUE | ACRC8 == TRUE ~ TRUE, is.na(TOPHUS) & is.na(NUMTOPHI) & is.na(ACRB) & is.na(ACRC8) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1,
           DURATION = case_when(!is.na(DURATION) ~ DURATION, is.na(DURATION) ~ DURATION2),
           AGE1ATK = case_when(!is.na(AGE1ATK) ~ AGE1ATK, is.na(AGE1ATK) ~ AGECOL - DURATION - 1)) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION) # All 4 accounted for
  
  rm(tmp, tmp2, logicfactor)
  
  
  # Gout in Aotearoa
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "Gout in Aotearoa")
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/GoutinAotearoa.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(1, 2), labels = c("FALSE", "TRUE")))
  }
  
  aotearoa_euro_pheno <- read_delim(here("Data/Phenotypes/NZPheno.txt"), delim = "\t") %>% 
    filter(SUBJECT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = SUBJECT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT2 = as.logical(factor(TOPHIGOUT, levels = c(0, 1, 2), labels = c("NA", "FALSE", "TRUE"))),
           TOPHIGOUT = case_when(GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE | TOPHUS == TRUE | TOPHIGOUT2 == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) & is.na(TOPHIGOUT2) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study == "Gout in Aotearoa")
  
  aotearoa_poly_pheno <- read_delim(here("Data/Phenotypes/NZPheno.txt"), delim = "\t") %>% 
    filter(SUBJECT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = SUBJECT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT2 = as.logical(factor(TOPHIGOUT, levels = c(0, 1, 2), labels = c("NA", "FALSE", "TRUE"))),
           TOPHIGOUT = case_when(GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE | TOPHUS == TRUE | TOPHIGOUT2 == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) & is.na(TOPHIGOUT2) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  rm(tmp, tmp2, logicfactor)
  
  
  # LPA (don't have any phenotypes of interest for this study)
  lpa_euro_pheno <- read_delim(here("Data/Phenotypes/LPAPheno.txt"), delim = "\t") %>% 
    filter(SUBJECT %in% CoreExPheno_Euro_Gout$Pheno.SampleID) # All 349 accounted for
  
  lpa_poly_pheno <- read_delim(here("Data/Phenotypes/LPAPheno.txt"), delim = "\t") %>% 
    filter(SUBJECT %in% CoreExPheno_Poly_Gout$Pheno.SampleID) # All 15 accounted for
  
  rm(lpa_euro_pheno, lpa_poly_pheno)
  
  
  # Ngati Porou
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "Ngati Porou")
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/NPH.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(1, 2), labels = c("FALSE", "TRUE")))
  }
  
  nph_euro_pheno <- read_delim(here("Data/Phenotypes/NPHPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT2 = as.logical(factor(TOPHIGOUT, levels = c(1, 2, 3), labels = c("NA", "FALSE", "TRUE"))),
           TOPHIGOUT = case_when(TOPHUS == TRUE | SUSTOPHUS == TRUE | GOUTCRITERIAB == TRUE | TOPHIGOUT2 == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) & is.na(TOPHIGOUT2) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study == "Ngati Porou")
  
  nph_poly_pheno <- read_delim(here("Data/Phenotypes/NPHPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT2 = as.logical(factor(TOPHIGOUT, levels = c(1, 2, 3), labels = c("NA", "FALSE", "TRUE"))),
           TOPHIGOUT = case_when(TOPHUS == TRUE | SUSTOPHUS == TRUE | GOUTCRITERIAB == TRUE | TOPHIGOUT2 == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) & is.na(TOPHIGOUT2) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  rm(tmp2, logicfactor)
  
  
  # RD
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study == "Renal Disease")
  
  tmp2 <- read_delim(here("Data/Phenotypes/For_selecting_vars/RD.txt"), delim = "\t", col_names = F)
  
  logicfactor <- function(x) {
    as.logical(factor(x, levels = c(1, 2), labels = c("FALSE", "TRUE")))
  }
  
  rd_euro_pheno <- read_delim(here("Data/Phenotypes/RDPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(NUMATK = rep(NA, nrow(.)),
           GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT = case_when(GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE | TOPHUS == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  tmp <- CoreExPheno_Poly_Gout %>% 
    filter(Pheno.Study == "Renal Disease")
  
  rd_poly_pheno <- read_delim(here("Data/Phenotypes/RDPheno.txt"), delim = "\t") %>% 
    filter(PATIENT %in% tmp$Pheno.SampleID) %>% 
    select(tmp2$X1, -AGECOL) %>% 
    rename(IID = PATIENT) %>% 
    left_join(ages, by = "IID") %>% 
    mutate(NUMATK = rep(NA, nrow(.)),
           GOUTCRITERIAB = logicfactor(GOUTCRITERIAB),
           SUSTOPHUS = logicfactor(SUSTOPHUS),
           TOPHUS = logicfactor(TOPHUS),
           TOPHIGOUT = case_when(GOUTCRITERIAB == TRUE | SUSTOPHUS == TRUE | TOPHUS == TRUE ~ TRUE, is.na(TOPHUS) & is.na(GOUTCRITERIAB) & is.na(SUSTOPHUS) ~ NA, TRUE ~ FALSE),
           DURATION = AGECOL - AGE1ATK + 1) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION)
  
  rm(tmp, tmp2, logicfactor, ages)
  
  
  # Combining AGRIA, Aotearoa, DM, NPH and RD into ANZ
  tmp <- agriapheno %>% 
    mutate(COHORT = rep("AGRIA", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp2 <- aotearoa_euro_pheno %>% 
    mutate(COHORT = rep("Gout in Aotearoa", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp3 <- dm_euro_pheno %>% 
    mutate(COHORT = rep("Diabetes Mellitus", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
    
  tmp4 <- nph_euro_pheno %>% 
    mutate(COHORT = rep("Ngati Porou Hauora", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp5 <- rd_euro_pheno %>% 
    mutate(COHORT = rep("Renal Disease", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  anz_euro_pheno <- rbind(tmp, tmp2, tmp3, tmp4, tmp5) %>% 
    mutate(COHORT = as.factor(COHORT)) %>% 
    arrange(IID)
  
  rm(aotearoa_euro_pheno, agriapheno, dm_euro_pheno, nph_euro_pheno, rd_euro_pheno, tmp, tmp2, tmp3, tmp4, tmp5)
  
  
  # Combining all Polynesian cohorts together
  tmp <- aotearoa_poly_pheno %>% 
    mutate(COHORT = rep("Gout in Aotearoa", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp2 <- dm_poly_pheno %>% 
    mutate(COHORT = rep("Diabetes Mellitus", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp3 <- eurogout_poly_pheno %>% 
    mutate(COHORT = rep("EuroGout", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp4 <- ironwood_poly_pheno %>% 
    mutate(COHORT = paste0("Ardea - ", STUDYACRONYM)) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp5 <- lasso_poly_pheno %>% 
    mutate(COHORT = rep("Ardea - LASSO", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
    
  tmp6 <- nph_poly_pheno %>% 
    mutate(COHORT = rep("Ngati Porou Hauora", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  tmp7 <- rd_poly_pheno %>% 
    mutate(COHORT = rep("Renal Disease", nrow(.))) %>% 
    select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT)
  
  poly_pheno <- rbind(tmp, tmp2, tmp3, tmp4, tmp5, tmp6, tmp7) %>% 
    mutate(COHORT = as.factor(COHORT)) %>% 
    arrange(IID) %>% 
    unique()
  
  rm(aotearoa_poly_pheno, dm_poly_pheno, eurogout_poly_pheno, ironwood_poly_pheno, lasso_poly_pheno, nph_poly_pheno, rd_poly_pheno, tmp, tmp2, tmp3, tmp4, tmp5, tmp6, tmp7)
  
  
  # Now I want to merge the rest of the phenotypes from the CoreExPheno files into the final-ish phenotype files
  anz_euro_pheno <- anz_euro_pheno %>% 
    left_join(CoreExPheno_Euro_Gout, by = c("IID" = "Pheno.SampleID")) %>% 
    rename(AGECOL = Pheno.Age,
           PC1 = Geno.PCVector1,
           PC2 = Geno.PCVector2,
           PC3 = Geno.PCVector3,
           PC4 = Geno.PCVector4,
           PC5 = Geno.PCVector5,
           PC6 = Geno.PCVector6,
           PC7 = Geno.PCVector7,
           PC8 = Geno.PCVector8,
           PC9 = Geno.PCVector9,
           PC10 = Geno.PCVector10) %>% 
    mutate(SEX = as.factor(Pheno.Gender),
           GOUT = rep(TRUE, nrow(.)),
           SURICACID = Pheno.CollectionUrate / 0.05949,
           ULT = as.logical(factor(Pheno.UrateTherapy, levels = c("No", "Yes"), labels = c("FALSE", "TRUE")))) %>% 
    select(IID, Geno.SampleID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10) %>% 
    filter(!(is.na(NUMATK) & is.na(TOPHIGOUT) & is.na(AGE1ATK)))
  
  eurogout_euro_pheno <- eurogout_euro_pheno %>% 
    left_join(CoreExPheno_Euro_Gout, by = c("IID" = "Pheno.SampleID")) %>% 
    filter(Geno.GeneticSex == Pheno.Gender, 
           !(is.na(NUMATK) & is.na(TOPHIGOUT) & is.na(AGE1ATK))) %>% 
    rename(AGECOL = Pheno.Age,
           PC1 = Geno.PCVector1,
           PC2 = Geno.PCVector2,
           PC3 = Geno.PCVector3,
           PC4 = Geno.PCVector4,
           PC5 = Geno.PCVector5,
           PC6 = Geno.PCVector6,
           PC7 = Geno.PCVector7,
           PC8 = Geno.PCVector8,
           PC9 = Geno.PCVector9,
           PC10 = Geno.PCVector10) %>% 
    mutate(SEX = as.factor(Pheno.Gender),
           GOUT = rep(TRUE, nrow(.)),
           SURICACID = Pheno.CollectionUrate / 0.05949,
           ULT = as.logical(factor(Pheno.UrateTherapy, levels = c("No", "Yes"), labels = c("FALSE", "TRUE"))),
           COHORT = rep("EuroGout", nrow(.))) %>% 
    select(IID, Geno.SampleID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10)
  
  ironwood_euro_pheno <- ironwood_euro_pheno %>% 
    mutate(COHORT = paste0("Ardea: ", STUDYACRONYM)) %>% 
    left_join(CoreExPheno_Euro_Gout, by = c("IID" = "Pheno.SampleID", "COHORT" = "Pheno.Study")) %>% 
    unique() %>% 
    filter(Geno.GeneticSex == Pheno.Gender, 
           !(is.na(NUMATK) & is.na(TOPHIGOUT) & is.na(AGE1ATK))) %>% 
    rename(AGECOL = Pheno.Age,
           PC1 = Geno.PCVector1,
           PC2 = Geno.PCVector2,
           PC3 = Geno.PCVector3,
           PC4 = Geno.PCVector4,
           PC5 = Geno.PCVector5,
           PC6 = Geno.PCVector6,
           PC7 = Geno.PCVector7,
           PC8 = Geno.PCVector8,
           PC9 = Geno.PCVector9,
           PC10 = Geno.PCVector10) %>% 
    mutate(SEX = as.factor(Pheno.Gender),
           GOUT = rep(TRUE, nrow(.)),
           SURICACID = Pheno.CollectionUrate / 0.05949,
           ULT = as.logical(factor(Pheno.UrateTherapy, levels = c("No", "Yes"), labels = c("FALSE", "TRUE"))),
           ULT = case_when(COHORT %in% c("Ardea: CLEAR1", "Ardea: CLEAR2") | (COHORT == "Ardea: CRYSTAL" & SURICACID < 8) ~ TRUE, COHORT == "Ardea: LIGHT" ~ NA, TRUE ~ ULT)) %>% 
    select(IID, Geno.SampleID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10)
  
  lasso_euro_pheno <- lasso_euro_pheno %>% 
    mutate(COHORT = rep("Ardea: 401", nrow(.))) %>% 
    left_join(CoreExPheno_Euro_Gout, by = c("IID" = "Pheno.SampleID", "COHORT" = "Pheno.Study")) %>% 
    filter(Geno.GeneticSex == Pheno.Gender, 
           !(is.na(NUMATK) & is.na(TOPHIGOUT) & is.na(AGE1ATK))) %>% 
    rename(AGECOL = Pheno.Age,
           PC1 = Geno.PCVector1,
           PC2 = Geno.PCVector2,
           PC3 = Geno.PCVector3,
           PC4 = Geno.PCVector4,
           PC5 = Geno.PCVector5,
           PC6 = Geno.PCVector6,
           PC7 = Geno.PCVector7,
           PC8 = Geno.PCVector8,
           PC9 = Geno.PCVector9,
           PC10 = Geno.PCVector10) %>% 
    mutate(SEX = as.factor(Pheno.Gender),
           GOUT = rep(TRUE, nrow(.)),
           SURICACID = Pheno.CollectionUrate / 0.05949,
           ULT = as.logical(factor(Pheno.UrateTherapy, levels = c("No", "Yes"), labels = c("FALSE", "TRUE"))),
           ULT = case_when(SURICACID < 8 ~ TRUE, TRUE ~ ULT)) %>% 
    select(IID, Geno.SampleID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10)
  
  
  # Now figuring out which Ardea cohorts are similar based on recruitment
  # CLEAR1 = Gout + inadequate response to Allopurinol (persistent HU after treatment) + at least 2 flares in prior year - all US
  # CLEAR2 = Exactly the same as CLEAR1 - half US, rest international
  # CRYSTAL = All gout recruited + > 8mg/dL off ULT or > 6 mg/dL on ULT + all had tophi - Most US, rest international
  # LIGHT = Gout + History of intolerance/contraindication to allop or febux + > 6.5 mg/dL urate - Most US, rest international
  # LASSO = Gout + at least 2 flares in prior year + > 8mg/dL off ULT or > 6.5 mg/dL on ULT - Most US, rest international
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: CLEAR1")) # All 242 United States
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: CLEAR2")) # 3 Aus, 4 Belgium, 17 Germany, 6 NZ, 37 South Africa, 1 Swiss, 60 Ukraine, 113 US
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: CRYSTAL")) # 14 Aus, 12 Canada, 4 NZ, 21 Poland, 2 Swiss, 122 US
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: LIGHT")) # 2 Aus, 9 Belgium, 7 Canada, 5 German, 2 NZ, 13 South Africa, 75 US
  
  tmp <- CoreExPheno_Euro_Gout %>% 
    filter(Pheno.Study %in% c("Ardea: 401")) # 63 Aus, 10 Belgium, 34 Canada, 5 German, 27 NZ, 119 South Africa, 593 US
  
  # So in summary, CLEAR1 and CLEAR2 are nearly identical except for some internationals in CLEAR2, LASSO same except also included HU gout patients off ULT, CRYSTAL same as LASSO except all had tophi instead of >2 flares, LIGHT is kind of different from rest
  
  # I could extract all 243 LASSO patients on ULT at screening and include them along with CLEAR1 and CLEAR2 for an identically selected cohort of 750 patients (all inadequate response to ULT + > 1 flare in last year)
  
  # The remaining 600 LASSO could stay as their own cohort (none on ULT, all HU and flares > 1 in last year)
  
  # Then CRYSTAL (N = 175) all have tophi, and either are HU or don't respond to ULT
  
  # LIGHT (N = 113) are all HU and can't take ALLO or FEBUX due to contraindication (might be able to combine with LASSO - no ULT)
  
  tmp <- ironwood_euro_pheno %>% 
    filter(COHORT %in% c("Ardea: CLEAR1", "Ardea: CLEAR2"))
  
  clear_lasso_euro_pheno <- lasso_euro_pheno %>% 
    filter(ULT == TRUE) %>% 
    rbind(tmp) %>% 
    arrange(IID)
  
  lasso_other_euro_pheno <- lasso_euro_pheno %>% 
    filter(is.na(ULT))
  
  rm(lasso_euro_pheno)
  
  crystal_euro_pheno <- ironwood_euro_pheno %>% 
    filter(COHORT == "Ardea: CRYSTAL")
  
  light_euro_pheno <- ironwood_euro_pheno %>% 
    filter(COHORT == "Ardea: LIGHT")
  
  rm(ironwood_euro_pheno, tmp, tmp2)
  
  # Now fixing up the polynesian gout phenotype files
  poly_pheno <- poly_pheno %>% 
    left_join(CoreExPheno_Poly_Gout, by = c("IID" = "Pheno.SampleID", "COHORT" = "Pheno.Study")) %>% 
    filter(Geno.GeneticSex == Pheno.Gender,
           Geno.AncestryMatch != "Self-reported ethnicity does not match genetic ancestry", 
           !(is.na(NUMATK) & is.na(TOPHIGOUT) & is.na(AGE1ATK))) %>% 
    rename(AGECOL = Pheno.Age,
           PC1 = Geno.PCVector1,
           PC2 = Geno.PCVector2,
           PC3 = Geno.PCVector3,
           PC4 = Geno.PCVector4,
           PC5 = Geno.PCVector5,
           PC6 = Geno.PCVector6,
           PC7 = Geno.PCVector7,
           PC8 = Geno.PCVector8,
           PC9 = Geno.PCVector9,
           PC10 = Geno.PCVector10,
           PC1_Oc = Geno.PCVector1_Oc,
           PC2_Oc = Geno.PCVector2_Oc,
           PC3_Oc = Geno.PCVector3_Oc,
           PC4_Oc = Geno.PCVector4_Oc,
           PC5_Oc = Geno.PCVector5_Oc,
           PC6_Oc = Geno.PCVector6_Oc,
           PC7_Oc = Geno.PCVector7_Oc,
           PC8_Oc = Geno.PCVector8_Oc,
           PC9_Oc = Geno.PCVector9_Oc,
           PC10_Oc = Geno.PCVector10_Oc) %>% 
    mutate(SEX = as.factor(Pheno.Gender),
           GOUT = rep(TRUE, nrow(.)),
           SURICACID = Pheno.CollectionUrate / 0.05949,
           ULT = as.logical(factor(Pheno.UrateTherapy, levels = c("No", "Yes"), labels = c("FALSE", "TRUE")))) %>% 
    select(IID, Geno.SampleID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10, PC1_Oc:PC10_Oc, Geno.PolynesianGroup)
  
  eastpoly_pheno <- poly_pheno %>% 
    filter(Geno.PolynesianGroup == "East Polynesian")
  
  westpoly_pheno <- poly_pheno %>% 
    filter(Geno.PolynesianGroup == "West Polynesian")
  
  rm(poly_pheno)
  
  # Finally cleaning up the control files
  control_anz_euro_pheno <- CoreExPheno_Euro_Control
  
  control_eastpoly_pheno <- CoreExPheno_EastPoly_Control
  
  control_westpoly_pheno <- CoreExPheno_WestPoly_Control
  
  rm(CoreExPheno_EastPoly_Control, CoreExPheno_Euro_Control, CoreExPheno_Euro_Gout, CoreExPheno_Poly_Gout, CoreExPheno_WestPoly_Control)
  
  save(anz_euro_pheno, clear_lasso_euro_pheno, control_anz_euro_pheno, control_eastpoly_pheno, control_westpoly_pheno, crystal_euro_pheno, eastpoly_pheno, eurogout_euro_pheno, lasso_other_euro_pheno, light_euro_pheno, westpoly_pheno, file = here("Output/Phenotypes.RData"))
}
```

The PRS was then calculated for all European samples. Using the results of the UKBB GWAS, 55 PRS metrics were generated, this includes the complete PRS, the 27 PRS's with a single variant excluded and the 27 variants individually.

```{r Euro_PRS, message = FALSE, warning = FALSE, cache = TRUE, dependson = "dataprep"}
make_prs_euro <- function(ids) {
  # Load in plink genotype files + rename column names + filter to only include IDs of cohort of interest
  map <- read_delim(here("Output/Temp/merged_PRS_UKBB.map"), delim = "\t", col_names = FALSE)
  x <- read_delim(here("Output/Temp/merged_PRS_UKBB.ped"), delim = " ", col_names = FALSE, col_types = cols(.default = col_character()))
  colnames(x)[seq(from = 7, to = ncol(x) - 1, by = 2)] <- str_c(map$X2, "_1")
  colnames(x)[seq(from = 8, to = (ncol(x)), by = 2)] <- str_c(map$X2, "_2")
  colnames(x)[1:6] <- c("FID", "IID", "PID", "MID", "SEX", "AFF")
  x <- x %>% filter(IID %in% ids)
  
  # Convert character genotypes into numeric genotypes based on risk allele = 1
  num_cols <- ncol(x)
  for (i in 1:nrow(UKBB_Gene_OR)) {
    x[[2 * i + 5]] <- x[[2 * i + 5]] %>% 
      str_replace("0", "NA") %>% 
      str_replace(UKBB_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
      str_replace(UKBB_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
      as.numeric()
    x[[2 * i + 6]] <- x[[2 * i + 6]] %>% 
      str_replace("0", "NA") %>% 
      str_replace(UKBB_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
      str_replace(UKBB_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
      as.numeric()
    x <- x %>% 
      mutate("TEMP" = (x[[2 * i + 5]] + x[[2 * i + 6]]))
    colnames(x) <- c(colnames(x[1:((num_cols - 1) + i)]), UKBB_Gene_OR[[i, "RSID"]])
  }
  x <- x %>% 
    select(2, (num_cols + 1):ncol(x))
  
  # Save this dataframe for individual SNP analysis
  x1 <- x
  
  # Now to calculate the PRS
  for(i in 1:nrow(UKBB_Gene_OR)) {
    x[i + 1] <- x[[i + 1]] * UKBB_Gene_OR[[i, "OR"]]
  }
  x$PRS <- rowSums(x[2:(ncol(x))])
  
  # Calculating leave-one-out PRS's
  num_cols <- ncol(x)
  for(i in 1:(ncol(x) - 2)) {
    x[i + 1] <- x[[num_cols]] - x[[i + 1]]
  }
  x <- x %>% 
    select(IID, PRS, 2:(ncol(x) - 1))
  colnames(x)[3:ncol(x)] <- str_c(paste0(colnames(x[2]), "_Less_"), colnames(x[3:ncol(x)]))
  
  # Joining them back together
  x <- x %>% left_join(x1)
  
  assign("prs", x, envir = .GlobalEnv)
}



# Australian/NZ cases
ids <- anz_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

anz_euro_pheno_prs <- anz_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(anz_euro_pheno_prs, file = here("Output/anz_euro_pheno_prs.RData"))

rm(ids, prs)

# Australian/NZ controls
ids <- control_anz_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

control_anz_euro_pheno_prs <- control_anz_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(control_anz_euro_pheno_prs, file = here("Output/control_anz_euro_pheno_prs.RData"))

rm(ids, prs)

# Ardea - CLEAR1, CLEAR2 and LASSO (ULT)
ids <- clear_lasso_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

clear_lasso_euro_pheno_prs <- clear_lasso_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(clear_lasso_euro_pheno_prs, file = here("Output/clear_lasso_euro_pheno_prs.RData"))

rm(ids, prs)

# Ardea - CRYSTAL
ids <- crystal_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

crystal_euro_pheno_prs <- crystal_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(crystal_euro_pheno_prs, file = here("Output/crystal_euro_pheno_prs.RData"))

rm(ids, prs)

# EuroGout
ids <- eurogout_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

eurogout_euro_pheno_prs <- eurogout_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(eurogout_euro_pheno_prs, file = here("Output/eurogout_euro_pheno_prs.RData"))

rm(ids, prs)

# Ardea - LASSO (all others)
ids <- lasso_other_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

lasso_other_euro_pheno_prs <- lasso_other_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(lasso_other_euro_pheno_prs, file = here("Output/lasso_other_euro_pheno_prs.RData"))

rm(ids, prs)

# Ardea - LIGHT
ids <- light_euro_pheno$Geno.SampleID

make_prs_euro(ids = ids)

light_euro_pheno_prs <- light_euro_pheno %>% 
    left_join(prs, by = c("Geno.SampleID" = "IID"))

save(light_euro_pheno_prs, file = here("Output/light_euro_pheno_prs.RData"))

rm(ids, prs, make_prs_euro)

# UK Biobank
load("/Volumes/userdata/student_users/nicksumpter/Documents/PhD/Cluster/self_report_med.RData")
tmp <- self_report_med %>% 
  mutate(IID = eid,
         ULT = case_when(allopurinol == TRUE | sulphinpyrazone == TRUE | probenecid == TRUE ~ TRUE, TRUE ~ FALSE)) %>% 
  select(IID, ULT)

load("/Volumes/userdata/student_users/nicksumpter/Documents/PhD/Cluster/final_data_prs.RData")
ukbb_euro_pheno <- final_data_prs %>% 
  select(eid, gout, age:urate) %>% 
  rename(IID = eid,
         GOUT = gout,
         AGECOL = age,
         SEX = sex,
         SURICACID = urate) %>% 
  mutate(COHORT = "UK Biobank") %>% 
  left_join(tmp) %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT)

ids <- ukbb_euro_pheno$IID

# Load in plink genotype files + rename column names + filter to only include IDs of cohort of interest
map <- read_delim(here("Output/Temp/merged_PRS.map"), delim = "\t", col_names = FALSE)
map <- map %>% 
  separate(X2, into = c("X2", "remove"), sep = ",") %>% 
  select(-remove)
x <- read_delim(here("Output/Temp/merged_PRS.ped"), delim = " ", col_names = FALSE, col_types = cols(.default = col_character()))
colnames(x)[seq(from = 7, to = ncol(x) - 1, by = 2)] <- str_c(map$X2, "_1")
colnames(x)[seq(from = 8, to = (ncol(x)), by = 2)] <- str_c(map$X2, "_2")
colnames(x)[1:6] <- c("FID", "IID", "PID", "MID", "SEX", "AFF")
tmp <- c("FID", "IID", "PID", "MID", "SEX", "AFF", paste0(UKBB_Gene_OR$RSID, "_1"), paste0(UKBB_Gene_OR$RSID, "_2"))
x <- x[, colnames(x) %in% tmp]
x <- x %>% mutate(IID = as.numeric(IID)) %>% filter(IID %in% ids)

# Convert character genotypes into numeric genotypes based on risk allele = 1
num_cols <- ncol(x)
for (i in 1:nrow(UKBB_Gene_OR)) {
  x[[2 * i + 5]] <- x[[2 * i + 5]] %>% 
      str_replace("0", "NA") %>% 
    str_replace(UKBB_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
    str_replace(UKBB_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
    as.numeric()
  x[[2 * i + 6]] <- x[[2 * i + 6]] %>% 
      str_replace("0", "NA") %>% 
    str_replace(UKBB_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
    str_replace(UKBB_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
    as.numeric()
  x <- x %>% 
    mutate("TEMP" = (x[[2 * i + 5]] + x[[2 * i + 6]]))
  colnames(x) <- c(colnames(x[1:((num_cols - 1) + i)]), UKBB_Gene_OR[[i, "RSID"]])
}
x <- x %>% 
  select(2, (num_cols + 1):ncol(x))

# Save this dataframe for individual SNP analysis
x1 <- x

# Now to calculate the PRS
for(i in 1:nrow(UKBB_Gene_OR)) {
  x[i + 1] <- x[[i + 1]] * UKBB_Gene_OR[[i, "OR"]]
}
x$PRS <- rowSums(x[2:(ncol(x))])

# Calculating leave-one-out PRS's
num_cols <- ncol(x)
for(i in 1:(ncol(x) - 2)) {
  x[i + 1] <- x[[num_cols]] - x[[i + 1]]
}
x <- x %>% 
  select(IID, PRS, 2:(ncol(x) - 1))
colnames(x)[3:ncol(x)] <- str_c(paste0(colnames(x[2]), "_Less_"), colnames(x[3:ncol(x)]))

# Joining them back together
x <- x %>% left_join(x1)

ukbb_euro_pheno_prs <- ukbb_euro_pheno %>% 
    left_join(x, by = c("IID"))

save(ukbb_euro_pheno_prs, file = here("Output/ukbb_euro_pheno_prs.RData"))

rm(ids, prs, ukbb_euro_pheno, final_data_prs, map, x, x1, i, ids, num_cols, tmp, self_report_med)
```

The PRS was then calculated for all Polynesian samples. Using the results of the UKBB GWAS (only using genotyped variants), 37 PRS metrics were generated, this includes the complete PRS, the 18 PRS's with a single variant excluded and the 18 variants individually.

```{r Poly_PRS, message = FALSE, warning = FALSE, cache = TRUE, dependson = "dataprep"}
make_prs_poly <- function(ids) {
  # Load in plink genotype files + rename column names + filter to only include IDs of cohort of interest
  map <- read_delim(here("Output/Temp/Poly_SNPs.map"), delim = "\t", col_names = FALSE)
  x <- read_delim(here("Output/Temp/Poly_SNPs.ped"), delim = " ", col_names = FALSE, col_types = cols(.default = col_character()))
  colnames(x)[seq(from = 7, to = ncol(x) - 1, by = 2)] <- str_c(map$X2, "_1")
  colnames(x)[seq(from = 8, to = (ncol(x)), by = 2)] <- str_c(map$X2, "_2")
  colnames(x)[1:6] <- c("FID", "IID", "PID", "MID", "SEX", "AFF")
  x <- x %>% filter(IID %in% ids)
  
  # Convert character genotypes into numeric genotypes based on risk allele = 1
  num_cols <- ncol(x)
  for (i in 1:nrow(Poly_Gene_OR)) {
    x[[2 * i + 5]] <- x[[2 * i + 5]] %>% 
      str_replace("0", "NA") %>% 
      str_replace(Poly_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
      str_replace(Poly_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
      as.numeric()
    x[[2 * i + 6]] <- x[[2 * i + 6]] %>% 
      str_replace("0", "NA") %>% 
      str_replace(Poly_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
      str_replace(Poly_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
      as.numeric()
    x <- x %>% 
      mutate("TEMP" = (x[[2 * i + 5]] + x[[2 * i + 6]]))
    colnames(x) <- c(colnames(x[1:((num_cols - 1) + i)]), Poly_Gene_OR[[i, "RSID"]])
  }
  x <- x %>% 
    select(2, (num_cols + 1):ncol(x))
  
  # Save this dataframe for individual SNP analysis
  x1 <- x
  
  # Now to calculate the PRS
  for(i in 1:nrow(Poly_Gene_OR)) {
    x[i + 1] <- x[[i + 1]] * Poly_Gene_OR[[i, "OR"]]
  }
  x$PRS <- rowSums(x[2:(ncol(x))])
  
  # Calculating leave-one-out PRS's
  num_cols <- ncol(x)
  for(i in 1:(ncol(x) - 2)) {
    x[i + 1] <- x[[num_cols]] - x[[i + 1]]
  }
  x <- x %>% 
    select(IID, PRS, 2:(ncol(x) - 1))
  colnames(x)[3:ncol(x)] <- str_c(paste0(colnames(x[2]), "_Less_"), colnames(x[3:ncol(x)]))
  
  # Joining them back together
  x <- x %>% left_join(x1)
  
  assign("prs", x, envir = .GlobalEnv)
}

# East Polynesian cases
ids <- eastpoly_pheno$Geno.SampleID

make_prs_poly(ids = ids)

eastpoly_pheno_prs <- eastpoly_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(eastpoly_pheno_prs, file = here("Output/eastpoly_pheno_prs.RData"))

rm(ids, prs, eastpoly_pheno)

# East Polynesian controls
ids <- control_eastpoly_pheno$Geno.SampleID

make_prs_poly(ids = ids)

control_eastpoly_pheno_prs <- control_eastpoly_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(control_eastpoly_pheno_prs, file = here("Output/control_eastpoly_pheno_prs.RData"))

rm(ids, prs, control_eastpoly_pheno)

# West Polynesian cases
ids <- westpoly_pheno$Geno.SampleID

make_prs_poly(ids = ids)

westpoly_pheno_prs <- westpoly_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(westpoly_pheno_prs, file = here("Output/westpoly_pheno_prs.RData"))

rm(ids, prs, westpoly_pheno)

# West Polynesian controls
ids <- control_westpoly_pheno$Geno.SampleID

make_prs_poly(ids = ids)

control_westpoly_pheno_prs <- control_westpoly_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(control_westpoly_pheno_prs, file = here("Output/control_westpoly_pheno_prs.RData"))

rm(ids, prs, control_westpoly_pheno)

# Australian/NZ cases
ids <- anz_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

anz_euro_pheno_prs_poly <- anz_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(anz_euro_pheno_prs_poly, file = here("Output/anz_euro_pheno_prs_poly.RData"))

rm(ids, prs, anz_euro_pheno)

# Australian/NZ controls
ids <- control_anz_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

control_anz_euro_pheno_prs_poly <- control_anz_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(control_anz_euro_pheno_prs_poly, file = here("Output/control_anz_euro_pheno_prs_poly.RData"))

rm(ids, prs, control_anz_euro_pheno)

# Ardea - CLEAR1, CLEAR2 and LASSO (ULT)
ids <- clear_lasso_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

clear_lasso_euro_pheno_prs_poly <- clear_lasso_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(clear_lasso_euro_pheno_prs_poly, file = here("Output/clear_lasso_euro_pheno_prs_poly.RData"))

rm(ids, prs, clear_lasso_euro_pheno)

# Ardea - CRYSTAL
ids <- crystal_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

crystal_euro_pheno_prs_poly <- crystal_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(crystal_euro_pheno_prs_poly, file = here("Output/crystal_euro_pheno_prs_poly.RData"))

rm(ids, prs, crystal_euro_pheno)

# EuroGout
ids <- eurogout_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

eurogout_euro_pheno_prs_poly <- eurogout_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(eurogout_euro_pheno_prs_poly, file = here("Output/eurogout_euro_pheno_prs_poly.RData"))

rm(ids, prs, eurogout_euro_pheno)

# Ardea - LASSO (all others)
ids <- lasso_other_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

lasso_other_euro_pheno_prs_poly <- lasso_other_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(lasso_other_euro_pheno_prs_poly, file = here("Output/lasso_other_euro_pheno_prs_poly.RData"))

rm(ids, prs, lasso_other_euro_pheno)

# Ardea - LIGHT
ids <- light_euro_pheno$Geno.SampleID

make_prs_poly(ids = ids)

light_euro_pheno_prs_poly <- light_euro_pheno %>% 
  left_join(prs, by = c("Geno.SampleID" = "IID")) %>% 
  filter(!is.na(PRS))

save(light_euro_pheno_prs_poly, file = here("Output/light_euro_pheno_prs_poly.RData"))

rm(ids, prs, light_euro_pheno, make_prs_poly)

# UK Biobank
load("/Volumes/userdata/student_users/nicksumpter/Documents/PhD/Cluster/self_report_med.RData")
tmp <- self_report_med %>% 
  mutate(IID = eid,
         ULT = case_when(allopurinol == TRUE | sulphinpyrazone == TRUE | probenecid == TRUE ~ TRUE, TRUE ~ FALSE)) %>% 
  select(IID, ULT)

load("/Volumes/userdata/student_users/nicksumpter/Documents/PhD/Cluster/final_data_prs.RData")
ukbb_euro_pheno <- final_data_prs %>% 
  select(eid, gout, age:urate) %>% 
  rename(IID = eid,
         GOUT = gout,
         AGECOL = age,
         SEX = sex,
         SURICACID = urate) %>% 
  mutate(COHORT = "UK Biobank") %>% 
  left_join(tmp) %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT)

ids <- ukbb_euro_pheno$IID

# Load in plink genotype files + rename column names + filter to only include IDs of cohort of interest
map <- read_delim(here("Output/Temp/merged_PRS.map"), delim = "\t", col_names = FALSE)
map <- map %>% 
  separate(X2, into = c("X2", "remove"), sep = ",") %>% 
  select(-remove)
x <- read_delim(here("Output/Temp/merged_PRS.ped"), delim = " ", col_names = FALSE, col_types = cols(.default = col_character()))
colnames(x)[seq(from = 7, to = ncol(x) - 1, by = 2)] <- str_c(map$X2, "_1")
colnames(x)[seq(from = 8, to = (ncol(x)), by = 2)] <- str_c(map$X2, "_2")
colnames(x)[1:6] <- c("FID", "IID", "PID", "MID", "SEX", "AFF")
tmp <- c("FID", "IID", "PID", "MID", "SEX", "AFF", paste0(Poly_Gene_OR$RSID, "_1"), paste0(Poly_Gene_OR$RSID, "_2"))
x <- x[, colnames(x) %in% tmp]
x <- x %>% mutate(IID = as.numeric(IID)) %>% filter(IID %in% ids)

# Convert character genotypes into numeric genotypes based on risk allele = 1
num_cols <- ncol(x)
for (i in 1:nrow(Poly_Gene_OR)) {
  x[[2 * i + 5]] <- x[[2 * i + 5]] %>% 
      str_replace("0", "NA") %>% 
    str_replace(Poly_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
    str_replace(Poly_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
    as.numeric()
  x[[2 * i + 6]] <- x[[2 * i + 6]] %>% 
      str_replace("0", "NA") %>% 
    str_replace(Poly_Gene_OR[[i, "Effect_Allele"]], "1") %>% 
    str_replace(Poly_Gene_OR[[i, "Alternate_Allele"]], "0") %>% 
    as.numeric()
  x <- x %>% 
    mutate("TEMP" = (x[[2 * i + 5]] + x[[2 * i + 6]]))
  colnames(x) <- c(colnames(x[1:((num_cols - 1) + i)]), Poly_Gene_OR[[i, "RSID"]])
}
x <- x %>% 
  select(2, (num_cols + 1):ncol(x))

# Save this dataframe for individual SNP analysis
x1 <- x

# Now to calculate the PRS
for(i in 1:nrow(Poly_Gene_OR)) {
  x[i + 1] <- x[[i + 1]] * Poly_Gene_OR[[i, "OR"]]
}
x$PRS <- rowSums(x[2:(ncol(x))])

# Calculating leave-one-out PRS's
num_cols <- ncol(x)
for(i in 1:(ncol(x) - 2)) {
  x[i + 1] <- x[[num_cols]] - x[[i + 1]]
}
x <- x %>% 
  select(IID, PRS, 2:(ncol(x) - 1))
colnames(x)[3:ncol(x)] <- str_c(paste0(colnames(x[2]), "_Less_"), colnames(x[3:ncol(x)]))

# Joining them back together
x <- x %>% left_join(x1)

ukbb_poly_pheno_prs <- ukbb_euro_pheno %>% 
    left_join(x, by = c("IID"))

save(ukbb_poly_pheno_prs, file = here("Output/ukbb_poly_pheno_prs.RData"))
```
