## Data Exploration

The purpose of this section is to explore the data that was generated in the previous section.

After loading in the data, I first produced a table of missing data for all of my variables of interest. I used this as evidence for filtering individuals. After filtering the data, and removing certain phenotypes that had too much missing data, I then re-observed the missingness and reported the proportion for each variable.

Next, I produced a table that shows the distribution of all variables of interest in all cohorts. I then plotted each of these distributions in all cohorts. Next, I plotted the relationship between several combinations of variables. Finally, I plotted the minor allele frequency of all SNPs in each cohort.

There is also code for observing the relationship of serum urate measurements over time, though this is not for this project.

```{r dataprep2}
load(here("Output/UKBB_Gene_OR.RData"))

load(here("Output/all_pheno_prs.RData"))

# Making FLARE_CAT variable and setting all control gout severity traits to NA and removing any non-Europeans with an imputed PRS
all_pheno_prs <- all_pheno_prs %>%
  mutate(FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"),
                                       "One every one to two months",
                                       "One or more per month"),
                            labels = c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52"),
                            ordered = TRUE),
         AGE1ATK = case_when(GOUT ~ AGE1ATK),
         DURATION = case_when(GOUT ~ DURATION),
         NUMATK = case_when(GOUT ~ NUMATK),
         TOPHIGOUT = case_when(GOUT ~ TOPHIGOUT),
         ULT = case_when(GOUT ~ ULT))

all_pheno_prs_male <- all_pheno_prs %>% 
  filter(SEX == "Male")

all_pheno_prs_female <- all_pheno_prs %>% 
  filter(SEX == "Female")

cohortstring <- c("UK Biobank - Gout - Male",
                  "UK Biobank - Gout - Female",
                  "UK Biobank - Control - Male",
                  "UK Biobank - Control - Female",
                  "Aus/NZ European - Gout - Male",
                  "Aus/NZ European - Gout - Female",
                  "Aus/NZ European - Control - Male",
                  "Aus/NZ European - Control - Female",
                  "GlobalGout - Gout - Male",
                  "GlobalGout - Gout - Female",
                  "GlobalGout - Control - Male",
                  "GlobalGout - Control - Female",
                  "Ardea - LASSO - Male",
                  "Ardea - LASSO - Female",
                  "Ardea - CLEAR1 - Male",
                  "Ardea - CLEAR1 - Female",
                  "Ardea - CLEAR2 - Male",
                  "Ardea - CLEAR2 - Female",
                  "Ardea - CRYSTAL - Male",
                  "Ardea - CRYSTAL - Female",
                  "Ardea - LIGHT - Male",
                  "Ardea - LIGHT - Female",
                  "East Polynesian - Gout - Male",
                  "East Polynesian - Gout - Female",
                  "East Polynesian - Control - Male",
                  "East Polynesian - Control - Female",
                  "East Polynesian - Gout - Male - NP",
                  "East Polynesian - Gout - Female - NP",
                  "East Polynesian - Control - Male - NP",
                  "East Polynesian - Control - Female - NP",
                  "West Polynesian - Gout - Male",
                  "West Polynesian - Gout - Female",
                  "West Polynesian - Control - Male",
                  "West Polynesian - Control - Female")

data_list <- list(all_pheno_prs_male %>% filter(GOUT,
                                                Pheno.Study == "UK Biobank"),
                  all_pheno_prs_female %>% filter(GOUT,
                                                  Pheno.Study == "UK Biobank"),
                  all_pheno_prs_male %>% filter(!GOUT,
                                                Pheno.Study == "UK Biobank"),
                  all_pheno_prs_female %>% filter(!GOUT,
                                                  Pheno.Study == "UK Biobank"),
                  all_pheno_prs_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                  Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                  Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs_male %>% filter(GOUT,
                                                Pheno.Study == "EuroGout",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(GOUT,
                                                  Pheno.Study == "EuroGout",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(!GOUT,
                                                Pheno.Study == "EuroGout",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(!GOUT,
                                                  Pheno.Study == "EuroGout",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: 401",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: 401",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: CLEAR1",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: CLEAR1",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: CLEAR2",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: CLEAR2",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: CRYSTAL",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: CRYSTAL",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: LIGHT",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: LIGHT",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study != "Ngati Porou"),
                  all_pheno_prs_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                  Pheno.Study != "Ngati Porou"),
                  all_pheno_prs_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study != "Ngati Porou"),
                  all_pheno_prs_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                  Pheno.Study != "Ngati Porou"),
                  all_pheno_prs_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study == "Ngati Porou"),
                  all_pheno_prs_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                  Pheno.Study == "Ngati Porou"),
                  all_pheno_prs_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study == "Ngati Porou"),
                  all_pheno_prs_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                  Pheno.Study == "Ngati Porou"),
                  all_pheno_prs_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  all_pheno_prs_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  all_pheno_prs_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  all_pheno_prs_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")))

clean_names <- c("UK&nbsp;Biobank<br/>Gout<br/>Male",
                 "UK&nbsp;Biobank<br/>Gout<br/>Female",
                 "UK&nbsp;Biobank<br/>Control<br/>Male",
                 "UK&nbsp;Biobank<br/>Control<br/>Female",
                 "Aus/NZ&nbsp;European<br/>Gout<br/>Male",
                 "Aus/NZ&nbsp;European<br/>Gout<br/>Female",
                 "Aus/NZ&nbsp;European<br/>Control<br/>Male",
                 "Aus/NZ&nbsp;European<br/>Control<br/>Female",
                 "GlobalGout<br/>Gout<br/>Male",
                 "GlobalGout<br/>Gout<br/>Female",
                 "GlobalGout<br/>Control<br/>Male",
                 "GlobalGout<br/>Control<br/>Female",
                 "Ardea<br/>LASSO<br/>Gout<br/>Male",
                 "Ardea<br/>LASSO<br/>Gout<br/>Female",
                 "Ardea<br/>CLEAR1<br/>Gout<br/>Male",
                 "Ardea<br/>CLEAR1<br/>Gout<br/>Female",
                 "Ardea<br/>CLEAR2<br/>Gout<br/>Male",
                 "Ardea<br/>CLEAR2<br/>Gout<br/>Female",
                 "Ardea<br/>CRYSTAL<br/>Gout<br/>Male",
                 "Ardea<br/>CRYSTAL<br/>Gout<br/>Female",
                 "Ardea<br/>LIGHT<br/>Gout<br/>Male",
                 "Ardea<br/>LIGHT<br/>Gout<br/>Female",
                 "East&nbsp;Polynesian<br/>Gout<br/>Male",
                 "East&nbsp;Polynesian<br/>Gout<br/>Female",
                 "East&nbsp;Polynesian<br/>Control<br/>Male",
                 "East&nbsp;Polynesian<br/>Control<br/>Female",
                 "East&nbsp;Polynesian<br/>Gout<br/>Male<br/>NP",
                 "East&nbsp;Polynesian<br/>Gout<br/>Female<br/>NP",
                 "East&nbsp;Polynesian<br/>Control<br/>Male<br/>NP",
                 "East&nbsp;Polynesian<br/>Control<br/>Female<br/>NP",
                 "West&nbsp;Polynesian<br/>Gout<br/>Male",
                 "West&nbsp;Polynesian<br/>Gout<br/>Female",
                 "West&nbsp;Polynesian<br/>Control<br/>Male",
                 "West&nbsp;Polynesian<br/>Control<br/>Female")
```

```{r functions2}
# Functions 
report <- function(x) {
    if(sum(is.na(x)) != length(x)) {
      paste0(sprintf(mean(x, na.rm = TRUE), fmt = "%#.1f"), " ± ", sprintf(sd(x, na.rm = TRUE), fmt = "%#.1f"))
    } else {
      paste0("NA")
    }
}

report_median <- function(x) {
    if(sum(is.na(x)) != length(x)) {
      paste0(median(x, na.rm =T), " (", summary(x)[[2]], " - ", summary(x)[[5]], ")")
    } else {
      paste0("NA")
    }
}

sumreport <- function(x) {
  if(sum(is.na(x)) != length(x)){
    paste0(sum(x, na.rm = TRUE), " (", sprintf((mean(x, na.rm = TRUE) * 100), fmt = "%#.1f"), ")")
  } else {
      paste0("NA")
  }
}

transpose_df <- function(df) {
  t_df <- data.table::transpose(df)
  colnames(t_df) <- rownames(df)
  rownames(t_df) <- colnames(df)
  t_df <- t_df %>%
    rownames_to_column() %>%
    as_tibble() %>% 
    row_to_names(row_number = 1)
  return(t_df)
}

missing <- function(x){
  if(sum(is.na(x)) == length(x)) {
    return("All")
    } else if(sum(!is.na(x)) == length(x)){
      return("None")
      } else {
      paste0(format(sum(is.na(x)), big.mark = ","), " (", format(round((sum(is.na(x)) / length(x) * 100), digits = 1), nsmall = 1), ")")
  }
}
```

<br/>

### Table of Missing Data Percentages

```{r MissingPercent}
table1 <- tibble("Cohort" = cohortstring,
                 "N" = unlist(lapply(data_list, function(x) nrow(x))),
                 "Age at Collection" = unlist(lapply(data_list, function(x) missing(x$AGECOL))),
                 "Serum Urate" = unlist(lapply(data_list, function(x) missing(x$URATE))),
                 "ULT" = unlist(lapply(data_list, function(x) missing(x$ULT))),
                 "Age at Onset" = unlist(lapply(data_list, function(x) missing(x$AGE1ATK))),
                 "Disease Duration" = unlist(lapply(data_list, function(x) missing(x$DURATION))),
                 "Flares" = unlist(lapply(data_list, function(x) missing(x$NUMATK))),
                 "Tophi" = unlist(lapply(data_list, function(x) missing(x$TOPHIGOUT))),
                 "PRS" = unlist(lapply(data_list, function(x) missing(x$PRS))),
                 "Prophylaxis" = unlist(lapply(data_list, function(x) missing(x$PROPHY))),
                 "BMI" = unlist(lapply(data_list, function(x) missing(x$BMI))),
                 "Hypertension" = unlist(lapply(data_list, function(x) missing(x$HYPERTENSION))),
                 "Type 2 Diabetes" = unlist(lapply(data_list, function(x) missing(x$DIABETES))),
                 "Heart Disease" = unlist(lapply(data_list, function(x) missing(x$HEART))),
                 "Kidney Disease" = unlist(lapply(data_list, function(x) missing(x$KIDNEY))),
                 "Dyslipidemia" = unlist(lapply(data_list, function(x) missing(x$LIPIDS))),
                 "Stroke" = unlist(lapply(data_list, function(x) missing(x$STROKE))),
                 "Alcoholic Drinks / Week" = unlist(lapply(data_list, function(x) missing(x$TOTALALC))),
                 "Sugar-Sweetened Drinks / Week" = unlist(lapply(data_list, function(x) missing(x$SUGDRINK))),
                 "Current Smoker" = unlist(lapply(data_list, function(x) missing(x$CURSMOKE))),
                 "Family History of Gout" = unlist(lapply(data_list, function(x) missing(x$FAMGOUT))),
                 "No. Relatives w/ Gout" = unlist(lapply(data_list, function(x) missing(x$FAMGOUTNUM))))

table1 <- transpose_df(table1) %>% 
  column_to_rownames(var = "Cohort") %>% 
  mutate(across(.cols = 1:ncol(table1), ~ str_replace_all(string = .x, pattern = " ", replacement = "&nbsp;")))

row.names(table1) <- str_replace_all(row.names(table1), " ", "&nbsp;")

table1 %>% 
  kable(col.names = clean_names,
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px") %>% 
  footnote("'All' = all missing, 'None' = none missing")

#datatable(table1, extension = "Responsive")
```

From this table we can tell the following:

1. Age at collection is only missing in GlobalGout, LASSO and West Polynesian Controls, and thus anybody missing this variable should be removed

2. Serum urate has less than 5% missingness in all European cohorts except for GlobalGout (with 22%), Polynesian controls also have approximately 10% missingness

3. ULT data seems to be missing for around 50% of people in Gout in Aotearoa, GlobalGout, and the CRYSTAL trial but is present for all other cohorts

4. Age at onset (and disease duration) are missing at around 10% or less for all cohorts except GlobalGout, which has over 40% missingness for this variable

5. Flare data is missing at up to 20% for most cohorts, but around 43% missing in GlobalGout

6. Tophi data is not missing in Ardea, around 10 - 25% missing in Gout in Aotearoa, and 67% missing in GlobalGout

7. The PRS is missing in up to 8% of individuals (mainly GlobalGout and UK Biobank)

8. Prophylaxis, comorbidity, lifestyle, and family history data are inconsistently phenotyped

Therefore, I will take the following actions:

1. I will remove anyone missing age at collection

2. I will remove anyone missing all three severity traits (excluding controls and UK Biobank)

3. I will remove anyone missing the PRS

<br/>

```{r MissingPercent2}
all_pheno_prs2 <- all_pheno_prs %>% 
  filter(!is.na(AGECOL),
         !is.na(PRS),
         (Pheno.Study == "UK Biobank" | !GOUT | GOUT & !(is.na(AGE1ATK) & is.na(NUMATK) & is.na(TOPHIGOUT))))

all_pheno_prs2_male <- all_pheno_prs2 %>% 
  filter(SEX == "Male")

all_pheno_prs2_female <- all_pheno_prs2 %>% 
  filter(SEX == "Female")

data_list2 <- list(all_pheno_prs2_male %>% filter(GOUT,
                                                Pheno.Study == "UK Biobank"),
                  all_pheno_prs2_female %>% filter(GOUT,
                                                  Pheno.Study == "UK Biobank"),
                  all_pheno_prs2_male %>% filter(!GOUT,
                                                Pheno.Study == "UK Biobank"),
                  all_pheno_prs2_female %>% filter(!GOUT,
                                                  Pheno.Study == "UK Biobank"),
                  all_pheno_prs2_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs2_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                  Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs2_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs2_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                  Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  all_pheno_prs2_male %>% filter(GOUT,
                                                Pheno.Study == "EuroGout",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(GOUT,
                                                  Pheno.Study == "EuroGout",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(!GOUT,
                                                Pheno.Study == "EuroGout",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(!GOUT,
                                                  Pheno.Study == "EuroGout",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(Pheno.Study == "Ardea: 401",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(Pheno.Study == "Ardea: 401",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(Pheno.Study == "Ardea: CLEAR1",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(Pheno.Study == "Ardea: CLEAR1",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(Pheno.Study == "Ardea: CLEAR2",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(Pheno.Study == "Ardea: CLEAR2",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(Pheno.Study == "Ardea: CRYSTAL",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(Pheno.Study == "Ardea: CRYSTAL",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(Pheno.Study == "Ardea: LIGHT",
                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_female %>% filter(Pheno.Study == "Ardea: LIGHT",
                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  all_pheno_prs2_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study != "Ngati Porou"),
                  all_pheno_prs2_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study != "Ngati Porou"),
                  all_pheno_prs2_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study != "Ngati Porou"),
                  all_pheno_prs2_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study != "Ngati Porou"),
                  all_pheno_prs2_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study == "Ngati Porou"),
                  all_pheno_prs2_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study == "Ngati Porou"),
                  all_pheno_prs2_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study == "Ngati Porou"),
                  all_pheno_prs2_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("East Polynesian"),
                                                Pheno.Study == "Ngati Porou"),
                  all_pheno_prs2_male %>% filter(GOUT,
                                                Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  all_pheno_prs2_female %>% filter(GOUT,
                                                  Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  all_pheno_prs2_male %>% filter(!GOUT,
                                                Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  all_pheno_prs2_female %>% filter(!GOUT,
                                                  Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")))

table1 <- tibble("Cohort" = cohortstring,
                 "N" = unlist(lapply(data_list2, function(x) nrow(x))),
                 "Age at Collection" = unlist(lapply(data_list2, function(x) missing(x$AGECOL))),
                 "Serum Urate" = unlist(lapply(data_list2, function(x) missing(x$URATE))),
                 "ULT" = unlist(lapply(data_list2, function(x) missing(x$ULT))),
                 "Age at Onset" = unlist(lapply(data_list2, function(x) missing(x$AGE1ATK))),
                 "Disease Duration" = unlist(lapply(data_list2, function(x) missing(x$DURATION))),
                 "Flares" = unlist(lapply(data_list2, function(x) missing(x$NUMATK))),
                 "Tophi" = unlist(lapply(data_list2, function(x) missing(x$TOPHIGOUT))),
                 "PRS" = unlist(lapply(data_list2, function(x) missing(x$PRS))),
                 "Prophylaxis" = unlist(lapply(data_list2, function(x) missing(x$PROPHY))),
                 "BMI" = unlist(lapply(data_list2, function(x) missing(x$BMI))),
                 "Hypertension" = unlist(lapply(data_list2, function(x) missing(x$HYPERTENSION))),
                 "Type 2 Diabetes" = unlist(lapply(data_list2, function(x) missing(x$DIABETES))),
                 "Heart Disease" = unlist(lapply(data_list2, function(x) missing(x$HEART))),
                 "Kidney Disease" = unlist(lapply(data_list2, function(x) missing(x$KIDNEY))),
                 "Dyslipidemia" = unlist(lapply(data_list2, function(x) missing(x$LIPIDS))),
                 "Stroke" = unlist(lapply(data_list2, function(x) missing(x$STROKE))),
                 "Alcoholic Drinks / Week" = unlist(lapply(data_list2, function(x) missing(x$TOTALALC))),
                 "Sugar-Sweetened Drinks / Week" = unlist(lapply(data_list2, function(x) missing(x$SUGDRINK))),
                 "Current Smoker" = unlist(lapply(data_list2, function(x) missing(x$CURSMOKE))),
                 "Family History of Gout" = unlist(lapply(data_list2, function(x) missing(x$FAMGOUT))),
                 "No. Relatives w/ Gout" = unlist(lapply(data_list2, function(x) missing(x$FAMGOUTNUM))))

table1 <- transpose_df(table1) %>% 
  column_to_rownames(var = "Cohort") %>% 
  mutate(across(.cols = 1:ncol(table1), ~ str_replace(string = .x, pattern = " ", replacement = "&nbsp;")))

row.names(table1) <- str_replace(row.names(table1), " ", "&nbsp;")

table1 %>% 
  kable(col.names = clean_names,
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px") %>% 
  footnote("'All' = all missing, 'None' = none missing")
```

So now the following is true about the final cohort:

1. Age at collection and the PRS are both fully phenotyped

2. Serum urate has less than 10% missingness in all cohorts except the GlobalGout controls (80% missing)

3. ULT has around 30-40% missingness in most gout cohorts except 4 of the 5 Ardea cohorts and the UK Biobank gout cohort

4. The three severity traits are completely available in all Ardea cohorts and completely missing in the UK Biobank. In the remaining cohorts, age at onset/disease duration has around 5 - 10% missingness, flares have around 5 - 15% missingness, and tophi have 10 - 20% missingness for Aus/NZ and around 45% missingness in GlobalGout

5. Prophylaxis, comorbidity, lifestyle and family history data are poorly phenotyped overall

Given the relatively low proportions of missingness for our most important variables, we are probably okay to just use complete cases only for each model. This will of course reduce power, and may introduce bias which I need to be okay with as a limitation.

<br/>

### Characteristics of each Cohort

```{r Table1}
table1 <- tibble("Cohort" = cohortstring, 
                 "N" = unlist(lapply(data_list2, nrow)),
                 "Age at Collection (years)" = unlist(lapply(data_list2, function(x) report(x$AGECOL))),
                 "Serum Urate (mg/dL)" = unlist(lapply(data_list2, function(x) report(x$URATE))),
                 "ULT" = unlist(lapply(data_list2, function(x) sumreport(x$ULT))),
                 "Age at Onset (years)" = unlist(lapply(data_list2, function(x) report(x$AGE1ATK))),
                 "Disease Duration (years)" = unlist(lapply(data_list2, function(x) report(x$DURATION))),
                 "Number of Flares in Last Year" = unlist(lapply(data_list2, function(x) report_median(x$NUMATK))),
                 "Presence of Tophi" = unlist(lapply(data_list2, function(x) sumreport(x$TOPHIGOUT))),
                 "PRS" = unlist(lapply(data_list2, function(x) report(x$PRS))),
                 "Prophylaxis" = unlist(lapply(data_list2, function(x) sumreport(x$PROPHY))),
                 "BMI" = unlist(lapply(data_list2, function(x) report(x$BMI))),
                 "Hypertension" = unlist(lapply(data_list2, function(x) sumreport(x$HYPERTENSION))),
                 "Type 2 Diabetes" = unlist(lapply(data_list2, function(x) sumreport(x$DIABETES))),
                 "Heart Disease" = unlist(lapply(data_list2, function(x) sumreport(x$HEART))),
                 "Kidney Disease" = unlist(lapply(data_list2, function(x) sumreport(x$KIDNEY))),
                 "Dyslipidemia" = unlist(lapply(data_list2, function(x) sumreport(x$LIPIDS))),
                 "Stroke" = unlist(lapply(data_list2, function(x) sumreport(x$STROKE))),
                 "Alcoholic Drinks / Week" = unlist(lapply(data_list2, function(x) report(x$TOTALALC))),
                 "Sugar-Sweetened Drinks / Week" = unlist(lapply(data_list2, function(x) report(x$SUGDRINK))),
                 "Current Smoker" = unlist(lapply(data_list2, function(x) sumreport(x$CURSMOKE))),
                 "Family History of Gout" = unlist(lapply(data_list2, function(x) sumreport(x$FAMGOUT))),
                 "No. Relatives w/ Gout" = unlist(lapply(data_list2, function(x) report(x$FAMGOUTNUM))))

table1 <- transpose_df(table1) %>% 
  column_to_rownames(var = "Cohort") %>% 
  mutate(across(.cols = 1:ncol(table1), ~ str_replace(string = .x, pattern = " ", replacement = "&nbsp;")))

row.names(table1) <- str_replace(row.names(table1), " ", "&nbsp;")

table1 %>% 
  kable(col.names = clean_names,
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px") %>% 
  footnote("Flares reported as median (inter-quartile range). All other numeric variables reported as mean ± sd. All binary variables reported as N (%).")
```

From this table we learn the following:

1. In general, gout cohorts consist of approximately 80 - 90% males, while control cohorts are closer to 50% male.

2. Male European gout cohorts are mostly 60 years old on average, except for Ardea cohorts who are younger (around 50-55). Male East Polynesian gout cohorts are around 50 - 60 years old, while the male West Polynesian gout cases are younger (average of 47 years old). Female gout cohorts are consistently 5-10 years older than their male counterparts. Controls are generally 5 to 10 years younger than cases.

3. Serum urate levels for gout cohorts tend to be between 6.5 and 9 mg/dL, while control cohorts sit around 4 to 6 mg/dL on average. Females in general have lower serum urate than males.

4. Most gout cohorts have high rates of ULT (many are over 90% on ULT - though this is likely related to missing data issues). Similar ULT usage is seen in males and females, though perhaps slightly less females are prescribed ULT than males.

5. Mean onset is around 40 - 50 for European cohorts and around 35 - 40 for Polynesians. Average disease duration is around 10 - 20 years for all cohorts. All females have a much higher age at onset compared to males and a shorter disease duration.

6. Gout cohorts have up to 10 flares per year on average. No obvious difference in flares between males and females, perhaps slightly lower in Polynesian females vs Polynesian males.

7. Gout cohorts tend to have between 15 and 50% tophaceous gout. Inconsistent decrease in number of tophi in females vs males.

8. The PRS sits around 4.0 to 4.4 for European gout cohorts, around 3.7 for European control cohorts, around 4.4 for East Polynesian gout cohorts, 4.8 for West Polynesian gout cohorts and around 4.2 to 4.3 for Polynesian control cohorts. No obvious difference between males and females.

9. For cohorts with prophylaxis data, almost all gout cases are on prophylaxis. No obvious sex differences.

10. European gout cases have a mean BMI of between 30 and 35 on average, while European controls are closer to 27 on average. Polynesian cases have a mean BMI of 36 while controls have around 33 on average. Female gout cases consistently have higher BMI and higher rates of all comorbidities than their male counterparts, while female controls have lower BMI and lower rates of all comorbidities than male controls.

12. Gout cohorts are around 50 - 100% hypertensive, while European control cohorts are around 35 - 50% and Polynesian control cohorts are 90 - 95% hypertensive

13. Type 2 diabetes seems to be more common in gout cohorts, at between 10 and 40%, while it sits at around 7 - 13% for European controls, and around 23% for Polynesian controls 

14. Heart disease is around 20 - 45% frequency in gout cohorts, though Ardea cohorts all show lower amounts at around 5 - 10% each. The control cohorts sit around 10 - 20% each

15. Kidney disease is relatively common among the Aus/NZ cohorts and the GlobalGout cohort, but not as common among the Ardea cohorts or UK Biobank. This may be because mainly the people involved in renal disease studies had kidney disease measurements

16. Dyslipidemia clearly associates with gout status, sitting at around 20 to 42% in European control cohorts vs 50 to 80% in European gout cohorts. Ardea cohorts have closer to 40 to 50% dyslipidemia however. The Polynesian cohorts appear to have very high proportions of dyslipidemia, particularly in the gout cohorts.

17. Stroke proportions are mostly low, except for the Aus/NZ control cohort who have very high proportions of this disease, and in general stroke associates with gout status.

18. Alcoholic drinks do associate with gout status, and generally Polynesians drink less than Europeans. Males tend to drink much more than females, regardless of gout status, but male gout vs controls have higher rates of drinking while females are no different between cases and controls.

19. Sugar sweetened drinks also seem to associate slightly with gout status, and Polynesians drink more of these than Europeans. Sugar-sweetened bevarages seems to be gout associated in all Europeans and West Polynesians but not East Polynesians. Also females tend to drink less of these than men.

20. Smoking status shows no correlation with gout status but does seem to be higher rates in GlobalGout, Ardea, and Polynesian cohorts. Smoking status may be positively gout associated in females but negatively associated in males (European specific).

21. Family history of gout is clearly correlated with gout status, and is much higher in Polynesians in general, which is also reflected by mean no. relatives with gout. As expected, family history of gout associates with gout status independent of sex.

Based on this information, I will take the following action:

1. Remove any cohort with fewer than ~30 individuals (i.e. female Ardea cohorts) as they are unlikely to contribute to the meta-analyses.

2. Ignore any phenotypes that are not phenotyped at more than 80% across all cohorts, given that there are likely biases as to inclusion in phenotyping (i.e. negative ULT cases reported as missing in some cohorts).

<br/>

### Plotting distribution of each variable {.tabset}

```{r}
# Plotting distributions of each of these variables in males and females separately, colored by cohort
all_cohorts <- all_pheno_prs2 %>% 
  mutate(SEX = factor(SEX, levels = c("Male", "Female")),
         GOUT2 = factor(GOUT, levels = c(TRUE, FALSE), labels = c("Gout", "Control")),
         GROUP = factor(case_when(GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "European Gout",
                                  !GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "European Control",
                                  GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian Gout",
                                  !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian Control",
                                  GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian Gout - NP",
                                  !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian Control - NP",
                                  GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian Gout",
                                  !GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian Control"), 
                        levels = c("European Gout", "European Control", "East Polynesian Gout", "East Polynesian Control", "East Polynesian Gout - NP", "East Polynesian Control - NP", "West Polynesian Gout", "West Polynesian Control")),
         GROUP2 = factor(case_when(GROUP == "European Gout" & SEX == "Male" ~ "European Gout - male",
                                  GROUP == "European Gout" & SEX == "Female" ~ "European Gout - female",
                                  GROUP == "European Control" & SEX == "Male" ~ "European Control - male",
                                  GROUP == "European Control" & SEX == "Female" ~ "European Control - female",
                                  GROUP == "East Polynesian Gout" & SEX == "Male" ~ "East Polynesian Gout - male",
                                  GROUP == "East Polynesian Gout" & SEX == "Female" ~ "East Polynesian Gout - female",
                                  GROUP == "East Polynesian Control" & SEX == "Male" ~ "East Polynesian Control - male",
                                  GROUP == "East Polynesian Control" & SEX == "Female" ~ "East Polynesian Control - female",
                                  GROUP == "East Polynesian Gout - NP" & SEX == "Male" ~ "East Polynesian Gout - NP - male",
                                  GROUP == "East Polynesian Gout - NP" & SEX == "Female" ~ "East Polynesian Gout - NP - female",
                                  GROUP == "East Polynesian Control - NP" & SEX == "Male" ~ "East Polynesian Control - NP - male",
                                  GROUP == "East Polynesian Control - NP" & SEX == "Female" ~ "East Polynesian Control - NP - female",
                                  GROUP == "West Polynesian Gout" & SEX == "Male" ~ "West Polynesian Gout - male",
                                  GROUP == "West Polynesian Gout" & SEX == "Female" ~ "West Polynesian Gout - female",
                                  GROUP == "West Polynesian Control" & SEX == "Male" ~ "West Polynesian Control - male",
                                  GROUP == "West Polynesian Control" & SEX == "Female" ~ "West Polynesian Control - female"), 
                        levels = c("European Gout - male", "European Gout - female", "European Control - male", "European Control - female", "East Polynesian Gout - male", "East Polynesian Gout - female", "East Polynesian Control - male", "East Polynesian Control - female", "East Polynesian Gout - NP - male", "East Polynesian Gout - NP - female", "East Polynesian Control - NP - male", "East Polynesian Control - NP - female", "West Polynesian Gout - male", "West Polynesian Gout - female", "West Polynesian Control - male", "West Polynesian Control - female")),
         GROUP3 = factor(case_when(GOUT & SEX == "Male" ~ "Male Gout",
                                   GOUT & SEX == "Female" ~ "Female Gout",
                                   !GOUT & SEX == "Male" ~ "Male Control",
                                   !GOUT & SEX == "Female" ~ "Female Control"), 
                         levels = c("Male Gout", "Female Gout", "Male Control", "Female Control")),
         COHORT2 = factor(case_when(GOUT & Pheno.Study == "UK Biobank" ~ "UK Biobank - Gout",
                                    !GOUT & Pheno.Study == "UK Biobank" ~ "UK Biobank - Control",
                                    GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") & Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease") ~ "Aus/NZ - Gout",
                                    !GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") & Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease") ~ "Aus/NZ - Control",
                                    GOUT & Pheno.Study == "EuroGout" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "GlobalGout - Gout",
                                    !GOUT & Pheno.Study == "EuroGout" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "GlobalGout - Control",
                                    Pheno.Study == "Ardea: 401" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - LASSO",
                                    Pheno.Study == "Ardea: CLEAR1" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CLEAR1",
                                    Pheno.Study == "Ardea: CLEAR2" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CLEAR2",
                                    Pheno.Study == "Ardea: CRYSTAL" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CRYSTAL",
                                    Pheno.Study == "Ardea: LIGHT" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - LIGHT",
                                    GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian - Gout",
                                    !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study != "Ngati Porou" ~ "East Polynesian - Control",
                                    GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian - Gout - NP",
                                    !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") & Pheno.Study == "Ngati Porou" ~ "East Polynesian - Control - NP",
                                    GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian - Gout",
                                    !GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian - Control"), 
                          levels = c("UK Biobank - Gout", "UK Biobank - Control", "Aus/NZ - Gout", "Aus/NZ - Control", "GlobalGout - Gout", "GlobalGout - Control", "Ardea - LASSO", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - CRYSTAL", "Ardea - LIGHT", "East Polynesian - Gout", "East Polynesian - Control", "East Polynesian - Gout - NP", "East Polynesian - Control - NP", "West Polynesian - Gout", "West Polynesian - Control"))) %>% 
  filter(!is.na(COHORT2))
```

#### Age at collection

```{r}
# Age at collection
all_cohorts %>% 
  ggplot(aes(x = AGECOL, y = COHORT2, color = SEX)) +
    geom_boxplot(position = position_dodge2(reverse = T)) +
    labs(x = "Age at Collection (years)") +
    theme(axis.title.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank()) + 
    scale_x_continuous(limits = c(0, 100)) + 
    scale_y_discrete(limits = rev(levels(all_cohorts$COHORT2)))
```

#### Serum urate

```{r}
# Serum urate
all_cohorts %>% 
  filter(!is.na(URATE)) %>% 
  ggplot(aes(x = URATE, y = COHORT2, color = SEX)) +
    geom_boxplot(position = position_dodge2(reverse = T)) +
    labs(x = "Serum Urate (mg/dL)") +
    theme(axis.title.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank()) + 
    scale_x_continuous(limits = c(0, max(all_cohorts$URATE, na.rm = T))) + 
    scale_y_discrete(limits = rev(levels(all_cohorts$COHORT2)))
```

#### Urate-lowering therapy

```{r}
# ULT
all_cohorts %>% 
  filter(GOUT) %>% 
  mutate(ULT = factor(case_when(ULT ~ "On ULT",
                                !ULT ~ "Not on ULT",
                                is.na(ULT) ~ "No Data"), 
                      levels = c("No Data", "Not on ULT", "On ULT"))) %>% 
  group_by(COHORT2, ULT, SEX) %>% 
  summarize(value = n()) %>% 
  ggplot(aes(x = COHORT2, y = value, fill = ULT, color = COHORT2)) +
    geom_bar(position = "fill", stat = "identity") +
    facet_wrap(~ SEX) +
    scale_fill_discrete(type = c("black", "#C0C0C0", "#505050"), limits = c("On ULT", "Not on ULT", "No Data")) +
    theme(axis.title.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.text.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.y = element_blank(),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank())
```

#### Age at onset

```{r}
# Age at onset
all_cohorts %>% 
  filter(GOUT, !is.na(AGE1ATK)) %>% 
  ggplot(aes(x = AGE1ATK, y = COHORT2, color = SEX)) +
    geom_boxplot(position = position_dodge2(reverse = T)) +
    labs(x = "Age at Onset (years)") +
    theme(axis.title.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank()) + 
    scale_x_continuous(limits = c(0, 100)) + 
    scale_y_discrete(limits = rev(levels(all_cohorts %>% filter(GOUT, !is.na(AGE1ATK)) %>% mutate(COHORT2 = factor(COHORT2)) %>% pull(COHORT2))))
```

#### Disease duration

```{r}
# Disease duration
all_cohorts %>% 
  filter(GOUT, !is.na(DURATION)) %>% 
  ggplot(aes(x = DURATION, y = COHORT2, color = SEX)) +
    geom_boxplot(position = position_dodge2(reverse = T)) +
    labs(x = "Disease Duration (years)") +
    theme(axis.title.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank()) + 
    scale_x_continuous(limits = c(0, max(all_cohorts$DURATION, na.rm = T))) + 
    scale_y_discrete(limits = rev(levels(all_cohorts %>% filter(GOUT, !is.na(DURATION)) %>% mutate(COHORT2 = factor(COHORT2)) %>% pull(COHORT2))))
```

#### Flare frequency

```{r}
# Flare frequency
all_cohorts %>% 
  filter(GOUT, !is.na(NUMATK)) %>% 
  mutate(NUMATK = case_when(NUMATK > 52 ~ 52,
                            TRUE ~ NUMATK)) %>% 
  ggplot(aes(x = NUMATK, y = COHORT2, color = SEX)) +
    geom_boxplot(position = position_dodge2(reverse = T)) +
    labs(x = "Number of Flares in Last Year") +
    theme(axis.title.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank()) + 
    scale_x_continuous(limits = c(0, 52)) + 
    scale_y_discrete(limits = rev(levels(all_cohorts %>% filter(GOUT, !is.na(NUMATK)) %>% mutate(COHORT2 = factor(COHORT2)) %>% pull(COHORT2))))
```

#### Flare frequency (categorical)

```{r}
all_cohorts %>% 
  filter(GOUT) %>% 
  mutate(FLARE_CAT = factor(case_when(is.na(FLARE_CAT) ~ "No Data",
                                      TRUE ~ as.character(FLARE_CAT)), 
                            levels = rev(c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52",
                                       "No Data")),
                            ordered = TRUE)) %>% 
  group_by(COHORT2, FLARE_CAT, SEX) %>% 
  summarize(value = n()) %>% 
  ggplot(aes(x = COHORT2, y = value, fill = FLARE_CAT, color = COHORT2)) +
    geom_bar(position = "fill", stat = "identity") +
    facet_wrap(~ SEX) +
    scale_fill_discrete(type = c("#C0C0C0", "#FDE725FF", "#9FDA3AFF", "#4AC16DFF", "#1FA187FF", "#277F8EFF", "#365C8DFF", "#46337EFF", "#440154FF")) +
    theme(axis.title.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.text.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.y = element_blank(),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank())
```

#### Tophi

```{r}
# Tophi
all_cohorts %>% 
  filter(GOUT) %>% 
  mutate(TOPHIGOUT = factor(case_when(TOPHIGOUT ~ "Tophi",
                                !TOPHIGOUT ~ "No Tophi",
                                is.na(TOPHIGOUT) ~ "No Data"), 
                      levels = c("No Data", "No Tophi", "Tophi"))) %>% 
  group_by(COHORT2, TOPHIGOUT, SEX) %>% 
  summarize(value = n()) %>% 
  ggplot(aes(x = COHORT2, y = value, fill = TOPHIGOUT, color = COHORT2)) +
    geom_bar(position = "fill", stat = "identity") +
    facet_wrap(~ SEX) +
    scale_fill_discrete(type = c("black", "#C0C0C0", "#505050"), limits = c("Tophi", "No Tophi", "No Data")) +
    theme(axis.title.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.text.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.y = element_blank(),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank())
```

#### PRS

```{r}
# PRS
all_cohorts %>% 
  ggplot(aes(x = PRS, y = COHORT2, color = SEX)) +
    geom_boxplot(position = position_dodge2(reverse = T)) +
    labs(x = "Gout PRS") +
    theme(axis.title.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank()) + 
    scale_y_discrete(limits = rev(levels(all_cohorts$COHORT2)))
```

<br/>

### Plotting relationship between age/sex and each outcome of interest {.tabset}

#### Gout

```{r}
# Plotting relationship between covariates (age + sex) and each outcome of interest (GOUT, SU, Onset, Duration, Flares, Tophi)
# Gout
all_cohorts %>%
  filter(!is.na(GROUP)) %>% 
  mutate(GROUP = factor(case_when(str_detect(GROUP, "European") ~ "European",
                                  str_detect(GROUP, "East") & Pheno.Study != "Ngati Porou" ~ "East Polynesian",
                                  str_detect(GROUP, "East") & Pheno.Study == "Ngati Porou" ~ "East Polynesian - NP",
                                  str_detect(GROUP, "West") ~ "West Polynesian"),
                        levels = c("European", "East Polynesian", "East Polynesian - NP", "West Polynesian"))) %>% 
  ggplot(mapping = aes(x = GOUT2, y = AGECOL, fill = GROUP)) +
    stat_summary(fun = mean, geom = "bar", position = position_dodge()) +
    stat_summary(fun.data = mean_se, geom = "errorbar", position = position_dodge(width = 0.9), width = 0.3, alpha = 0.75) +
    facet_wrap(~ SEX) +
    labs(y = "Mean Age at Collection (years)") +
    #scale_fill_discrete(limits = c("Gout", "Control")) +
    theme(axis.title.x = element_blank(),
            panel.grid.major.x = element_blank(),
            panel.grid.minor.x = element_blank(),
            legend.title = element_blank())
```

#### Serum urate v Age

```{r}
# Serum urate
all_cohorts %>% 
  filter(!is.na(URATE)) %>% 
  mutate(ULT2 = factor(case_when(!ULT ~ "Not on ULT",
                                 ULT ~ "On ULT",
                                 is.na(ULT) ~ "No Data / Control"),
                       levels = c("On ULT", "Not on ULT", "No Data / Control"))) %>% 
  ggplot(mapping = aes(x = AGECOL, y = URATE, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX * ULT2) +
    labs(x = "Mean Age at Collection (years)", 
         y = "Serum Urate (mg/dL)") +
    theme(legend.title = element_blank())
```

#### Serum urate v Duration

```{r}
all_cohorts %>% 
  filter(!is.na(URATE)) %>% 
  mutate(ULT2 = factor(case_when(!ULT ~ "Not on ULT",
                                 ULT ~ "On ULT",
                                 is.na(ULT) ~ "No Data / Control"),
                       levels = c("On ULT", "Not on ULT", "No Data / Control"))) %>% 
  ggplot(mapping = aes(x = DURATION, y = URATE, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX * ULT2) +
    labs(x = "Disease Duration (years)", 
         y = "Serum Urate (mg/dL)") +
    theme(legend.title = element_blank())
```

#### ULT

```{r}
# ULT
all_cohorts %>% 
  filter(GOUT) %>% 
  mutate(ULT2 = factor(case_when(!ULT ~ "Not on ULT",
                          ULT ~ "On ULT",
                          is.na(ULT) ~ "No Data"),
                       levels = c("On ULT", "Not on ULT", "No Data"))) %>% 
  ggplot(mapping = aes(x = ULT2, y = AGECOL, fill = GROUP)) +
    stat_summary(fun = mean, geom = "bar", position = position_dodge()) +
    stat_summary(fun.data = mean_se, geom = "errorbar", position = position_dodge(width = 0.9), width = 0.3, alpha = 0.75) +
    facet_wrap(~ SEX) +
    labs(y = "Mean Age at Collection (years)") +
    theme(axis.title.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          legend.title = element_blank())
```

#### Onset

```{r}
# Age at onset
all_cohorts %>% 
  filter(!is.na(AGE1ATK)) %>% 
  ggplot(mapping = aes(x = AGECOL, y = AGE1ATK, color = COHORT2)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Age at Collection (years)", y = "Age at Onset (years)") +
    theme(legend.title = element_blank())
```

#### Duration

```{r}
# Disease duration
all_cohorts %>% 
  filter(GOUT, !is.na(AGE1ATK)) %>% 
  ggplot(mapping = aes(x = AGECOL, y = DURATION, color = COHORT2)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Age at Collection (years)", y = "Disease Duration (years)") +
    theme(legend.title = element_blank())
```

#### Flare frequency (all)

```{r}
# Flare frequency
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         !str_detect(COHORT2, "Ardea")) %>% 
  ggplot(aes(x = FLARE_CAT, y = AGECOL, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Age at Collection (years)") +
    theme(legend.title = element_blank())
```

#### Flare frequency (> 1)

```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         NUMATK >= 2) %>% 
  ggplot(aes(x = FLARE_CAT, y = AGECOL, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Age at Collection (years)") +
    theme(legend.title = element_blank())
```

#### Tophi

```{r}
# Tophi
all_cohorts %>% 
  filter(GOUT,
         !str_detect(COHORT2, "CRYSTAL")) %>% 
  mutate(TOPHIGOUT = factor(case_when(TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi",
                                      is.na(TOPHIGOUT) ~ "No Data"), 
                            levels = c("Tophi", "No Tophi", "No Data"))) %>% 
  ggplot(mapping = aes(x = TOPHIGOUT, y = AGECOL, fill = GROUP)) +
    stat_summary(fun = mean, geom = "bar", position = position_dodge()) +
    stat_summary(fun.data = mean_se, geom = "errorbar", position = position_dodge(width = 0.9), width = 0.3, alpha = 0.75) +
    facet_wrap(~ SEX) +
    labs(y = "Mean Age at Collection (years)") +
    theme(axis.title.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          legend.title = element_blank())
```

<br/>

### Plotting relationship between PRS and each outcome of interest {.tabset}

#### Gout vs PRS
    
```{r}
all_cohorts %>%
  filter(!is.na(GROUP)) %>% 
  ggplot(aes(x = GROUP, y = PRS, fill = GOUT2)) +
    stat_summary(fun = mean, geom = "bar", position = position_dodge()) +
    stat_summary(fun.data = mean_se, geom = "errorbar", position = position_dodge(width = 0.9), width = 0.3, alpha = 0.75) +
    facet_wrap(~ SEX) +
    labs(y = "Gout PRS") +
    scale_fill_discrete(limits = c("Gout", "Control")) +
    theme(axis.title.x = element_blank(),
            panel.grid.major.x = element_blank(),
            panel.grid.minor.x = element_blank(),
            legend.title = element_blank())
```

#### Serum urate vs PRS
    
```{r}
all_cohorts %>% 
  filter(!is.na(URATE)) %>% 
  mutate(ULT2 = factor(case_when(!ULT ~ "Not on ULT",
                                 ULT ~ "On ULT",
                                 is.na(ULT) ~ "No Data / Control"),
                       levels = c("On ULT", "Not on ULT", "No Data / Control"))) %>% 
  ggplot(mapping = aes(x = PRS, y = URATE, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX * ULT2) +
    labs(x = "Gout PRS", 
         y = "Serum Urate (mg/dL)") +
    theme(legend.title = element_blank())
```

#### ULT vs PRS
    
```{r}
all_cohorts %>% 
  filter(GOUT) %>% 
  mutate(ULT2 = factor(case_when(!ULT ~ "Not on ULT",
                          ULT ~ "On ULT",
                          is.na(ULT) ~ "No Data"),
                       levels = c("On ULT", "Not on ULT", "No Data"))) %>% 
  ggplot(mapping = aes(x = ULT2, y = PRS, fill = GROUP)) +
    stat_summary(fun = mean, geom = "bar", position = position_dodge()) +
    stat_summary(fun.data = mean_se, geom = "errorbar", position = position_dodge(width = 0.9), width = 0.3, alpha = 0.75) +
    facet_wrap(~ SEX) +
    labs(y = "Gout PRS") +
    theme(axis.title.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          legend.title = element_blank())
```

#### Onset vs PRS
    
```{r}
all_cohorts %>% 
  filter(!is.na(AGE1ATK)) %>% 
  ggplot(mapping = aes(x = PRS, y = AGE1ATK, color = COHORT2)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Gout PRS", y = "Age at Onset (years)") +
    theme(legend.title = element_blank())
```

#### Duration vs PRS
    
```{r}
all_cohorts %>% 
  filter(!is.na(DURATION)) %>% 
  ggplot(mapping = aes(x = PRS, y = DURATION, color = COHORT2)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Gout PRS", y = "Disease Duration (years)") +
    theme(legend.title = element_blank())
```

#### Flares (all) vs PRS
    
```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         !str_detect(COHORT2, "Ardea")) %>% 
  ggplot(aes(x = FLARE_CAT, y = PRS, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Gout PRS") +
    theme(legend.title = element_blank())
```

#### Flares (> 1) vs PRS
    
```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         NUMATK >= 2) %>% 
  ggplot(aes(x = FLARE_CAT, y = PRS, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Gout PRS") +
    theme(legend.title = element_blank())
```

#### Tophi vs PRS
    
```{r}
all_cohorts %>% 
  filter(GOUT,
         !str_detect(COHORT2, "CRYSTAL")) %>% 
  mutate(TOPHIGOUT = factor(case_when(TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi",
                                      is.na(TOPHIGOUT) ~ "No Data"), 
                            levels = c("Tophi", "No Tophi", "No Data"))) %>% 
  ggplot(mapping = aes(x = TOPHIGOUT, y = PRS, fill = GROUP)) +
    stat_summary(fun = mean, geom = "bar", position = position_dodge()) +
    stat_summary(fun.data = mean_se, geom = "errorbar", position = position_dodge(width = 0.9), width = 0.3, alpha = 0.75) +
    facet_wrap(~ SEX) +
    labs(y = "Gout PRS") +
    theme(axis.title.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          legend.title = element_blank())
```

<br/>

### Plotting relationship between PRS and age/sex {.tabset}

#### Age vs Genotyped PRS

```{r}
all_cohorts %>% 
  ggplot(mapping = aes(x = AGECOL, y = PRS, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Mean Age at Collection (years)", 
         y = "Gout PRS") +
    theme(legend.title = element_blank())
```

<br/>

### Plotting relationship between severity traits {.tabset}

#### Onset vs Tophi

```{r}
# Plotting relationships between severity traits
all_cohorts %>% 
  filter(!is.na(AGE1ATK),
         !str_detect(COHORT2, "CRYSTAL")) %>% 
  mutate(TOPHIGOUT = factor(case_when(is.na(TOPHIGOUT) ~ "No Data",
                                      TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi"),
                            levels = c("Tophi", "No Tophi", "No Data"))) %>% 
  ggplot(aes(y = AGE1ATK, x = TOPHIGOUT, color = GROUP)) +
  geom_boxplot() +
  facet_wrap(~ SEX) +
  labs(y = "Age at Onset (years)") +
  theme(axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        legend.title = element_blank())
```

#### Duration vs Tophi

```{r}
all_cohorts %>% 
  filter(!is.na(DURATION),
         !str_detect(COHORT2, "CRYSTAL")) %>% 
  mutate(TOPHIGOUT = factor(case_when(is.na(TOPHIGOUT) ~ "No Data",
                                      TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi"),
                            levels = c("Tophi", "No Tophi", "No Data"))) %>% 
  ggplot(aes(y = DURATION, x = TOPHIGOUT, color = GROUP)) +
  geom_boxplot() +
  facet_wrap(~ SEX) +
  labs(y = "Disease Duration (years)") +
  theme(axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        legend.title = element_blank())
```

#### Flares (all) vs Tophi

```{r}
all_cohorts %>% 
  filter(!is.na(NUMATK),
         !str_detect(COHORT2, "CRYSTAL"),
         !str_detect(COHORT2, "Ardea")) %>% 
  mutate(TOPHIGOUT = factor(case_when(is.na(TOPHIGOUT) ~ "No Data",
                                      TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi"),
                            levels = c("Tophi", "No Tophi", "No Data")),
         NUMATK = case_when(NUMATK > 52 ~ 52,
                            TRUE ~ NUMATK)) %>% 
  ggplot(aes(y = NUMATK, x = TOPHIGOUT, color = GROUP)) +
  geom_boxplot() +
  facet_wrap(~ SEX) +
  labs(y = "Number of Flares in Last Year") +
  theme(axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        legend.title = element_blank())
```

#### Flares (> 1) vs Tophi

```{r}
all_cohorts %>% 
  filter(!is.na(NUMATK),
         !str_detect(COHORT2, "CRYSTAL"),
         NUMATK >= 2) %>% 
  mutate(TOPHIGOUT = factor(case_when(is.na(TOPHIGOUT) ~ "No Data",
                                      TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi"),
                            levels = c("Tophi", "No Tophi", "No Data")),
         NUMATK = case_when(NUMATK > 52 ~ 52,
                            TRUE ~ NUMATK)) %>% 
  ggplot(aes(y = NUMATK, x = TOPHIGOUT, color = GROUP)) +
  geom_boxplot() +
  facet_wrap(~ SEX) +
  labs(y = "Number of Flares in Last Year") +
  theme(axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        legend.title = element_blank())
```

#### Flares (all - cat) vs Tophi

```{r}
all_cohorts %>% 
  filter(GOUT,
         !str_detect(COHORT2, "CRYSTAL"),
         !str_detect(COHORT2, "Ardea")) %>% 
  mutate(FLARE_CAT = factor(case_when(is.na(FLARE_CAT) ~ "No Data",
                                      TRUE ~ as.character(FLARE_CAT)), 
                            levels = rev(c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52",
                                       "No Data")),
                            ordered = TRUE),
         TOPHIGOUT = factor(case_when(is.na(TOPHIGOUT) ~ "No Data",
                                      TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi"),
                            levels = c("Tophi", "No Tophi", "No Data"))) %>% 
  group_by(GROUP, TOPHIGOUT, FLARE_CAT, SEX) %>% 
  summarize(value = n()) %>% 
  ggplot(aes(fill = FLARE_CAT, y = value, x = TOPHIGOUT, color = TOPHIGOUT)) +
  geom_bar(position = "fill", stat = "identity") +
  facet_wrap(~ SEX * GROUP) +
  scale_fill_discrete(type = c("#C0C0C0", "#FDE725FF", "#9FDA3AFF", "#4AC16DFF", "#1FA187FF", "#277F8EFF", "#365C8DFF", "#46337EFF", "#440154FF")) +
  theme(axis.title.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.text.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.y = element_blank(),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank())
```

#### Flares (> 1 - cat) vs Tophi

```{r}
all_cohorts %>% 
  filter(GOUT,
         !str_detect(COHORT2, "CRYSTAL"),
         NUMATK >= 2) %>% 
  mutate(FLARE_CAT = factor(case_when(is.na(FLARE_CAT) ~ "No Data",
                                      TRUE ~ as.character(FLARE_CAT)), 
                            levels = rev(c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52",
                                       "No Data")),
                            ordered = TRUE),
         TOPHIGOUT = factor(case_when(is.na(TOPHIGOUT) ~ "No Data",
                                      TOPHIGOUT ~ "Tophi",
                                      !TOPHIGOUT ~ "No Tophi"),
                            levels = c("Tophi", "No Tophi", "No Data"))) %>% 
  group_by(GROUP, TOPHIGOUT, FLARE_CAT, SEX) %>% 
  summarize(value = n()) %>% 
  ggplot(aes(fill = FLARE_CAT, y = value, x = TOPHIGOUT, color = TOPHIGOUT)) +
  geom_bar(position = "fill", stat = "identity") +
  facet_wrap(~ SEX * GROUP) +
  scale_fill_discrete(type = c("#C0C0C0", "#FDE725FF", "#9FDA3AFF", "#4AC16DFF", "#1FA187FF", "#277F8EFF", "#365C8DFF", "#46337EFF", "#440154FF")) +
  theme(axis.title.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.text.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.y = element_blank(),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          legend.title = element_blank())
```

#### Onset vs Flares (all)

```{r}
all_cohorts %>% 
  filter(!is.na(AGE1ATK),
         !is.na(NUMATK),
         !str_detect(COHORT2, "Ardea")) %>% 
  ggplot(aes(x = AGE1ATK, y = NUMATK, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Age at Onset (years)", 
         y = "Number of Flares in Last Year") +
    theme(legend.title = element_blank())
```

#### Onset vs Flares (> 1)

```{r}
all_cohorts %>% 
  filter(!is.na(AGE1ATK),
         !is.na(NUMATK),
         NUMATK >= 2) %>% 
  ggplot(aes(x = AGE1ATK, y = NUMATK, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Age at Onset (years)", 
         y = "Number of Flares in Last Year") +
    theme(legend.title = element_blank())
```

#### Duration vs Flares (all)

```{r}
all_cohorts %>% 
  filter(!is.na(DURATION),
         !is.na(NUMATK),
         !str_detect(COHORT2, "Ardea")) %>% 
  ggplot(aes(x = DURATION, y = NUMATK, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Disease Duration (years)", 
         y = "Number of Flares in Last Year") +
    theme(legend.title = element_blank())
```

#### Duration vs Flares (> 1)

```{r}
all_cohorts %>% 
  filter(!is.na(DURATION),
         !is.na(NUMATK),
         NUMATK >= 2) %>% 
  ggplot(aes(x = DURATION, y = NUMATK, color = GROUP)) +
    geom_smooth(method = "lm", se = F) +
    facet_wrap(~ SEX) +
    labs(x = "Disease Duration (years)", 
         y = "Number of Flares in Last Year") +
    theme(legend.title = element_blank())
```

#### Onset vs Flares (all - cat)

```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         !str_detect(COHORT2, "Ardea")) %>% 
  ggplot(aes(x = FLARE_CAT, y = AGE1ATK, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Age at Onset (years)") +
    theme(legend.title = element_blank())
```

#### Onset vs Flares (> 1 - cat)

```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         NUMATK >= 2) %>% 
  ggplot(aes(x = FLARE_CAT, y = AGE1ATK, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Age at Onset (years)") +
    theme(legend.title = element_blank())
```

#### Duration vs Flares (all - cat)

```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         !str_detect(COHORT2, "Ardea")) %>% 
  ggplot(aes(x = FLARE_CAT, y = DURATION, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Disease Duration (years)") +
    theme(legend.title = element_blank())
```

#### Duration vs Flares (> 1 - cat)

```{r}
all_cohorts %>% 
  filter(GOUT, !is.na(FLARE_CAT),
         NUMATK >= 2) %>% 
  ggplot(aes(x = FLARE_CAT, y = DURATION, color = COHORT2)) +
    stat_summary(geom = "point", fun = mean) +
    stat_summary(geom = "errorbar", fun.data = mean_se, width = 0.3) +
    stat_summary(geom = "line", fun = mean, aes(group = COHORT2)) +
    facet_wrap(~ SEX) +
    labs(x = "Number of Flares in Last Year (categorical)", y = "Disease Duration (years)") +
    theme(legend.title = element_blank())
```

<br/>

### Table of MAF for each SNP in each cohort

```{r MAF}
# Making a table with SNPs as rows and columns representing MAF for each cohort for that SNP
SNPlist <- UKBB_Gene_OR$RSID

get_maf <- function(cohort, snps){
  test <- cohort %>% 
    select(all_of(snps)) %>% 
    mutate_all(factor, levels = 0:2) %>% 
    na.omit()
  
  tmp <- c()
  for(i in 1:ncol(test)){
    tmp[i] <- ((sum(test[[i]] == 1) + (sum(test[[i]] == 2) * 2)) / (2 * nrow(test))) %>% sprintf(fmt = "%#.3f")
  }
  return(tmp)
}

freq_table <- tibble("Cohort" = rep(cohortstring, each = length(SNPlist)), 
                     "MAF" = unlist(lapply(data_list2, function(x) get_maf(cohort = x, snps = SNPlist))),
                     "SNP" = rep(SNPlist, length(cohortstring))) %>%
  pivot_wider(names_from = SNP, values_from = MAF)

freq_table <- transpose_df(freq_table) %>% 
  column_to_rownames(var = "Cohort") %>% 
  mutate(across(.cols = 1:ncol(freq_table), ~ str_replace(string = .x, pattern = " ", replacement = "&nbsp;")))

row.names(freq_table) <- str_replace(row.names(freq_table), " ", "&nbsp;")

freq_table %>% 
  kable(col.names = clean_names,
        align = "c",
        escape = F) %>% 
  kable_styling("striped") %>% 
  scroll_box(width = "900px", height = "475px")

# rs35332062 is monomorphic in females in western polynesians so it should be removed from the individual SNP analysis, though it can be left in the PRS
```
