# Running all models of interest

The purpose of this document is to run all models of interest for assessing the relationship between gout genetic risk and severity of gout

```{r dataprep, message = FALSE, warning = FALSE, cache = TRUE}
# Datasets
load(here("Output/all_pheno_prs.RData"))

# Making FLARE_CAT variable and setting all control gout severity traits to NA and removing any non-Europeans with an imputed PRS
all_pheno_prs <- all_pheno_prs %>%
  mutate(FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"),
                                       "One every one to two months",
                                       "One or more per month"),
                            labels = c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52"),
                            ordered = TRUE),
         AGE1ATK = case_when(GOUT ~ AGE1ATK),
         DURATION = case_when(GOUT ~ DURATION),
         NUMATK = case_when(GOUT ~ NUMATK),
         TOPHIGOUT = case_when(GOUT ~ TOPHIGOUT),
         ULT = case_when(GOUT ~ ULT),
         SEX = factor(SEX, levels = c("Male", "Female")),
         GOUT2 = factor(GOUT, levels = c(TRUE, FALSE), labels = c("Gout", "Control")),
         GROUP = factor(case_when(GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "European Gout",
                                  !GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "European Control",
                                  GOUT & Geno.SpecificAncestry %in% c("East Polynesian") ~ "East Polynesian Gout",
                                  !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") ~ "East Polynesian Control",
                                  GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian Gout",
                                  !GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian Control"), 
                        levels = c("European Gout", "European Control", "East Polynesian Gout", "East Polynesian Control", "West Polynesian Gout", "West Polynesian Control")),
         GROUP2 = factor(case_when(GROUP == "European Gout" & SEX == "Male" ~ "European Gout - male",
                                  GROUP == "European Gout" & SEX == "Female" ~ "European Gout - female",
                                  GROUP == "European Control" & SEX == "Male" ~ "European Control - male",
                                  GROUP == "European Control" & SEX == "Female" ~ "European Control - female",
                                  GROUP == "East Polynesian Gout" & SEX == "Male" ~ "East Polynesian Gout - male",
                                  GROUP == "East Polynesian Gout" & SEX == "Female" ~ "East Polynesian Gout - female",
                                  GROUP == "East Polynesian Control" & SEX == "Male" ~ "East Polynesian Control - male",
                                  GROUP == "East Polynesian Control" & SEX == "Female" ~ "East Polynesian Control - female",
                                  GROUP == "West Polynesian Gout" & SEX == "Male" ~ "West Polynesian Gout - male",
                                  GROUP == "West Polynesian Gout" & SEX == "Female" ~ "West Polynesian Gout - female",
                                  GROUP == "West Polynesian Control" & SEX == "Male" ~ "West Polynesian Control - male",
                                  GROUP == "West Polynesian Control" & SEX == "Female" ~ "West Polynesian Control - female"), 
                        levels = c("European Gout - male", "European Gout - female", "European Control - male", "European Control - female", "East Polynesian Gout - male", "East Polynesian Gout - female", "East Polynesian Control - male", "East Polynesian Control - female", "West Polynesian Gout - male", "West Polynesian Gout - female", "West Polynesian Control - male", "West Polynesian Control - female")),
         GROUP3 = factor(case_when(GOUT & SEX == "Male" ~ "Male Gout",
                                   GOUT & SEX == "Female" ~ "Female Gout",
                                   !GOUT & SEX == "Male" ~ "Male Control",
                                   !GOUT & SEX == "Female" ~ "Female Control"), 
                         levels = c("Male Gout", "Female Gout", "Male Control", "Female Control")),
         COHORT2 = factor(case_when(GOUT & Pheno.Study == "UK Biobank" ~ "UK Biobank - Gout",
                                    !GOUT & Pheno.Study == "UK Biobank" ~ "UK Biobank - Control",
                                    GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") & Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease") ~ "Aus/NZ - Gout",
                                    !GOUT & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") & Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease") ~ "Aus/NZ - Control",
                                    GOUT & Pheno.Study == "EuroGout" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "GlobalGout - Gout",
                                    !GOUT & Pheno.Study == "EuroGout" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "GlobalGout - Control",
                                    Pheno.Study == "Ardea: 401" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - LASSO",
                                    Pheno.Study == "Ardea: CLEAR1" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CLEAR1",
                                    Pheno.Study == "Ardea: CLEAR2" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CLEAR2",
                                    Pheno.Study == "Ardea: CRYSTAL" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - CRYSTAL",
                                    Pheno.Study == "Ardea: LIGHT" & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") ~ "Ardea - LIGHT",
                                    GOUT & Geno.SpecificAncestry %in% c("East Polynesian") ~ "East Polynesian - Gout",
                                    !GOUT & Geno.SpecificAncestry %in% c("East Polynesian") ~ "East Polynesian - Control",
                                    GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian - Gout",
                                    !GOUT & Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan") ~ "West Polynesian - Control"), 
                          levels = c("UK Biobank - Gout", "UK Biobank - Control", "Aus/NZ - Gout", "Aus/NZ - Control", "GlobalGout - Gout", "GlobalGout - Control", "Ardea - LASSO", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - CRYSTAL", "Ardea - LIGHT", "East Polynesian - Gout", "East Polynesian - Control", "West Polynesian - Gout", "West Polynesian - Control")),
         NUMATK = round(NUMATK)) %>% 
  filter(Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian") | !(Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")) & is.na(PRS1),
         !is.na(AGECOL),
         !is.na(PRS2),
         !(is.na(PRS1) & Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
         (Pheno.Study == "UK Biobank" | !GOUT | GOUT & !(is.na(AGE1ATK) & is.na(NUMATK) & is.na(TOPHIGOUT))),
         !is.na(COHORT2))

all_pheno_prs_male <- all_pheno_prs %>% 
  filter(SEX == "Male")

all_pheno_prs_female <- all_pheno_prs %>% 
  filter(SEX == "Female")

data_list <- list("UK Biobank - Gout - Male" = all_pheno_prs_male %>% filter(GOUT,
                                                                             Pheno.Study == "UK Biobank"),
                  "UK Biobank - Gout - Female" = all_pheno_prs_female %>% filter(GOUT,
                                                                                 Pheno.Study == "UK Biobank"),
                  "UK Biobank - Control - Male" = all_pheno_prs_male %>% filter(!GOUT,
                                                                                Pheno.Study == "UK Biobank"),
                  "UK Biobank - Control - Female" = all_pheno_prs_female %>% filter(!GOUT,
                                                                                    Pheno.Study == "UK Biobank"),
                  "Aus/NZ European - Gout - Male" = all_pheno_prs_male %>% filter(GOUT,
                                                                                  Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                                                  Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "Aus/NZ European - Gout - Female" = all_pheno_prs_female %>% filter(GOUT,
                                                                                      Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                                                      Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "Aus/NZ European - Control - Male" = all_pheno_prs_male %>% filter(!GOUT,
                                                                                     Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                                                     Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "Aus/NZ European - Control - Female" = all_pheno_prs_female %>% filter(!GOUT,
                                                                                         Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian"),
                                                                                         Pheno.Study %in% c("AGRIA", "Diabetes Mellitus", "Gout in Aotearoa", "LPA", "Ngati Porou", "Renal Disease")),
                  "GlobalGout - Gout - Male" = all_pheno_prs_male %>% filter(GOUT,
                                                                             Pheno.Study == "EuroGout",
                                                                             Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "GlobalGout - Gout - Female" = all_pheno_prs_female %>% filter(GOUT,
                                                                                 Pheno.Study == "EuroGout",
                                                                                 Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "GlobalGout - Control - Male" = all_pheno_prs_male %>% filter(!GOUT,
                                                                                Pheno.Study == "EuroGout",
                                                                                Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "GlobalGout - Control - Female" = all_pheno_prs_female %>% filter(!GOUT,
                                                                                    Pheno.Study == "EuroGout",
                                                                                    Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LASSO - Male" = all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: 401",
                                                                         Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LASSO - Female" = all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: 401",
                                                                             Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR1 - Male" = all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: CLEAR1",
                                                                          Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR1 - Female" = all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: CLEAR1",
                                                                              Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR2 - Male" = all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: CLEAR2",
                                                                          Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CLEAR2 - Female" = all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: CLEAR2",
                                                                              Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CRYSTAL - Male" = all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: CRYSTAL",
                                                                           Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - CRYSTAL - Female" = all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: CRYSTAL",
                                                                               Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LIGHT - Male" = all_pheno_prs_male %>% filter(Pheno.Study == "Ardea: LIGHT",
                                                                         Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "Ardea - LIGHT - Female" = all_pheno_prs_female %>% filter(Pheno.Study == "Ardea: LIGHT",
                                                                             Geno.SpecificAncestry %in% c("European", "European; Iberian", "Iberian")),
                  "East Polynesian - Gout - Male" = all_pheno_prs_male %>% filter(GOUT,
                                                                                  Geno.SpecificAncestry %in% c("East Polynesian")),
                  "East Polynesian - Gout - Female" = all_pheno_prs_female %>% filter(GOUT,
                                                                                      Geno.SpecificAncestry %in% c("East Polynesian")),
                  "East Polynesian - Control - Male" = all_pheno_prs_male %>% filter(!GOUT,
                                                                                     Geno.SpecificAncestry %in% c("East Polynesian")),
                  "East Polynesian - Control - Female" = all_pheno_prs_female %>% filter(!GOUT,
                                                                                         Geno.SpecificAncestry %in% c("East Polynesian")),
                  "West Polynesian - Gout - Male" = all_pheno_prs_male %>% filter(GOUT,
                                                                                  Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  "West Polynesian - Gout - Female" = all_pheno_prs_female %>% filter(GOUT,
                                                                                      Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  "West Polynesian - Control - Male" = all_pheno_prs_male %>% filter(!GOUT,
                                                                                     Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")),
                  "West Polynesian - Control - Female" = all_pheno_prs_female %>% filter(!GOUT,
                                                                                         Geno.SpecificAncestry %in% c("West Polynesian", "Niuean", "Pukapukan")))

load(here("Output/UKBB_Gene_OR.RData"))
load(here("Output/Poly_Gene_OR.RData"))
```

```{r functions, message = FALSE, warning = FALSE, cache = TRUE}
# Functions
# We need a function for each type of model:

onset_meta_euro <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f, data = anz_gout %>% filter(SEX == "Male"))
    tmp2 <- lm(f, data = eurogout %>% filter(SEX == "Male"))
    tmp3 <- lm(f, data = ardea_clear1 %>% filter(SEX == "Male"))
    tmp4 <- lm(f, data = ardea_clear2 %>% filter(SEX == "Male"))
    tmp5 <- lm(f, data = ardea_lasso %>% filter(SEX == "Male"))
    tmp6 <- lm(f, data = ardea_crystal %>% filter(SEX == "Male"))
    tmp7 <- lm(f, data = ardea_light %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]], 
                                                 tmp6$coefficients[[2]], 
                                                 tmp7$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2], 
                                                   summary(tmp6)$coefficients[2, 2], 
                                                   summary(tmp7)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - CRYSTAL", "Ardea - LIGHT"))))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f, data = anz_gout %>% filter(SEX == "Female"))
    tmp2 <- lm(f, data = eurogout %>% filter(SEX == "Female"))
    tmp3 <- lm(f, data = ardea_lasso %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - LASSO"))))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



onset_meta_direct_euro <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f, data = anz_gout_poly %>% filter(SEX == "Male"))
    tmp2 <- lm(f, data = eurogout_poly %>% filter(SEX == "Male"))
    tmp3 <- lm(f, data = ardea_clear1_poly %>% filter(SEX == "Male"))
    tmp4 <- lm(f, data = ardea_clear2_poly %>% filter(SEX == "Male"))
    tmp5 <- lm(f, data = ardea_lasso_poly %>% filter(SEX == "Male"))
    tmp6 <- lm(f, data = ardea_crystal_poly %>% filter(SEX == "Male"))
    tmp7 <- lm(f, data = ardea_light_poly %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]], 
                                                 tmp6$coefficients[[2]], 
                                                 tmp7$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2], 
                                                   summary(tmp6)$coefficients[2, 2], 
                                                   summary(tmp7)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - CRYSTAL", "Ardea - LIGHT"))))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f, data = anz_gout_poly %>% filter(SEX == "Female"))
    tmp2 <- lm(f, data = eurogout_poly %>% filter(SEX == "Female"))
    tmp3 <- lm(f, data = ardea_lasso_poly %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - LASSO"))))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



onset_meta_direct_poly <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(eastpoly_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f, data = eastpoly_gout %>% filter(SEX == "Male"))
    tmp2 <- lm(f, data = westpoly_gout %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"))))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f, data = eastpoly_gout %>% filter(SEX == "Female"))
    tmp2 <- lm(f, data = westpoly_gout %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"))))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



onset_meta_direct_both <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates1 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
    covariates2 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  } else{
    covariates1 <- covariates
    covariates2 <- covariates
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("AGE1ATK", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f1, data = eastpoly_gout %>% filter(SEX == "Male"))
    tmp2 <- lm(f1, data = westpoly_gout %>% filter(SEX == "Male"))
    tmp3 <- lm(f2, data = anz_gout_poly %>% filter(SEX == "Male"))
    tmp4 <- lm(f2, data = eurogout_poly %>% filter(SEX == "Male"))
    tmp5 <- lm(f2, data = ardea_clear1_poly %>% filter(SEX == "Male"))
    tmp6 <- lm(f2, data = ardea_clear2_poly %>% filter(SEX == "Male"))
    tmp7 <- lm(f2, data = ardea_lasso_poly %>% filter(SEX == "Male"))
    tmp8 <- lm(f2, data = ardea_crystal_poly %>% filter(SEX == "Male"))
    tmp9 <- lm(f2, data = ardea_light_poly %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]], 
                                                 tmp6$coefficients[[2]], 
                                                 tmp7$coefficients[[2]], 
                                                 tmp8$coefficients[[2]], 
                                                 tmp9$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2], 
                                                   summary(tmp6)$coefficients[2, 2], 
                                                   summary(tmp7)$coefficients[2, 2], 
                                                   summary(tmp8)$coefficients[2, 2], 
                                                   summary(tmp9)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - CRYSTAL", "Ardea - LIGHT"))))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("AGE1ATK", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("AGE1ATK", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- lm(f1, data = eastpoly_gout %>% filter(SEX == "Female"))
    tmp2 <- lm(f1, data = westpoly_gout %>% filter(SEX == "Female"))
    tmp3 <- lm(f2, data = anz_gout_poly %>% filter(SEX == "Female"))
    tmp4 <- lm(f2, data = eurogout_poly %>% filter(SEX == "Female"))
    tmp5 <- lm(f2, data = ardea_lasso_poly %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout", "Ardea - LASSO"))))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



tophi_meta_euro <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f, family = binomial, data = anz_gout %>% filter(SEX == "Male"))
    tmp2 <- glm(f, family = binomial, data = eurogout %>% filter(SEX == "Male"))
    tmp3 <- glm(f, family = binomial, data = ardea_clear1 %>% filter(SEX == "Male"))
    tmp4 <- glm(f, family = binomial, data = ardea_clear2 %>% filter(SEX == "Male"))
    tmp5 <- glm(f, family = binomial, data = ardea_lasso %>% filter(SEX == "Male"))
    tmp6 <- glm(f, family = binomial, data = ardea_light %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]], 
                                                 tmp6$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2], 
                                                   summary(tmp6)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - LIGHT"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f, family = binomial, data = anz_gout %>% filter(SEX == "Female"))
    tmp2 <- glm(f, family = binomial, data = eurogout %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



tophi_meta_direct_euro <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f, family = binomial, data = anz_gout_poly %>% filter(SEX == "Male"))
    tmp2 <- glm(f, family = binomial, data = eurogout_poly %>% filter(SEX == "Male"))
    tmp3 <- glm(f, family = binomial, data = ardea_clear1_poly %>% filter(SEX == "Male"))
    tmp4 <- glm(f, family = binomial, data = ardea_clear2_poly %>% filter(SEX == "Male"))
    tmp5 <- glm(f, family = binomial, data = ardea_lasso_poly %>% filter(SEX == "Male"))
    tmp6 <- glm(f, family = binomial, data = ardea_light_poly %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]], 
                                                 tmp6$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2], 
                                                   summary(tmp6)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - LIGHT"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f, family = binomial, data = anz_gout_poly %>% filter(SEX == "Female"))
    tmp2 <- glm(f, family = binomial, data = eurogout_poly %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



tophi_meta_direct_poly <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(eastpoly_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f, family = binomial, data = eastpoly_gout %>% filter(SEX == "Male"))
    tmp2 <- glm(f, family = binomial, data = westpoly_gout %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f, family = binomial, data = eastpoly_gout %>% filter(SEX == "Female"))
    tmp2 <- glm(f, family = binomial, data = westpoly_gout %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



tophi_meta_direct_both <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates1 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
    covariates2 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  } else{
    covariates1 <- covariates
    covariates2 <- covariates
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("TOPHIGOUT", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("TOPHIGOUT", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f1, family = binomial, data = eastpoly_gout %>% filter(SEX == "Male"))
    tmp2 <- glm(f1, family = binomial, data = westpoly_gout %>% filter(SEX == "Male"))
    tmp3 <- glm(f2, family = binomial, data = anz_gout_poly %>% filter(SEX == "Male"))
    tmp4 <- glm(f2, family = binomial, data = eurogout_poly %>% filter(SEX == "Male"))
    tmp5 <- glm(f2, family = binomial, data = ardea_clear1_poly %>% filter(SEX == "Male"))
    tmp6 <- glm(f2, family = binomial, data = ardea_clear2_poly %>% filter(SEX == "Male"))
    tmp7 <- glm(f2, family = binomial, data = ardea_lasso_poly %>% filter(SEX == "Male"))
    tmp8 <- glm(f2, family = binomial, data = ardea_light_poly %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]], 
                                                 tmp5$coefficients[[2]], 
                                                 tmp6$coefficients[[2]], 
                                                 tmp7$coefficients[[2]], 
                                                 tmp8$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2], 
                                                   summary(tmp5)$coefficients[2, 2], 
                                                   summary(tmp6)$coefficients[2, 2], 
                                                   summary(tmp7)$coefficients[2, 2], 
                                                   summary(tmp8)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - LIGHT"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("TOPHIGOUT", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("TOPHIGOUT", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- glm(f1, family = binomial, data = eastpoly_gout %>% filter(SEX == "Female"))
    tmp2 <- glm(f1, family = binomial, data = westpoly_gout %>% filter(SEX == "Female"))
    tmp3 <- glm(f2, family = binomial, data = anz_gout_poly %>% filter(SEX == "Female"))
    tmp4 <- glm(f2, family = binomial, data = eurogout_poly %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[2]], 
                                                 tmp2$coefficients[[2]], 
                                                 tmp3$coefficients[[2]], 
                                                 tmp4$coefficients[[2]]), 
                                          seTE = c(summary(tmp1)$coefficients[2, 2], 
                                                   summary(tmp2)$coefficients[2, 2], 
                                                   summary(tmp3)$coefficients[2, 2], 
                                                   summary(tmp4)$coefficients[2, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_euro1 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout %>% filter(SEX == "Male"))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout %>% filter(SEX == "Female"))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_direct_euro1 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Male"))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Female"))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_direct_poly1 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(eastpoly_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Male"))
    tmp2 <- polr(f, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Female"))
    tmp2 <- polr(f, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_direct_both1 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates1 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
    covariates2 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  } else{
    covariates1 <- covariates
    covariates2 <- covariates
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("FLARE_CAT", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("FLARE_CAT", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f1, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Male"))
    tmp2 <- polr(f1, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Male"))
    tmp3 <- polr(f2, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Male"))
    tmp4 <- polr(f2, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Male"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]], 
                                                 tmp4$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2], 
                                                   summary(tmp4)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("FLARE_CAT", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("FLARE_CAT", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f1, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Female"))
    tmp2 <- polr(f1, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Female"))
    tmp3 <- polr(f2, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Female"))
    tmp4 <- polr(f2, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Female"))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]], 
                                                 tmp4$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2], 
                                                   summary(tmp4)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_euro2 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp3 <- polr(f, Hess = TRUE, data = ardea_clear1 %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp4 <- polr(f, Hess = TRUE, data = ardea_clear2 %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp5 <- polr(f, Hess = TRUE, data = ardea_lasso %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp6 <- polr(f, Hess = TRUE, data = ardea_light %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp7 <- polr(f, Hess = TRUE, data = ardea_crystal %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]], 
                                                 tmp4$coefficients[[1]], 
                                                 tmp5$coefficients[[1]], 
                                                 tmp6$coefficients[[1]], 
                                                 tmp7$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2], 
                                                   summary(tmp4)$coefficients[1, 2], 
                                                   summary(tmp5)$coefficients[1, 2], 
                                                   summary(tmp6)$coefficients[1, 2], 
                                                   summary(tmp7)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - LIGHT", "Ardea - CRYSTAL"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp3 <- polr(f, Hess = TRUE, data = ardea_lasso %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - LASSO"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_direct_euro2 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp3 <- polr(f, Hess = TRUE, data = ardea_clear1_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp4 <- polr(f, Hess = TRUE, data = ardea_clear2_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp5 <- polr(f, Hess = TRUE, data = ardea_lasso_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp6 <- polr(f, Hess = TRUE, data = ardea_light_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp7 <- polr(f, Hess = TRUE, data = ardea_crystal_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]], 
                                                 tmp4$coefficients[[1]], 
                                                 tmp5$coefficients[[1]], 
                                                 tmp6$coefficients[[1]], 
                                                 tmp7$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2], 
                                                   summary(tmp4)$coefficients[1, 2], 
                                                   summary(tmp5)$coefficients[1, 2], 
                                                   summary(tmp6)$coefficients[1, 2], 
                                                   summary(tmp7)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - LIGHT", "Ardea - CRYSTAL"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp3 <- polr(f, Hess = TRUE, data = ardea_lasso_poly %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2]), 
                                          studlab = c("Aus/NZ", "EuroGout", "Ardea - LASSO"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_direct_poly2 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(eastpoly_gout, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables <- c(prsnames[i], covariates)
    f <- as.formula(paste("FLARE_CAT", paste(variables, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



flare_meta_direct_both2 <- function(covariates, PCs, modelname) {
  prsnames <- colnames(select(anz_gout_poly, PRS, starts_with("rs")))
  x <- length(prsnames)
  
  if(PCs){
    covariates1 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC1_Oc", "PC2_Oc", "PC3_Oc", "PC4_Oc", "PC5_Oc", "PC6_Oc", "PC7_Oc", "PC8_Oc", "PC9_Oc", "PC10_Oc")
    covariates2 <- c(covariates, "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
  } else{
    covariates1 <- covariates
    covariates2 <- covariates
  }
  
  tmplistmale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("FLARE_CAT", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("FLARE_CAT", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f1, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f1, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp3 <- polr(f2, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp4 <- polr(f2, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp5 <- polr(f2, Hess = TRUE, data = ardea_clear1_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp6 <- polr(f2, Hess = TRUE, data = ardea_clear2_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp7 <- polr(f2, Hess = TRUE, data = ardea_lasso_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp8 <- polr(f2, Hess = TRUE, data = ardea_light_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp9 <- polr(f2, Hess = TRUE, data = ardea_crystal_poly %>% filter(SEX == "Male", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]], 
                                                 tmp4$coefficients[[1]], 
                                                 tmp5$coefficients[[1]], 
                                                 tmp6$coefficients[[1]], 
                                                 tmp7$coefficients[[1]], 
                                                 tmp8$coefficients[[1]], 
                                                 tmp9$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2], 
                                                   summary(tmp4)$coefficients[1, 2], 
                                                   summary(tmp5)$coefficients[1, 2], 
                                                   summary(tmp6)$coefficients[1, 2], 
                                                   summary(tmp7)$coefficients[1, 2], 
                                                   summary(tmp8)$coefficients[1, 2], 
                                                   summary(tmp9)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout", "Ardea - CLEAR1", "Ardea - CLEAR2", "Ardea - LASSO", "Ardea - LIGHT", "Ardea - CRYSTAL"), sm = "OR")))
    tmplistmale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistmale"), tmplistmale, envir = .GlobalEnv)
  
  tmplistfemale <- list()
  for(i in 1:x) {
    variables1 <- c(prsnames[i], covariates1)
    f1 <- as.formula(paste("FLARE_CAT", paste(variables1, collapse = " + "), sep = " ~ "))
    variables2 <- c(prsnames[i], covariates2)
    f2 <- as.formula(paste("FLARE_CAT", paste(variables2, collapse = " + "), sep = " ~ "))
    tmp1 <- polr(f1, Hess = TRUE, data = eastpoly_gout %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp2 <- polr(f1, Hess = TRUE, data = westpoly_gout %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp3 <- polr(f2, Hess = TRUE, data = anz_gout_poly %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp4 <- polr(f2, Hess = TRUE, data = eurogout_poly %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    tmp5 <- polr(f2, Hess = TRUE, data = ardea_lasso_poly %>% filter(SEX == "Female", !(FLARE_CAT %in% c("0", "1"))) %>% mutate(FLARE_CAT = factor(FLARE_CAT, ordered = TRUE)))
    assign(paste0(modelname, i), (metagen(TE = c(tmp1$coefficients[[1]], 
                                                 tmp2$coefficients[[1]], 
                                                 tmp3$coefficients[[1]], 
                                                 tmp4$coefficients[[1]], 
                                                 tmp5$coefficients[[1]]), 
                                          seTE = c(summary(tmp1)$coefficients[1, 2], 
                                                   summary(tmp2)$coefficients[1, 2], 
                                                   summary(tmp3)$coefficients[1, 2], 
                                                   summary(tmp4)$coefficients[1, 2], 
                                                   summary(tmp5)$coefficients[1, 2]), 
                                          studlab = c("East Polynesian", "West Polynesian", "Aus/NZ", "EuroGout", "Ardea - LASSO"), sm = "OR")))
    tmplistfemale[[i]] <- get(paste0(modelname, i))
  }
  assign(paste0(modelname, "modlistfemale"), tmplistfemale, envir = .GlobalEnv)
}



clean_lapply <- function(modelname, functionname){
  c(unlist(lapply(get(paste0(modelname, "modlistmale")), functionname)), unlist(lapply(get(paste0(modelname, "modlistfemale")), functionname)))
}

OR <- function(x) {
  sprintf(exp(coef(x))[[2]], fmt = "%#.4f")
}

CI_OR <- function(x) {
  paste0("[", sprintf(exp(confint.default(x))[2, 1], fmt = "%#.4f"), ", ", sprintf(exp(confint.default(x))[2, 2], fmt = "%#.4f"), "]")
}

Pval <- function(x) {
  signif(summary(x)$coefficients[2, 4], 3)
}

OR_meta_fixed <- function(x) {
  sprintf(exp(x$TE.fixed), fmt = "%#.4f")
}

CI_OR_meta_fixed <- function(x) {
  paste0("[", sprintf(exp(x$lower.fixed), fmt = "%#.4f"), ", ", sprintf(exp(x$upper.fixed), fmt = "%#.4f"), "]")
}

Beta_meta_fixed <- function(x) {
  sprintf(x$TE.fixed, fmt = "%#.4f")
}

CI_Beta_meta_fixed <- function(x) {
  paste0("[", sprintf(x$lower.fixed, fmt = "%#.4f"), ", ", sprintf(x$upper.fixed, fmt = "%#.4f"), "]")
}

Pval_meta_fixed <- function(x) {
  signif(x$pval.fixed, 3)
}

OR_meta_random <- function(x) {
  sprintf(exp(x$TE.random), fmt = "%#.4f")
}

CI_OR_meta_random <- function(x) {
  paste0("[", sprintf(exp(x$lower.random), fmt = "%#.4f"), ", ", sprintf(exp(x$upper.random), fmt = "%#.4f"), "]")
}

Beta_meta_random <- function(x) {
  sprintf(x$TE.random, fmt = "%#.4f")
}

CI_Beta_meta_random <- function(x) {
  paste0("[", sprintf(x$lower.random, fmt = "%#.4f"), ", ", sprintf(x$upper.random, fmt = "%#.4f"), "]")
}

Pval_meta_random <- function(x) {
  signif(x$pval.random, 3)
}
```

```{r Assumptions}
# Linear regression has the following assumptions:
# 1. Linear relationship between variables (test this in scatter plot)
# 2. Normality of residuals
# 3. Homoscedasticity of residuals
# 4. No multicollinearity (for models with multiple predictors)

# Logistic regression has the following assumptions:
# 1. Linear relationship between the predictor and the log-odds (basically, percentage of cases should increase exponentially as the predictor increases)
# 2. No multicollinearity (for models with multiple predictors)

# These should be tested for all models, but I might be able to just test some representative models instead - the main thing is that extremely non-normal variables should not be used for linear regression



# First let's run a representative linear model of AGE1ATK vs PRS in the full male gout cohort
tmp <- all_pheno_prs %>% 
  filter(SEX == "Male",
         GOUT,
         Geno.SpecificAncestry %in% c("European", "Iberian", "European; Iberian"))

ggplot(tmp, aes(x = PRS1, y = AGE1ATK, color = COHORT2)) +
  geom_point(shape = 1) +
  geom_smooth(se = FALSE) # No evidence of non-linearity

mod <- lm(AGE1ATK ~ PRS1 + Geno.PCVector1 + Geno.PCVector2 + Geno.PCVector3 + Geno.PCVector4 + Geno.PCVector5 + Geno.PCVector6 + Geno.PCVector7 + Geno.PCVector8 + Geno.PCVector9 + Geno.PCVector10, data = tmp)

test <- augment(mod)

ggplot(test, aes(x = .fitted, y = .resid)) +
  geom_point() # no obvious pattern, so homoscedasticity seems to be met

ggplot(data = test, mapping = aes(x = .resid)) + 
  geom_histogram(mapping = aes(y = ..density..), bins = 30, fill = 'gray', color = 'black') +
  stat_function(fun = 'dnorm', 
                args = list(mean = mean(test$.resid), sd = sd(test$.resid)), 
                color = 'red')

ggplot(data = test, mapping = aes(sample = .resid)) +
  geom_qq() +
  geom_qq_line()

ggplot(data = test, mapping = aes(x = .resid)) +
  geom_boxplot() +
  theme(axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

test %>% 
  pull(.resid) %>% 
  stat.desc(basic = FALSE, desc = FALSE, norm = TRUE) %>% 
  enframe() %>% 
  pivot_wider(names_from = name, values_from = value)

# Normality of residuals is completely met here (ignore 2SE and shapiro-wilk test as this is large data)

vif(mod) # no multicollinearity (no values over 5 or 10)

# So we should have no problem running models of the PRS vs age at onset (as long as we don't include variables that exhibit multicollinearity)


# Now let's do some experiments with the flare phenotype
# First let's run a representative linear model of NUMATK vs PRS in the full male gout cohort
tmp <- tmp %>% 
  mutate(NUMATK = case_when(NUMATK > 52 ~ 52,
                            TRUE ~ NUMATK)) %>% 
  filter(NUMATK >= 2)

ggplot(tmp, aes(x = PRS1, y = NUMATK, color = COHORT2)) +
  geom_point(shape = 1) +
  geom_smooth(se = FALSE)

mod <- lm(NUMATK ~ PRS1 + Geno.PCVector1 + Geno.PCVector2 + Geno.PCVector3 + Geno.PCVector4 + Geno.PCVector5 + Geno.PCVector6 + Geno.PCVector7 + Geno.PCVector8 + Geno.PCVector9 + Geno.PCVector10, data = tmp)

test <- augment(mod)

ggplot(test, aes(x = .fitted, y = .resid)) +
  geom_point() # weird pattern, so homoscedasticity seems to not be met

ggplot(data = test, mapping = aes(x = .resid)) + 
  geom_histogram(mapping = aes(y = ..density..), bins = 30, fill = 'gray', color = 'black') +
  stat_function(fun = 'dnorm', 
                args = list(mean = mean(test$.resid), sd = sd(test$.resid)), 
                color = 'red')

ggplot(data = test, mapping = aes(sample = .resid)) +
  geom_qq() +
  geom_qq_line()

ggplot(data = test, mapping = aes(x = .resid)) +
  geom_boxplot() +
  theme(axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

test %>% 
  pull(.resid) %>% 
  stat.desc(basic = FALSE, desc = FALSE, norm = TRUE) %>% 
  enframe() %>% 
  pivot_wider(names_from = name, values_from = value)

# Normality of residuals is not met here (ignore 2SE and shapiro-wilk test as this is large data)

vif(mod) # no multicollinearity (no values over 5 or 10)

# So it is not appropriate to use the NUMATK variable in a linear regression model

# I will go ahead and test whether categorizing the variable (FLARE_CAT) and running an independent ANOVA or ordinal logistic regression model will be appropriate - note I am not sure how easy it will be to adjust for covariates in an ANOVA (perhaps I could run an ANCOVA but it might not be doing what I think it's doing)

# The ordinal logistic regression test should tell us the odds of being in any combination of higher flare categories vs the remainder of the categories (i.e. highest flare group vs all others, or highest 5 flare groups vs lowest flare group)

# I'll try running an example from a paper to see if I can replicate their results in R

# dat <- tibble("Group" = rep(c("Activator", "Headgear"), each = 50),
#               "Happiness" = factor(c(rep("Unhappy", 30), rep("Somewhat happy", 14), rep("Very happy", 6), rep("Unhappy", 14), rep("Somewhat happy", 25), rep("Very happy", 11)), levels = c("Unhappy", "Somewhat happy", "Very happy"), ordered = T))
# 
# mod <- MASS::polr(Happiness ~ Group, data = dat, Hess = TRUE)
# 
# modsum <- tibble("Group of interest" = row.names(summary(mod)$coefficients)[1],
#                  "OR" = exp(summary(mod)$coefficients)[1],
#                  "Lower CI" = exp(confint.default(mod))[1],
#                  "Upper CI" = exp(confint.default(mod))[2],
#                  "P-value" = pnorm(abs(summary(mod)$coefficients[1, "t value"]), lower.tail = FALSE) * 2)

# To interpret the odds ratio, we can say that for the group of children that wore Headgear, the odds of being more happy (i.e. being in a happiness tier or higher compared to a lower tier) were 3.27 times those of the the children that used an Activator 

# The parallel to this in our example would be that for those individuals with 1 unit more PRS than another group, the odds of being in a higher flare category (i.e. >= 3 flares vs < 3 flares) were X times those of the other group (i.e. those with 1 unit PRS lower - the reference group). Let's test this using an example model

mod <- polr(FLARE_CAT ~ PRS2, data = tmp, Hess = TRUE)

modsum <- tibble("Group of interest" = row.names(summary(mod)$coefficients)[1],
                 "OR" = exp(summary(mod)$coefficients)[1],
                 "Lower CI" = exp(confint.default(mod))[1],
                 "Upper CI" = exp(confint.default(mod))[2],
                 "P-value" = pnorm(abs(summary(mod)$coefficients[1, "t value"]), lower.tail = FALSE) * 2)

# This suggests that the model is significant for the full male gout cohort, with an OR of 1.11 [1.00, 1.24], p = 0.047 per PRS unit

# Can try to visualize this by looking at proportions of FLARE_CAT within bins of PRS

tmp %>% 
  filter(!is.na(PRS2)) %>% 
  mutate(PRS_bin = factor(ntile(PRS2, 6), ordered = T),
         FLARE_CAT = factor(case_when(is.na(FLARE_CAT) ~ "No Data",
                                      TRUE ~ as.character(FLARE_CAT)), 
                            levels = rev(c(paste0(0:5), 
                                       "6 - 11",
                                       "12 - 52",
                                       "No Data")),
                            ordered = TRUE)) %>%
  group_by(PRS_bin, FLARE_CAT, SEX, GROUP) %>% 
  summarize(value = n()) %>% 
  ggplot(aes(fill = FLARE_CAT, y = value, x = PRS_bin)) +
  geom_bar(position = "fill", stat = "identity") +
  facet_wrap(~ SEX * GROUP) +
  scale_fill_discrete(type = c("#C0C0C0", "#FDE725FF", "#9FDA3AFF", "#4AC16DFF", "#1FA187FF", "#277F8EFF", "#365C8DFF", "#46337EFF", "#440154FF")) +
  theme(axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.title = element_blank())

# The ANOVA test will essentially ask "does the mean PRS differ between flare categories"
# I should further note that I am unclear on how to meta-analyze ANOVA results - perhaps just running them within each cohort and determining how many show significant differences and, of those, which groups were different and in which direction? The other option is to pull cohorts but that runs into potentially weird issues of bias. So I think in the end I should run both models and see if I get a different answer, then decide how to present them in the paper.

# The assumptions of a one-way independent ANOVA are:

# 1. Independence of the datapoints (met)
# 2. Normality of model residuals within groups
# 3. Homogeneity of variance of model residuals within groups

mod <- tmp %>% 
  mutate(IID = factor(IID),
         FLARE_CAT = factor(FLARE_CAT)) %>% 
  ezANOVA(data = .,
               dv = PRS2,
               between = FLARE_CAT,
               wid = IID,
               type = 3,
               return_aov = TRUE)

mod

tmp %>% 
  ggplot(aes(x = FLARE_CAT, y = PRS2)) +
  geom_boxplot()

rm(tmp, mod, modsum, test)
```

```{r NewGoutModels}
# We need to model both PRS and all SNPs against gout, adjusting for global PCs. It needs to be in males and females separately, and needs to be both adjusted for age at collection and unadjusted. It also needs to run with the 10 oceanian PCs for Polynesian cohorts. We want the output of all of these models to be stored in a list object which we can then extract all of the important elements from in a table format.

OR <- function(x) {
  sprintf(exp(coef(x))[[2]], fmt = "%#.4f")
}

LCL_OR <- function(x) {
  sprintf(exp(confint.default(x))[2, 1], fmt = "%#.4f")
}

UCL_OR <- function(x) {
  sprintf(exp(confint.default(x))[2, 2], fmt = "%#.4f")
}

Pval <- function(x) {
  signif(summary(x)$coefficients[2, 4], 3)
}

gout_data_list <- list("UK Biobank - Male" = full_join(data_list[["UK Biobank - Gout - Male"]], 
                                                       data_list[["UK Biobank - Control - Male"]]),
                       "UK Biobank - Female" = full_join(data_list[["UK Biobank - Gout - Female"]], 
                                                         data_list[["UK Biobank - Control - Female"]]),
                       "Aus/NZ European - Male" = full_join(data_list[["Aus/NZ European - Gout - Male"]],
                                                            data_list[["Aus/NZ European - Control - Male"]]),
                       "Aus/NZ European - Female" = full_join(data_list[["Aus/NZ European - Gout - Female"]],
                                                              data_list[["Aus/NZ European - Control - Female"]]),
                       "GlobalGout - Male" = full_join(data_list[["GlobalGout - Gout - Male"]],
                                                       data_list[["GlobalGout - Control - Male"]]),
                       "GlobalGout - Female" = full_join(data_list[["GlobalGout - Gout - Female"]],
                                                         data_list[["GlobalGout - Control - Female"]]),
                       "East Polynesian - Male" = full_join(data_list[["East Polynesian - Gout - Male"]],
                                                            data_list[["East Polynesian - Control - Male"]]),
                       "East Polynesian - Female" = full_join(data_list[["East Polynesian - Gout - Female"]],
                                                              data_list[["East Polynesian - Control - Female"]]),
                       "West Polynesian - Male" = full_join(data_list[["West Polynesian - Gout - Male"]],
                                                            data_list[["West Polynesian - Control - Male"]]),
                       "West Polynesian - Female" = full_join(data_list[["West Polynesian - Gout - Female"]],
                                                              data_list[["West Polynesian - Control - Female"]]))

modlist <- vector("list", length(gout_data_list))
for(i in seq_along(gout_data_list)){
  prsnames <- gout_data_list[[i]] %>% 
    select(PRS1, PRS2, starts_with("rs")) %>% 
    select(where(~ !all(is.na(.x)))) %>% 
    colnames()
  
  covariates <- c()
  
  if(!str_detect(names(gout_data_list)[i], "UK Biobank")){
    covariates <- c(covariates, "Geno.PCVector1", "Geno.PCVector2", "Geno.PCVector3", "Geno.PCVector4", "Geno.PCVector5", "Geno.PCVector6", "Geno.PCVector7", "Geno.PCVector8", "Geno.PCVector9", "Geno.PCVector10")
  }
  
  if(str_detect(names(gout_data_list)[i], "Polynesian")){
    covariates <- c(covariates, "Geno.PCVector1_Oc", "Geno.PCVector2_Oc", "Geno.PCVector3_Oc", "Geno.PCVector4_Oc", "Geno.PCVector5_Oc", "Geno.PCVector6_Oc", "Geno.PCVector7_Oc", "Geno.PCVector8_Oc", "Geno.PCVector9_Oc", "Geno.PCVector10_Oc")
  }
  
  covariates2 <- c(covariates, "AGECOL")
  
  tmplist <- vector("list", 2 * length(prsnames))
  for(j in seq_along(prsnames)) {
    variables <- c(prsnames[j], covariates)
    f <- as.formula(paste("GOUT", paste(variables, collapse = " + "), sep = " ~ "))
    assign(paste0("Model_", i, "_", j), glm(f, family = binomial, data = gout_data_list[[i]]))
    # columns will be: Cohort, N, N case, N control, Outcome, Predictor, Covariates, OR, LCL, UCL, Pval, log-odds, standard error
    modstring <- c(names(gout_data_list)[[i]],
                   nrow(gout_data_list[[i]] %>% filter(!is.na(prsnames[[j]]))),
                   nrow(gout_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), GOUT)),
                   nrow(gout_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), !GOUT)),
                   "Gout",
                   prsnames[[j]],
                   paste(covariates, collapse = " + "),
                   OR(get(paste0("Model_", i, "_", j))),
                   LCL_OR(get(paste0("Model_", i, "_", j))),
                   UCL_OR(get(paste0("Model_", i, "_", j))),
                   Pval(get(paste0("Model_", i, "_", j))),
                   coef(get(paste0("Model_", i, "_", j)))[[2]],
                   summary(get(paste0("Model_", i, "_", j)))$coefficients[2, 2])
    tmplist[[j]] <- modstring
    
    variables <- c(prsnames[j], covariates2)
    f <- as.formula(paste("GOUT", paste(variables, collapse = " + "), sep = " ~ "))
    assign(paste0("Model_", i, "_", j, "_adj"), glm(f, family = binomial, data = gout_data_list[[i]]))
    modstring1 <- c(names(gout_data_list)[[i]],
                   nrow(gout_data_list[[i]] %>% filter(!is.na(prsnames[[j]]))),
                   nrow(gout_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), GOUT)),
                   nrow(gout_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), !GOUT)),
                   "Gout",
                   prsnames[[j]],
                   paste(covariates2, collapse = " + "),
                   OR(get(paste0("Model_", i, "_", j, "_adj"))),
                   LCL_OR(get(paste0("Model_", i, "_", j, "_adj"))),
                   UCL_OR(get(paste0("Model_", i, "_", j, "_adj"))),
                   Pval(get(paste0("Model_", i, "_", j, "_adj"))),
                   coef(get(paste0("Model_", i, "_", j, "_adj")))[[2]],
                   summary(get(paste0("Model_", i, "_", j, "_adj")))$coefficients[2, 2])
    tmplist[[j + length(prsnames)]] <- modstring1
  }
  
  modlist[[i]] <- tmplist
}

remove <- ls()
remove <- as_tibble(remove) %>% 
  filter(str_detect(value, "Model_"))
remove <- remove$value
rm(list = remove, remove)



out <- data.frame()
for(i in 1:length(modlist)){
  tmp <- modlist[[i]] %>% 
    map(as.data.frame) %>% 
    bind_cols() %>% 
    data.table::transpose()
  
  colnames(tmp) <- c("Cohort", "N", "N case", "N control", "Outcome", "Predictor", "Covariates", "OR", "LCL", "UCL", "Pval", "log-odds", "SE")
  
  out <- rbind(out, tmp)
}

GoutModels <- out

GoutModels <- GoutModels %>% 
  mutate(across(c(Cohort, Outcome, Predictor, Covariates), factor),
         across(c(N, `N case`, `N control`, OR, LCL, UCL, Pval, `log-odds`, SE), as.numeric))

#save(GoutModels, file = here("Output/GoutModels.RData"))

rm(modlist, out, tmp, tmplist, covariates, covariates2, f, i, j, modstring, modstring1, prsnames, variables)
```

```{r NewSeverityModels}
# We need to model both PRSs and all SNPs against age at onset, tophi, and flare category (by both ordinal logistic regression and ANOVA), adjusting for global PCs. It needs to be in males and females separately, and needs to be both adjusted for disease duration (for tophi) and unadjusted. It also needs to run with the 10 oceanian PCs for Polynesian cohorts. We want the output of all of these models to be stored in a list object which we can then extract all of the important elements from in a table format.

onset_data_list <- list("Aus/NZ European - Male" = data_list[["Aus/NZ European - Gout - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "Aus/NZ European - Female" = data_list[["Aus/NZ European - Gout - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "GlobalGout - Male" = data_list[["GlobalGout - Gout - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "GlobalGout - Female" = data_list[["GlobalGout - Gout - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - LASSO - Male" = data_list[["Ardea - LASSO - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - LASSO - Female" = data_list[["Ardea - LASSO - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - CLEAR1 - Male" = data_list[["Ardea - CLEAR1 - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - CLEAR1 - Female" = data_list[["Ardea - CLEAR1 - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - CLEAR2 - Male" = data_list[["Ardea - CLEAR2 - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - CLEAR2 - Female" = data_list[["Ardea - CLEAR2 - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - CRYSTAL - Male" = data_list[["Ardea - CRYSTAL - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - CRYSTAL - Female" = data_list[["Ardea - CRYSTAL - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - LIGHT - Male" = data_list[["Ardea - LIGHT - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "Ardea - LIGHT - Female" = data_list[["Ardea - LIGHT - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "East Polynesian - Male" = data_list[["East Polynesian - Gout - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "East Polynesian - Female" = data_list[["East Polynesian - Gout - Female"]] %>% filter(!is.na(AGE1ATK)),
                        "West Polynesian - Male" = data_list[["West Polynesian - Gout - Male"]] %>% filter(!is.na(AGE1ATK)),
                        "West Polynesian - Female" = data_list[["West Polynesian - Gout - Female"]] %>% filter(!is.na(AGE1ATK)))

Beta <- function(x) {
  sprintf(coef(x)[[2]], fmt = "%#.4f")
}

LCL <- function(x) {
  sprintf(confint.default(x)[2, 1], fmt = "%#.4f")
}

UCL <- function(x) {
  sprintf(confint.default(x)[2, 2], fmt = "%#.4f")
}

modlist <- vector("list", length(onset_data_list))
for(i in seq_along(onset_data_list)){
  prsnames <- onset_data_list[[i]] %>% 
    select(PRS1, PRS2, starts_with("rs")) %>% 
    select(where(~ !all(is.na(.x)))) %>% 
    colnames()
  
  covariates <- c("Geno.PCVector1", "Geno.PCVector2", "Geno.PCVector3", "Geno.PCVector4", "Geno.PCVector5", "Geno.PCVector6", "Geno.PCVector7", "Geno.PCVector8", "Geno.PCVector9", "Geno.PCVector10")
  
  if(str_detect(names(onset_data_list)[i], "Polynesian")){
    covariates <- c(covariates, "Geno.PCVector1_Oc", "Geno.PCVector2_Oc", "Geno.PCVector3_Oc", "Geno.PCVector4_Oc", "Geno.PCVector5_Oc", "Geno.PCVector6_Oc", "Geno.PCVector7_Oc", "Geno.PCVector8_Oc", "Geno.PCVector9_Oc", "Geno.PCVector10_Oc")
  }
  
  tmplist <- vector("list", length(prsnames))
  for(j in seq_along(prsnames)) {
    variables <- c(prsnames[j], covariates)
    f <- as.formula(paste("AGE1ATK", paste(variables, collapse = " + "), sep = " ~ "))
    assign(paste0("Model_", i, "_", j), lm(f, data = onset_data_list[[i]]))
    # columns will be: Cohort, N, Outcome, Predictor, Covariates, Beta, LCL, UCL, Pval, standard error
    modstring <- c(names(onset_data_list)[[i]],
                   nrow(onset_data_list[[i]] %>% filter(!is.na(prsnames[[j]]))),
                   "Age at Onset (years)",
                   prsnames[[j]],
                   paste(covariates, collapse = " + "),
                   Beta(get(paste0("Model_", i, "_", j))),
                   LCL(get(paste0("Model_", i, "_", j))),
                   UCL(get(paste0("Model_", i, "_", j))),
                   Pval(get(paste0("Model_", i, "_", j))),
                   summary(get(paste0("Model_", i, "_", j)))$coefficients[2, 2])
    tmplist[[j]] <- modstring
  }
  
  modlist[[i]] <- tmplist
}

remove <- ls()
remove <- as_tibble(remove) %>% 
  filter(str_detect(value, "Model_"))
remove <- remove$value
rm(list = remove, remove)



out <- data.frame()
for(i in 1:length(modlist)){
  tmp <- modlist[[i]] %>% 
    map(as.data.frame) %>% 
    bind_cols() %>% 
    data.table::transpose()
  
  colnames(tmp) <- c("Cohort", "N", "Outcome", "Predictor", "Covariates", "Beta", "LCL", "UCL", "Pval", "SE")
  
  out <- rbind(out, tmp)
}

OnsetModels <- out

OnsetModels <- OnsetModels %>% 
  mutate(across(c(Cohort, Outcome, Predictor, Covariates), factor),
         across(c(N, Beta, LCL, UCL, Pval, SE), as.numeric))

#save(OnsetModels, file = here("Output/OnsetModels.RData"))

rm(modlist, out, tmp, tmplist, covariates, f, i, j, modstring, prsnames, variables)



# Tophi
tophi_data_list <- list("Aus/NZ European - Male" = data_list[["Aus/NZ European - Gout - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Aus/NZ European - Female" = data_list[["Aus/NZ European - Gout - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "GlobalGout - Male" = data_list[["GlobalGout - Gout - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "GlobalGout - Female" = data_list[["GlobalGout - Gout - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - LASSO - Male" = data_list[["Ardea - LASSO - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - LASSO - Female" = data_list[["Ardea - LASSO - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - CLEAR1 - Male" = data_list[["Ardea - CLEAR1 - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - CLEAR1 - Female" = data_list[["Ardea - CLEAR1 - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - CLEAR2 - Male" = data_list[["Ardea - CLEAR2 - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - CLEAR2 - Female" = data_list[["Ardea - CLEAR2 - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - CRYSTAL - Male" = data_list[["Ardea - CRYSTAL - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - CRYSTAL - Female" = data_list[["Ardea - CRYSTAL - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - LIGHT - Male" = data_list[["Ardea - LIGHT - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "Ardea - LIGHT - Female" = data_list[["Ardea - LIGHT - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "East Polynesian - Male" = data_list[["East Polynesian - Gout - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "East Polynesian - Female" = data_list[["East Polynesian - Gout - Female"]] %>% filter(!is.na(TOPHIGOUT)),
                        "West Polynesian - Male" = data_list[["West Polynesian - Gout - Male"]] %>% filter(!is.na(TOPHIGOUT)),
                        "West Polynesian - Female" = data_list[["West Polynesian - Gout - Female"]] %>% filter(!is.na(TOPHIGOUT)))

modlist <- vector("list", length(tophi_data_list))
for(i in seq_along(tophi_data_list)){
  prsnames <- tophi_data_list[[i]] %>% 
    select(PRS1, PRS2, starts_with("rs")) %>% 
    select(where(~ !all(is.na(.x)))) %>% 
    colnames()
  
  covariates <- c("Geno.PCVector1", "Geno.PCVector2", "Geno.PCVector3", "Geno.PCVector4", "Geno.PCVector5", "Geno.PCVector6", "Geno.PCVector7", "Geno.PCVector8", "Geno.PCVector9", "Geno.PCVector10")
  
  
  if(str_detect(names(tophi_data_list)[i], "Polynesian")){
    covariates <- c(covariates, "Geno.PCVector1_Oc", "Geno.PCVector2_Oc", "Geno.PCVector3_Oc", "Geno.PCVector4_Oc", "Geno.PCVector5_Oc", "Geno.PCVector6_Oc", "Geno.PCVector7_Oc", "Geno.PCVector8_Oc", "Geno.PCVector9_Oc", "Geno.PCVector10_Oc")
  }
  
  covariates2 <- c(covariates, "DURATION")
  
  tmplist <- vector("list", 2 * length(prsnames))
  for(j in seq_along(prsnames)) {
    variables <- c(prsnames[j], covariates)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    assign(paste0("Model_", i, "_", j), glm(f, family = binomial, data = tophi_data_list[[i]]))
    # columns will be: Cohort, N, N case, N control, Outcome, Predictor, Covariates, OR, LCL, UCL, Pval, log-odds, standard error
    modstring <- c(names(tophi_data_list)[[i]],
                   nrow(tophi_data_list[[i]] %>% filter(!is.na(prsnames[[j]]))),
                   nrow(tophi_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), TOPHIGOUT)),
                   nrow(tophi_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), !TOPHIGOUT)),
                   "Presence of Tophi",
                   prsnames[[j]],
                   paste(covariates, collapse = " + "),
                   OR(get(paste0("Model_", i, "_", j))),
                   LCL_OR(get(paste0("Model_", i, "_", j))),
                   UCL_OR(get(paste0("Model_", i, "_", j))),
                   Pval(get(paste0("Model_", i, "_", j))),
                   coef(get(paste0("Model_", i, "_", j)))[[2]],
                   summary(get(paste0("Model_", i, "_", j)))$coefficients[2, 2])
    tmplist[[j]] <- modstring
    
    variables <- c(prsnames[j], covariates2)
    f <- as.formula(paste("TOPHIGOUT", paste(variables, collapse = " + "), sep = " ~ "))
    assign(paste0("Model_", i, "_", j, "_adj"), glm(f, family = binomial, data = tophi_data_list[[i]]))
    modstring1 <- c(names(tophi_data_list)[[i]],
                   nrow(tophi_data_list[[i]] %>% filter(!is.na(prsnames[[j]]))),
                   nrow(tophi_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), TOPHIGOUT)),
                   nrow(tophi_data_list[[i]] %>% filter(!is.na(prsnames[[j]]), !TOPHIGOUT)),
                   "Presence of Tophi",
                   prsnames[[j]],
                   paste(covariates2, collapse = " + "),
                   OR(get(paste0("Model_", i, "_", j, "_adj"))),
                   LCL_OR(get(paste0("Model_", i, "_", j, "_adj"))),
                   UCL_OR(get(paste0("Model_", i, "_", j, "_adj"))),
                   Pval(get(paste0("Model_", i, "_", j, "_adj"))),
                   coef(get(paste0("Model_", i, "_", j, "_adj")))[[2]],
                   summary(get(paste0("Model_", i, "_", j, "_adj")))$coefficients[2, 2])
    tmplist[[j + length(prsnames)]] <- modstring1
  }
  
  modlist[[i]] <- tmplist
}

remove <- ls()
remove <- as_tibble(remove) %>% 
  filter(str_detect(value, "Model_"))
remove <- remove$value
rm(list = remove, remove)



out <- data.frame()
for(i in 1:length(modlist)){
  tmp <- modlist[[i]] %>% 
    map(as.data.frame) %>% 
    bind_cols() %>% 
    data.table::transpose()
  
  colnames(tmp) <- c("Cohort", "N", "N case", "N control", "Outcome", "Predictor", "Covariates", "OR", "LCL", "UCL", "Pval", "log-odds", "SE")
  
  out <- rbind(out, tmp)
}

TophiModels <- out

TophiModels <- TophiModels %>% 
  mutate(across(c(Cohort, Outcome, Predictor, Covariates), factor),
         across(c(N, `N case`, `N control`, OR, LCL, UCL, Pval, `log-odds`, SE), as.numeric))

#save(TophiModels, file = here("Output/TophiModels.RData"))

rm(modlist, out, tmp, tmplist, covariates, covariates2, f, i, j, modstring, modstring1, prsnames, variables)


# Flare models
```


```{r MainSeverityModels, echo = FALSE}
onset_meta_euro(covariates = c(), PCs = FALSE, modelname = "metaONSET")

onset_meta_direct_euro(covariates = c(), PCs = FALSE, modelname = "metaONSETdirecteuro")

onset_meta_direct_poly(covariates = c(), PCs = FALSE, modelname = "metaONSETdirectpoly")

onset_meta_direct_both(covariates = c(), PCs = FALSE, modelname = "metaONSETdirectboth")

onset_meta_euro(covariates = c(), PCs = TRUE, modelname = "metaONSETadj")

onset_meta_direct_euro(covariates = c(), PCs = TRUE, modelname = "metaONSETdirecteuroadj")

onset_meta_direct_poly(covariates = c(), PCs = TRUE, modelname = "metaONSETdirectpolyadj")

onset_meta_direct_both(covariates = c(), PCs = TRUE, modelname = "metaONSETdirectbothadj")

tophi_meta_euro(covariates = c(), PCs = FALSE, modelname = "metaTOPHI")

tophi_meta_direct_euro(covariates = c(), PCs = FALSE, modelname = "metaTOPHIdirecteuro")

tophi_meta_direct_poly(covariates = c(), PCs = FALSE, modelname = "metaTOPHIdirectpoly")

tophi_meta_direct_both(covariates = c(), PCs = FALSE, modelname = "metaTOPHIdirectboth")

tophi_meta_euro(covariates = "DURATION", PCs = TRUE, modelname = "metaTOPHIadj")

tophi_meta_direct_euro(covariates = "DURATION", PCs = TRUE, modelname = "metaTOPHIdirecteuroadj")

tophi_meta_direct_poly(covariates = "DURATION", PCs = TRUE, modelname = "metaTOPHIdirectpolyadj")

tophi_meta_direct_both(covariates = "DURATION", PCs = TRUE, modelname = "metaTOPHIdirectbothadj")

flare_meta_euro1(covariates = c(), PCs = FALSE, modelname = "metaFLARE1")

flare_meta_direct_euro1(covariates = c(), PCs = FALSE, modelname = "metaFLARE1directeuro")

flare_meta_direct_poly1(covariates = c(), PCs = FALSE, modelname = "metaFLARE1directpoly")

flare_meta_direct_both1(covariates = c(), PCs = FALSE, modelname = "metaFLARE1directboth")

flare_meta_euro1(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE1adj")

flare_meta_direct_euro1(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE1directeuroadj")

flare_meta_direct_poly1(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE1directpolyadj") # Issue with female west poly model

flare_meta_direct_both1(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE1directbothadj") # Issue with female west poly model

flare_meta_euro2(covariates = c(), PCs = FALSE, modelname = "metaFLARE2")

flare_meta_direct_euro2(covariates = c(), PCs = FALSE, modelname = "metaFLARE2directeuro")

flare_meta_direct_poly2(covariates = c(), PCs = FALSE, modelname = "metaFLARE2directpoly")

flare_meta_direct_both2(covariates = c(), PCs = FALSE, modelname = "metaFLARE2directboth")

flare_meta_euro2(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE2adj")

flare_meta_direct_euro2(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE2directeuroadj")

flare_meta_direct_poly2(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE2directpolyadj") # Issue with female west poly model

flare_meta_direct_both2(covariates = "AGE1ATK", PCs = TRUE, modelname = "metaFLARE2directbothadj") # Issue with female west poly model



# Creating table of results
prsnames1 <- anz_gout %>% 
  select(PRS, starts_with("rs")) %>% 
  colnames()

x <- prsnames1 %>% 
  length()

prsnames2 <- anz_gout_poly %>% 
  select(PRS, starts_with("rs")) %>% 
  colnames()

y <- prsnames2 %>% 
  length()

NAstring1 <- rep(NA, x)

NAstring2 <- rep(NA, y)

OutcomeString <- c(rep("Age at onset (years)", 2*x),
                   rep("Age at onset (years)", 6*y),
                   rep("Age at onset (years)", 2*x),
                   rep("Age at onset (years)", 6*y),
                   rep("Presence of tophi", 2*x),
                   rep("Presence of tophi", 6*y),
                   rep("Presence of tophi", 2*x),
                   rep("Presence of tophi", 6*y),
                   rep("Higher flare frequency (categorical)", 2*x),
                   rep("Higher flare frequency (categorical)", 6*y),
                   rep("Higher flare frequency (categorical)", 2*x),
                   rep("Higher flare frequency (categorical)", 6*y),
                   rep("Higher flare frequency (categorical > 1)", 2*x),
                   rep("Higher flare frequency (categorical > 1)", 6*y),
                   rep("Higher flare frequency (categorical > 1)", 2*x),
                   rep("Higher flare frequency (categorical > 1)", 6*y))

PredictorString <- c(rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6),
                     rep(prsnames1, 2),
                     rep(prsnames2, 6))

CovariateString <- c(rep("None", 2*x + 6*y),
                     rep("10 Global PCs", 2*x + 2*y),
                     rep("10 Global PCs + 10 Oceanian PCs", 2*y),
                     rep("10 Global PCs", 2*y),
                     rep("None", 2*x + 6*y),
                     rep("Disease duration + 10 Global PCs", 2*x + 2*y),
                     rep("Disease duration + 10 Global PCs + 10 Oceanian PCs", 2*y),
                     rep("Disease duration + 10 Global PCs", 2*y),
                     rep("None", 2*x + 6*y),
                     rep("Age at onset + 10 Global PCs", 2*x + 2*y),
                     rep("Age at onset + 10 Global PCs + 10 Oceanian PCs", 2*y),
                     rep("Age at onset + 10 Global PCs", 2*y),
                     rep("None", 2*x + 6*y),
                     rep("Age at onset + 10 Global PCs", 2*x + 2*y),
                     rep("Age at onset + 10 Global PCs + 10 Oceanian PCs", 2*y),
                     rep("Age at onset + 10 Global PCs", 2*y))

CohortString <- rep(c(rep("European Gout", 2*x + 2*y),
                      rep("Polynesian Gout", 2*y),
                      rep("All Gout", 2*y)), 8)

BetaFixedString <- c(clean_lapply("metaONSET", Beta_meta_fixed),
                     clean_lapply("metaONSETdirecteuro", Beta_meta_fixed),
                     clean_lapply("metaONSETdirectpoly", Beta_meta_fixed),
                     clean_lapply("metaONSETdirectboth", Beta_meta_fixed),
                     clean_lapply("metaONSETadj", Beta_meta_fixed),
                     clean_lapply("metaONSETdirecteuroadj", Beta_meta_fixed),
                     clean_lapply("metaONSETdirectpolyadj", Beta_meta_fixed),
                     clean_lapply("metaONSETdirectbothadj", Beta_meta_fixed),
                     rep(NAstring1, 12),
                     rep(NAstring2, 36))

ORFixedString <- c(rep(NAstring1, 4),
                   rep(NAstring2, 12),
                   clean_lapply("metaTOPHI", OR_meta_fixed),
                   clean_lapply("metaTOPHIdirecteuro", OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectpoly", OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectboth", OR_meta_fixed),
                   clean_lapply("metaTOPHIadj", OR_meta_fixed),
                   clean_lapply("metaTOPHIdirecteuroadj", OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectpolyadj", OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectbothadj", OR_meta_fixed),
                   clean_lapply("metaFLARE1", OR_meta_fixed),
                   clean_lapply("metaFLARE1directeuro", OR_meta_fixed),
                   clean_lapply("metaFLARE1directpoly", OR_meta_fixed),
                   clean_lapply("metaFLARE1directboth", OR_meta_fixed),
                   clean_lapply("metaFLARE1adj", OR_meta_fixed),
                   clean_lapply("metaFLARE1directeuroadj", OR_meta_fixed),
                   clean_lapply("metaFLARE1directpolyadj", OR_meta_fixed),
                   clean_lapply("metaFLARE1directbothadj", OR_meta_fixed),
                   clean_lapply("metaFLARE2", OR_meta_fixed),
                   clean_lapply("metaFLARE2directeuro", OR_meta_fixed),
                   clean_lapply("metaFLARE2directpoly", OR_meta_fixed),
                   clean_lapply("metaFLARE2directboth", OR_meta_fixed),
                   clean_lapply("metaFLARE2adj", OR_meta_fixed),
                   clean_lapply("metaFLARE2directeuroadj", OR_meta_fixed),
                   clean_lapply("metaFLARE2directpolyadj", OR_meta_fixed),
                   clean_lapply("metaFLARE2directbothadj", OR_meta_fixed))

CIFixedString <- c(clean_lapply("metaONSET", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETdirecteuro", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETdirectpoly", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETdirectboth", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETadj", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETdirecteuroadj", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETdirectpolyadj", CI_Beta_meta_fixed),
                   clean_lapply("metaONSETdirectbothadj", CI_Beta_meta_fixed),
                   clean_lapply("metaTOPHI", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIdirecteuro", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectpoly", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectboth", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIadj", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIdirecteuroadj", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectpolyadj", CI_OR_meta_fixed),
                   clean_lapply("metaTOPHIdirectbothadj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1directeuro", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1directpoly", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1directboth", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1adj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1directeuroadj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1directpolyadj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE1directbothadj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2directeuro", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2directpoly", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2directboth", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2adj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2directeuroadj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2directpolyadj", CI_OR_meta_fixed),
                   clean_lapply("metaFLARE2directbothadj", CI_OR_meta_fixed))

PvalFixedString <- c(clean_lapply("metaONSET", Pval_meta_fixed),
                     clean_lapply("metaONSETdirecteuro", Pval_meta_fixed),
                     clean_lapply("metaONSETdirectpoly", Pval_meta_fixed),
                     clean_lapply("metaONSETdirectboth", Pval_meta_fixed),
                     clean_lapply("metaONSETadj", Pval_meta_fixed),
                     clean_lapply("metaONSETdirecteuroadj", Pval_meta_fixed),
                     clean_lapply("metaONSETdirectpolyadj", Pval_meta_fixed),
                     clean_lapply("metaONSETdirectbothadj", Pval_meta_fixed),
                     clean_lapply("metaTOPHI", Pval_meta_fixed),
                     clean_lapply("metaTOPHIdirecteuro", Pval_meta_fixed),
                     clean_lapply("metaTOPHIdirectpoly", Pval_meta_fixed),
                     clean_lapply("metaTOPHIdirectboth", Pval_meta_fixed),
                     clean_lapply("metaTOPHIadj", Pval_meta_fixed),
                     clean_lapply("metaTOPHIdirecteuroadj", Pval_meta_fixed),
                     clean_lapply("metaTOPHIdirectpolyadj", Pval_meta_fixed),
                     clean_lapply("metaTOPHIdirectbothadj", Pval_meta_fixed),
                     clean_lapply("metaFLARE1", Pval_meta_fixed),
                     clean_lapply("metaFLARE1directeuro", Pval_meta_fixed),
                     clean_lapply("metaFLARE1directpoly", Pval_meta_fixed),
                     clean_lapply("metaFLARE1directboth", Pval_meta_fixed),
                     clean_lapply("metaFLARE1adj", Pval_meta_fixed),
                     clean_lapply("metaFLARE1directeuroadj", Pval_meta_fixed),
                     clean_lapply("metaFLARE1directpolyadj", Pval_meta_fixed),
                     clean_lapply("metaFLARE1directbothadj", Pval_meta_fixed),
                     clean_lapply("metaFLARE2", Pval_meta_fixed),
                     clean_lapply("metaFLARE2directeuro", Pval_meta_fixed),
                     clean_lapply("metaFLARE2directpoly", Pval_meta_fixed),
                     clean_lapply("metaFLARE2directboth", Pval_meta_fixed),
                     clean_lapply("metaFLARE2adj", Pval_meta_fixed),
                     clean_lapply("metaFLARE2directeuroadj", Pval_meta_fixed),
                     clean_lapply("metaFLARE2directpolyadj", Pval_meta_fixed),
                     clean_lapply("metaFLARE2directbothadj", Pval_meta_fixed))

BetaRandomString <- c(clean_lapply("metaONSET", Beta_meta_random),
                      clean_lapply("metaONSETdirecteuro", Beta_meta_random),
                      clean_lapply("metaONSETdirectpoly", Beta_meta_random),
                      clean_lapply("metaONSETdirectboth", Beta_meta_random),
                      clean_lapply("metaONSETadj", Beta_meta_random),
                      clean_lapply("metaONSETdirecteuroadj", Beta_meta_random),
                      clean_lapply("metaONSETdirectpolyadj", Beta_meta_random),
                      clean_lapply("metaONSETdirectbothadj", Beta_meta_random),
                      rep(NAstring1, 12),
                      rep(NAstring2, 36))

ORRandomString <- c(rep(NAstring1, 4),
                    rep(NAstring2, 12),
                    clean_lapply("metaTOPHI", OR_meta_random),
                    clean_lapply("metaTOPHIdirecteuro", OR_meta_random),
                    clean_lapply("metaTOPHIdirectpoly", OR_meta_random),
                    clean_lapply("metaTOPHIdirectboth", OR_meta_random),
                    clean_lapply("metaTOPHIadj", OR_meta_random),
                    clean_lapply("metaTOPHIdirecteuroadj", OR_meta_random),
                    clean_lapply("metaTOPHIdirectpolyadj", OR_meta_random),
                    clean_lapply("metaTOPHIdirectbothadj", OR_meta_random),
                    clean_lapply("metaFLARE1", OR_meta_random),
                    clean_lapply("metaFLARE1directeuro", OR_meta_random),
                    clean_lapply("metaFLARE1directpoly", OR_meta_random),
                    clean_lapply("metaFLARE1directboth", OR_meta_random),
                    clean_lapply("metaFLARE1adj", OR_meta_random),
                    clean_lapply("metaFLARE1directeuroadj", OR_meta_random),
                    clean_lapply("metaFLARE1directpolyadj", OR_meta_random),
                    clean_lapply("metaFLARE1directbothadj", OR_meta_random),
                    clean_lapply("metaFLARE2", OR_meta_random),
                    clean_lapply("metaFLARE2directeuro", OR_meta_random),
                    clean_lapply("metaFLARE2directpoly", OR_meta_random),
                    clean_lapply("metaFLARE2directboth", OR_meta_random),
                    clean_lapply("metaFLARE2adj", OR_meta_random),
                    clean_lapply("metaFLARE2directeuroadj", OR_meta_random),
                    clean_lapply("metaFLARE2directpolyadj", OR_meta_random),
                    clean_lapply("metaFLARE2directbothadj", OR_meta_random))

CIRandomString <- c(clean_lapply("metaONSET", CI_Beta_meta_random),
                    clean_lapply("metaONSETdirecteuro", CI_Beta_meta_random),
                    clean_lapply("metaONSETdirectpoly", CI_Beta_meta_random),
                    clean_lapply("metaONSETdirectboth", CI_Beta_meta_random),
                    clean_lapply("metaONSETadj", CI_Beta_meta_random),
                    clean_lapply("metaONSETdirecteuroadj", CI_Beta_meta_random),
                    clean_lapply("metaONSETdirectpolyadj", CI_Beta_meta_random),
                    clean_lapply("metaONSETdirectbothadj", CI_Beta_meta_random),
                    clean_lapply("metaTOPHI", CI_OR_meta_random),
                    clean_lapply("metaTOPHIdirecteuro", CI_OR_meta_random),
                    clean_lapply("metaTOPHIdirectpoly", CI_OR_meta_random),
                    clean_lapply("metaTOPHIdirectboth", CI_OR_meta_random),
                    clean_lapply("metaTOPHIadj", CI_OR_meta_random),
                    clean_lapply("metaTOPHIdirecteuroadj", CI_OR_meta_random),
                    clean_lapply("metaTOPHIdirectpolyadj", CI_OR_meta_random),
                    clean_lapply("metaTOPHIdirectbothadj", CI_OR_meta_random),
                    clean_lapply("metaFLARE1", CI_OR_meta_random),
                    clean_lapply("metaFLARE1directeuro", CI_OR_meta_random),
                    clean_lapply("metaFLARE1directpoly", CI_OR_meta_random),
                    clean_lapply("metaFLARE1directboth", CI_OR_meta_random),
                    clean_lapply("metaFLARE1adj", CI_OR_meta_random),
                    clean_lapply("metaFLARE1directeuroadj", CI_OR_meta_random),
                    clean_lapply("metaFLARE1directpolyadj", CI_OR_meta_random),
                    clean_lapply("metaFLARE1directbothadj", CI_OR_meta_random),
                    clean_lapply("metaFLARE2", CI_OR_meta_random),
                    clean_lapply("metaFLARE2directeuro", CI_OR_meta_random),
                    clean_lapply("metaFLARE2directpoly", CI_OR_meta_random),
                    clean_lapply("metaFLARE2directboth", CI_OR_meta_random),
                    clean_lapply("metaFLARE2adj", CI_OR_meta_random),
                    clean_lapply("metaFLARE2directeuroadj", CI_OR_meta_random),
                    clean_lapply("metaFLARE2directpolyadj", CI_OR_meta_random),
                    clean_lapply("metaFLARE2directbothadj", CI_OR_meta_random))

PvalRandomString <- c(clean_lapply("metaONSET", Pval_meta_random),
                      clean_lapply("metaONSETdirecteuro", Pval_meta_random),
                      clean_lapply("metaONSETdirectpoly", Pval_meta_random),
                      clean_lapply("metaONSETdirectboth", Pval_meta_random),
                      clean_lapply("metaONSETadj", Pval_meta_random),
                      clean_lapply("metaONSETdirecteuroadj", Pval_meta_random),
                      clean_lapply("metaONSETdirectpolyadj", Pval_meta_random),
                      clean_lapply("metaONSETdirectbothadj", Pval_meta_random),
                      clean_lapply("metaTOPHI", Pval_meta_random),
                      clean_lapply("metaTOPHIdirecteuro", Pval_meta_random),
                      clean_lapply("metaTOPHIdirectpoly", Pval_meta_random),
                      clean_lapply("metaTOPHIdirectboth", Pval_meta_random),
                      clean_lapply("metaTOPHIadj", Pval_meta_random),
                      clean_lapply("metaTOPHIdirecteuroadj", Pval_meta_random),
                      clean_lapply("metaTOPHIdirectpolyadj", Pval_meta_random),
                      clean_lapply("metaTOPHIdirectbothadj", Pval_meta_random),
                      clean_lapply("metaFLARE1", Pval_meta_random),
                      clean_lapply("metaFLARE1directeuro", Pval_meta_random),
                      clean_lapply("metaFLARE1directpoly", Pval_meta_random),
                      clean_lapply("metaFLARE1directboth", Pval_meta_random),
                      clean_lapply("metaFLARE1adj", Pval_meta_random),
                      clean_lapply("metaFLARE1directeuroadj", Pval_meta_random),
                      clean_lapply("metaFLARE1directpolyadj", Pval_meta_random),
                      clean_lapply("metaFLARE1directbothadj", Pval_meta_random),
                      clean_lapply("metaFLARE2", Pval_meta_random),
                      clean_lapply("metaFLARE2directeuro", Pval_meta_random),
                      clean_lapply("metaFLARE2directpoly", Pval_meta_random),
                      clean_lapply("metaFLARE2directboth", Pval_meta_random),
                      clean_lapply("metaFLARE2adj", Pval_meta_random),
                      clean_lapply("metaFLARE2directeuroadj", Pval_meta_random),
                      clean_lapply("metaFLARE2directpolyadj", Pval_meta_random),
                      clean_lapply("metaFLARE2directbothadj", Pval_meta_random))

SexString <- rep(c(rep(rep(c("Male", "Female"), each = x), 1),
                   rep(rep(c("Male", "Female"), each = y), 3)), 8)

PRSString <- rep(c(rep("Imputed", 2*x),
                   rep("Directly genotyped", 6*y)), 8)

Meta3traitsvsPRS <- tibble("Outcome" = OutcomeString,
                           "Predictor" = PredictorString,
                           "PRS type" = PRSString,
                           "Covariates" = CovariateString, 
                           "Cohort" = CohortString,
                           "Sex" = SexString, 
                           "Beta (Fixed)" = BetaFixedString, 
                           "OR (Fixed)" = ORFixedString, 
                           "95% CI (Fixed)" = CIFixedString, 
                           "Pval (Fixed)" = PvalFixedString, 
                           "Beta (Random)" = BetaRandomString, 
                           "OR (Random)" = ORRandomString, 
                           "95% CI (Random)" = CIRandomString, 
                           "Pval (Random)" = PvalRandomString)
Meta3traitsvsPRS <- Meta3traitsvsPRS %>% mutate("Signif. (Fixed)" = case_when(`Pval (Fixed)` < (0.05 / nrow(UKBB_Gene_OR)) ~ "**", `Pval (Fixed)` >= (0.05 / nrow(UKBB_Gene_OR)) & `Pval (Fixed)` < 0.05 ~ "*", TRUE ~ "-"),
                                                "Signif. (Random)" = case_when(`Pval (Random)` < (0.05 / nrow(UKBB_Gene_OR)) ~ "**", `Pval (Random)` >= (0.05 / nrow(UKBB_Gene_OR)) & `Pval (Random)` < 0.05 ~ "*", TRUE ~ "-"))
save(Meta3traitsvsPRS, file = here("Output/Meta3traitsvsPRS.RData"))

remove <- ls()
remove <- as_tibble(remove) %>% 
  filter(str_detect(value, "modlist"))
remove <- remove$value
rm(list = remove, remove)
   
remove2 <- ls()
remove2 <- as_tibble(remove2) %>% 
  filter(str_detect(value, "String"))
remove2 <- remove2$value
rm(list = remove2, remove2, prsnames1, prsnames2, x, y)
```

```{r SigModels}
# All significant gout models:
GoutvsPRS_sig <- GoutvsPRS %>% 
  filter(`Signif.` == "**",
         Cohort != "UK Biobank")

table(GoutvsPRS_sig$Predictor)

# All significant severity models:
Meta3traitsvsPRS_sig <- Meta3traitsvsPRS %>% 
  filter(`Signif. (Fixed)` == "**" | `Signif. (Random)` == "**")
```

```{r Relationships, echo = FALSE}
# Now I want to determine the relationship between the three severity traits themselves alongside age and sex
# I will do this for each cohort independently
# First testing how sex and age affect each trait (which I kind of already tested in the models)
tmp1 <- lm(AGE1ATK ~ SEX, data = anz_gout)
tmp2 <- lm(AGE1ATK ~ SEX, data = eurogout)
tmp3 <- lm(AGE1ATK ~ SEX, data = ardea_clear_lasso)
tmp4 <- lm(AGE1ATK ~ SEX, data = ardea_lasso_other)
tmp5 <- lm(AGE1ATK ~ SEX, data = ardea_crystal)
tmp6 <- lm(AGE1ATK ~ SEX, data = ardea_light)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]], tmp3$coefficients[[2]], tmp4$coefficients[[2]], tmp5$coefficients[[2]], tmp6$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2], summary(tmp3)$coefficients[2,2], summary(tmp4)$coefficients[2,2], summary(tmp5)$coefficients[2,2], summary(tmp6)$coefficients[2,2]), studlab = c("Aus/NZ", "EuroGout", "Ardea - CLEAR + LASSO", "Ardea - LASSO (other)", "Ardea - CRYSTAL", "Ardea - LIGHT"))
summary(mod)
mod$pval.fixed # Females have around 13 years later age at onset on average (p = 7.6e-63)

# No need to test age vs age at onset (obviously correlated)


# Flare activity vs sex
tmp1 <- glm(FLAREHIGH ~ SEX, family = "binomial", data = anz_gout)
tmp2 <- glm(FLAREHIGH ~ SEX, family = "binomial", data = eurogout)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout"), sm = "OR")
summary(mod)
mod$pval.fixed # Non-significant reduced flare activity in females (p = 0.99)

# Flare activity vs age
tmp1 <- glm(FLAREHIGH ~ AGECOL, family = "binomial", data = anz_gout)
tmp2 <- glm(FLAREHIGH ~ AGECOL, family = "binomial", data = eurogout)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout"), sm = "OR")
summary(mod)
mod$pval.fixed # Reduced flare activity in older people (OR of 0.98 per year older, p = 7.0e-9)


# Tophi vs sex
tmp1 <- glm(TOPHIGOUT ~ SEX, family = "binomial", data = anz_gout)
tmp2 <- glm(TOPHIGOUT ~ SEX, family = "binomial", data = eurogout)
tmp3 <- glm(TOPHIGOUT ~ SEX, family = "binomial", data = ardea_clear_lasso)
tmp4 <- glm(TOPHIGOUT ~ SEX, family = "binomial", data = ardea_lasso_other)
tmp5 <- glm(TOPHIGOUT ~ SEX, family = "binomial", data = ardea_light)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]], tmp3$coefficients[[2]], tmp4$coefficients[[2]], tmp5$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2], summary(tmp3)$coefficients[2,2], summary(tmp4)$coefficients[2,2], summary(tmp5)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout", "Ardea - CLEAR + LASSO", "Ardea - LASSO (other)", "Ardea - LIGHT"), sm = "OR")
summary(mod)
mod$pval.fixed # Non-significant reduction in tophi presence in females (p = 0.66)

# Tophi vs age
tmp1 <- glm(TOPHIGOUT ~ AGECOL, family = "binomial", data = anz_gout)
tmp2 <- glm(TOPHIGOUT ~ AGECOL, family = "binomial", data = eurogout)
tmp3 <- glm(TOPHIGOUT ~ AGECOL, family = "binomial", data = ardea_clear_lasso)
tmp4 <- glm(TOPHIGOUT ~ AGECOL, family = "binomial", data = ardea_lasso_other)
tmp5 <- glm(TOPHIGOUT ~ AGECOL, family = "binomial", data = ardea_light)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]], tmp3$coefficients[[2]], tmp4$coefficients[[2]], tmp5$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2], summary(tmp3)$coefficients[2,2], summary(tmp4)$coefficients[2,2], summary(tmp5)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout", "Ardea - CLEAR + LASSO", "Ardea - LASSO (other)", "Ardea - LIGHT"), sm = "OR")
summary(mod) 
mod$pval.fixed # More tophi presence in older individuals (due to duration increase, OR 1.01, p = 1.5e-5)


# Now to test relationship between severity traits
# Flare activity vs age at onset
tmp1 <- glm(FLAREHIGH ~ AGE1ATK, family = "binomial", data = anz_gout)
tmp2 <- glm(FLAREHIGH ~ AGE1ATK, family = "binomial", data = eurogout)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout"), sm = "OR")
summary(mod)
mod$pval.fixed # Non-significant decrease in flare activity per year later age at onset (p = 0.10)

# Tophi positivity vs age at onset
tmp1 <- glm(TOPHIGOUT ~ AGE1ATK, family = "binomial", data = anz_gout)
tmp2 <- glm(TOPHIGOUT ~ AGE1ATK, family = "binomial", data = eurogout)
tmp3 <- glm(TOPHIGOUT ~ AGE1ATK, family = "binomial", data = ardea_clear_lasso)
tmp4 <- glm(TOPHIGOUT ~ AGE1ATK, family = "binomial", data = ardea_lasso_other)
tmp5 <- glm(TOPHIGOUT ~ AGE1ATK, family = "binomial", data = ardea_light)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]], tmp3$coefficients[[2]], tmp4$coefficients[[2]], tmp5$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2], summary(tmp3)$coefficients[2,2], summary(tmp4)$coefficients[2,2], summary(tmp5)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout", "Ardea - CLEAR + LASSO", "Ardea - LASSO (other)", "Ardea - LIGHT"), sm = "OR")
summary(mod) 
mod$pval.fixed # Decrease in tophus positivity per year later age at onset (OR = 0.99, p = 0.003)

# Flare activity vs tophi
tmp1 <- glm(FLAREHIGH ~ TOPHIGOUT, family = "binomial", data = anz_gout)
tmp2 <- glm(FLAREHIGH ~ TOPHIGOUT, family = "binomial", data = eurogout)

mod <- metagen(TE = c(tmp1$coefficients[[2]], tmp2$coefficients[[2]]), seTE = c(summary(tmp1)$coefficients[2,2], summary(tmp2)$coefficients[2,2]), studlab = c("Aus/NZ", "GlobalGout"), sm = "OR")
summary(mod)
mod$pval.fixed # Flare activity significantly increases with presence of tophi (OR = 1.81, p = 7.4e-9)


# Cleaning up
rm(tmp1, tmp2, tmp3, tmp4, tmp5, tmp6, mod)
```

```{r TophusGWAS}
alltophi <- allgout %>% 
  filter(!is.na(TOPHIGOUT)) %>% 
  mutate(FID = IID) %>% 
  select(FID, IID, TOPHIGOUT, COHORT2, DURATION, SEX, SURICACID:PC10)

write_delim(alltophi, delim = "\t", file = here("Output/TophiGWAS/tophi.covar"))

system(paste0('source ~/.bashrc; parallel "plink1.9b6.10 --vcf /Volumes/archive/merrimanlab/raid_backup/New_Zealand_Chip_data/CoreExome/QC_MergedBatches/Imputed_Genotypes/QC1-10_Impute_EUR_only/CZ-MB1.2-QC1.10_EUR_imputed_chr{}.vcf.gz --make-bed --out ', here("Output/TophiGWAS/"), 'matched_snps_chr{}.txt" ::: {1..22}'))

system(paste0('source ~/.bashrc; plink1.9b6.10 --bfile ', here("Output/TophiGWAS/"), 'merged_poly_multi --logistic sex --ci 0.95 --covar ', here("Output/TophiGWAS/", "tophi.covar"), ' --covar-name PC1-PC10 --out ', here("Output/TophiGWAS/"), 'allgout'))
```

```{r Oldcode}
anz_gout_poly <- anz_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "Aus/NZ",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)

anz_control <- control_anz_euro_pheno_prs %>%
  mutate(GOUT = FALSE,
         SEX = factor(Pheno.Gender, levels = c("Female", "Male")),
         SURICACID = Pheno.CollectionUrate / 0.05949) %>% 
  rename(IID = Pheno.SampleID,
         AGECOL = Pheno.Age,
         COHORT = Pheno.Study,
         PC1 = Geno.PCVector1,
         PC2 = Geno.PCVector2,
         PC3 = Geno.PCVector3,
         PC4 = Geno.PCVector4,
         PC5 = Geno.PCVector5,
         PC6 = Geno.PCVector6,
         PC7 = Geno.PCVector7,
         PC8 = Geno.PCVector8,
         PC9 = Geno.PCVector9,
         PC10 = Geno.PCVector10) %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, PRS:rs738408)

anz_control_poly <- control_anz_euro_pheno_prs_poly %>%
  mutate(GOUT = FALSE,
         SEX = factor(Pheno.Gender, levels = c("Female", "Male")),
         SURICACID = Pheno.CollectionUrate / 0.05949) %>% 
  rename(IID = Pheno.SampleID,
         AGECOL = Pheno.Age,
         COHORT = Pheno.Study,
         PC1 = Geno.PCVector1,
         PC2 = Geno.PCVector2,
         PC3 = Geno.PCVector3,
         PC4 = Geno.PCVector4,
         PC5 = Geno.PCVector5,
         PC6 = Geno.PCVector6,
         PC7 = Geno.PCVector7,
         PC8 = Geno.PCVector8,
         PC9 = Geno.PCVector9,
         PC10 = Geno.PCVector10) %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, PRS:rs738409, -rs35332062)

tmp <- anz_gout %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, PRS:rs738408)

anz_full <- rbind(tmp, anz_control)

tmp <- anz_gout_poly %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, PRS:rs738409)

anz_full_poly <- rbind(tmp, anz_control_poly)


eurogout <- eurogout_euro_pheno_prs %>% 
  mutate(COHORT2 = "EuroGout",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID)

eurogout_poly <- eurogout_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "EuroGout",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)


ardea_clear1 <- clear1_euro_pheno_prs %>% 
  mutate(COHORT2 = "Ardea - CLEAR1",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID)

ardea_clear1_poly <- clear1_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "Ardea - CLEAR1",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)


ardea_clear2 <- clear2_euro_pheno_prs %>% 
  mutate(COHORT2 = "Ardea - CLEAR2",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID)

ardea_clear2_poly <- clear2_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "Ardea - CLEAR2",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)


ardea_lasso <- lasso_euro_pheno_prs %>% 
  mutate(COHORT2 = "Ardea - LASSO",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID)

ardea_lasso_poly <- lasso_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "Ardea - LASSO",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)


ardea_crystal <- crystal_euro_pheno_prs %>% 
  mutate(COHORT2 = "Ardea - CRYSTAL",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID)

ardea_crystal_poly <- crystal_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "Ardea - CRYSTAL",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)


ardea_light <- light_euro_pheno_prs %>% 
  mutate(COHORT2 = "Ardea - LIGHT",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID)

ardea_light_poly <- light_euro_pheno_prs_poly %>% 
  mutate(COHORT2 = "Ardea - LIGHT",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)
  

allgout <- rbind(anz_gout, eurogout, ardea_clear1, ardea_clear2, ardea_crystal, ardea_lasso, ardea_light)

allgout_poly <- rbind(anz_gout_poly, eurogout_poly, ardea_clear1_poly, ardea_clear2_poly, ardea_crystal_poly, ardea_lasso_poly, ardea_light_poly)

rm(anz_euro_pheno_prs, clear1_euro_pheno_prs, clear2_euro_pheno_prs, control_anz_euro_pheno_prs, crystal_euro_pheno_prs, eurogout_euro_pheno_prs, lasso_euro_pheno_prs, light_euro_pheno_prs, anz_euro_pheno_prs_poly, clear1_euro_pheno_prs_poly, clear2_euro_pheno_prs_poly, control_anz_euro_pheno_prs_poly, crystal_euro_pheno_prs_poly, eurogout_euro_pheno_prs_poly, lasso_euro_pheno_prs_poly, light_euro_pheno_prs_poly, tmp)


# Polynesian cohort prep
eastpoly_gout <- eastpoly_pheno_prs %>% 
  mutate(COHORT2 = "East Polynesian",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)

eastpoly_control <- control_eastpoly_pheno_prs %>%
  mutate(GOUT = FALSE,
         SEX = factor(Pheno.Gender, levels = c("Female", "Male")),
         SURICACID = Pheno.CollectionUrate / 0.05949) %>% 
  rename(IID = Pheno.SampleID,
         AGECOL = Pheno.Age,
         COHORT = Pheno.Study,
         PC1 = Geno.PCVector1,
         PC2 = Geno.PCVector2,
         PC3 = Geno.PCVector3,
         PC4 = Geno.PCVector4,
         PC5 = Geno.PCVector5,
         PC6 = Geno.PCVector6,
         PC7 = Geno.PCVector7,
         PC8 = Geno.PCVector8,
         PC9 = Geno.PCVector9,
         PC10 = Geno.PCVector10,
         PC1_Oc = Geno.PCVector1_Oc,
         PC2_Oc = Geno.PCVector2_Oc,
         PC3_Oc = Geno.PCVector3_Oc,
         PC4_Oc = Geno.PCVector4_Oc,
         PC5_Oc = Geno.PCVector5_Oc,
         PC6_Oc = Geno.PCVector6_Oc,
         PC7_Oc = Geno.PCVector7_Oc,
         PC8_Oc = Geno.PCVector8_Oc,
         PC9_Oc = Geno.PCVector9_Oc,
         PC10_Oc = Geno.PCVector10_Oc) %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, Geno.PolynesianGroup, PC1_Oc:PC10_Oc, PRS:rs738409, -rs35332062)

tmp <- eastpoly_gout %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, Geno.PolynesianGroup, PC1_Oc:PC10_Oc, PRS:rs738409)

eastpoly_full <- rbind(tmp, eastpoly_control)

westpoly_gout <- westpoly_pheno_prs %>% 
  mutate(COHORT2 = "West Polynesian",
         NUMATK = round(NUMATK),
         FLARE_CAT = factor(case_when(between(NUMATK, 0, 5) ~ paste0(as.character(NUMATK), " flares in last year"),
                                      between(NUMATK, 6, 11) ~ "One every one to two months", 
                                      between(NUMATK, 12, 52) ~ "One or more per month"),
                            levels = c(paste0(0:5, " flares in last year"), "One every one to two months", "One or more per month"),
                            labels = c(paste0(0:5), "6 - 11", "12 - 52"),
                            ordered = TRUE)) %>% 
  select(-Geno.SampleID, -rs35332062)

westpoly_control <- control_westpoly_pheno_prs %>%
  mutate(GOUT = rep(FALSE, nrow(.)),
         SEX = factor(Pheno.Gender, levels = c("Female", "Male")),
         SURICACID = Pheno.CollectionUrate / 0.05949) %>% 
  rename(IID = Pheno.SampleID,
         AGECOL = Pheno.Age,
         COHORT = Pheno.Study,
         PC1 = Geno.PCVector1,
         PC2 = Geno.PCVector2,
         PC3 = Geno.PCVector3,
         PC4 = Geno.PCVector4,
         PC5 = Geno.PCVector5,
         PC6 = Geno.PCVector6,
         PC7 = Geno.PCVector7,
         PC8 = Geno.PCVector8,
         PC9 = Geno.PCVector9,
         PC10 = Geno.PCVector10,
         PC1_Oc = Geno.PCVector1_Oc,
         PC2_Oc = Geno.PCVector2_Oc,
         PC3_Oc = Geno.PCVector3_Oc,
         PC4_Oc = Geno.PCVector4_Oc,
         PC5_Oc = Geno.PCVector5_Oc,
         PC6_Oc = Geno.PCVector6_Oc,
         PC7_Oc = Geno.PCVector7_Oc,
         PC8_Oc = Geno.PCVector8_Oc,
         PC9_Oc = Geno.PCVector9_Oc,
         PC10_Oc = Geno.PCVector10_Oc) %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, Geno.PolynesianGroup, PC1_Oc:PC10_Oc, PRS:rs738409, -rs35332062)

tmp <- westpoly_gout %>% 
  select(IID, COHORT, SEX, AGECOL, GOUT, SURICACID, PC1:PC10, Geno.PolynesianGroup, PC1_Oc:PC10_Oc, PRS:rs738409)

westpoly_full <- rbind(tmp, westpoly_control)

allgout_poly2 <- allgout_poly %>% 
  full_join(eastpoly_gout) %>% 
  full_join(westpoly_gout)

rm(control_eastpoly_pheno_prs, control_westpoly_pheno_prs, eastpoly_pheno_prs, westpoly_pheno_prs, tmp)


# UK Biobank prep
ukbb_gout <- ukbb_euro_pheno_prs %>% 
  filter(GOUT == TRUE,
         !is.na(PRS)) %>% 
  mutate(NUMATK = NA,
         TOPHIGOUT = NA,
         AGE1ATK = NA,
         DURATION = NA,
         PC1 = NA,
         PC2 = NA,
         PC3 = NA,
         PC4 = NA,
         PC5 = NA,
         PC6 = NA,
         PC7 = NA,
         PC8 = NA,
         PC9 = NA,
         PC10 = NA) %>% 
  select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10, PRS:rs738408)

ukbb_control <- ukbb_euro_pheno_prs %>% 
  filter(GOUT == FALSE,
         !is.na(PRS)) %>%
  mutate(NUMATK = NA,
         TOPHIGOUT = NA,
         AGE1ATK = NA,
         DURATION = NA,
         PC1 = NA,
         PC2 = NA,
         PC3 = NA,
         PC4 = NA,
         PC5 = NA,
         PC6 = NA,
         PC7 = NA,
         PC8 = NA,
         PC9 = NA,
         PC10 = NA,
         ULT = NA) %>% 
  select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10, PRS:rs738408)

ukbb <- rbind(ukbb_gout, ukbb_control)

rm(ukbb_euro_pheno_prs)

ukbb_gout_poly <- ukbb_poly_pheno_prs %>% 
  filter(GOUT == TRUE,
         !is.na(PRS)) %>% 
  mutate(NUMATK = NA,
         TOPHIGOUT = NA,
         AGE1ATK = NA,
         DURATION = NA,
         PC1 = NA,
         PC2 = NA,
         PC3 = NA,
         PC4 = NA,
         PC5 = NA,
         PC6 = NA,
         PC7 = NA,
         PC8 = NA,
         PC9 = NA,
         PC10 = NA) %>% 
  select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10, PRS:rs738409, -rs35332062)

ukbb_control_poly <- ukbb_poly_pheno_prs %>% 
  filter(GOUT == FALSE,
         !is.na(PRS)) %>%
  mutate(NUMATK = NA,
         TOPHIGOUT = NA,
         AGE1ATK = NA,
         DURATION = NA,
         PC1 = NA,
         PC2 = NA,
         PC3 = NA,
         PC4 = NA,
         PC5 = NA,
         PC6 = NA,
         PC7 = NA,
         PC8 = NA,
         PC9 = NA,
         PC10 = NA,
         ULT = NA) %>% 
  select(IID, NUMATK, TOPHIGOUT, AGE1ATK, DURATION, COHORT, SEX, AGECOL, GOUT, SURICACID, ULT, PC1:PC10, PRS:rs738409, -rs35332062)

ukbb_poly <- rbind(ukbb_gout_poly, ukbb_control_poly)

rm(ukbb_poly_pheno_prs)
```

